---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-"
)
```

# TOmicsVis
## 1. Introduction
**TOmicsVis**: TranscriptOmics Visualization.

**Github**: [https://github.com/benben-miao/TOmicsVis/](https://github.com/benben-miao/TOmicsVis/)

**Website**: [https://benben-miao.github.io/TOmicsVis/](https://benben-miao.github.io/TOmicsVis/)

**Author**: [benben-miao github](https://github.com/benben-miao)

**BioSciTools**: [Biology Science Tools](https://github.com/BioSciTools/BioSciTools.github.io)

## 2. Install package
```{r}
# 1. TOmicsVis
# install.packages("devtools")
# devtools::install_github("benben-miao/TOmicsVis")
library(TOmicsVis)

# 2. Extra package
# install.packages("ggplot2")
library(ggplot2)
```

## 3. Usage cases
### 3.1 Samples Statistics
#### 3.1.1 quantile_plot
Quantile plot for visualizing data distribution.
```{r quantile_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load box_data example datasets
data(quantile_data)

# 2. Run quantile_plot plot function
quantile_plot(
  quantile_data,
  my_shape = "fill_circle",
  point_size = 1.5,
  conf_int = TRUE,
  conf_level = 0.95,
  split_panel = "One_Panel",
  legend_pos = "right",
  legend_dir = "vertical",
  sci_fill_color = "Sci_AAAS",
  sci_color_alpha = 0.75,
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::quantile_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html](https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::quantile_plot
```

#### 3.1.2 corr_heatmap
Correlation Heatmap for samples/groups based on Pearson algorithm.
```{r corr_heatmap, fig.width=10.00, fig.height=6.18}
# 1. Load gene_exp example dataset
data(gene_exp)
head(gene_exp)

# 2. Run corr_heatmap plot function
corr_heatmap(
  gene_exp,
  corr_method = "pearson",
  cell_shape = "square",
  fill_type = "full",
  lable_size = 3,
  lable_digits = 3,
  color_low = "blue",
  color_mid = "white",
  color_high = "red",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::corr_heatmap` or reference page [https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html](https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::corr_heatmap
```

#### 3.1.3 pca_plot
PCA dimensional reduction visualization for RNA-Seq.
```{r pca_plot, fig.width=10.00, fig.height=6.18}
# 1. Load pca_sample_gene and pca_group_sample example datasets
data(pca_sample_gene)
data(pca_group_sample)

# 2. Run pca_plot plot function
pca_plot(
  pca_sample_gene,
  pca_group_sample,
  point_size = 5,
  text_size = 5,
  ellipse_alpha = 0.3,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::pca_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html](https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::pca_plot
```

### 3.2 Traits Analysis
#### 3.2.1 box_plot
Box plot support two levels and multiple groups with P value.
```{r box_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load box_data example datasets
data(box_data)

# 2. Run box_plot plot function
box_plot(
  box_data,
  test_method = "wilcox.test",
  test_label = "p.format",
  notch = TRUE,
  group_level = "Three_Column",
  add_element = "dotplot",
  my_shape = "fill_circle",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.5,
  sci_color_alpha = 1,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::box_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/box_plot.html](https://benben-miao.github.io/TOmicsVis/reference/box_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::box_plot
```

#### 3.2.2 violin_plot
Violin plot support two levels and multiple groups with P value.
```{r violin_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load box_data example datasets
data(box_data)

# 2. Run violin_plot plot function
violin_plot(
  box_data,
  test_method = "wilcox.test",
  test_label = "p.format",
  group_level = "Three_Column",
  violin_orientation = "vertical",
  add_element = "boxplot",
  element_alpha = 0.5,
  my_shape = "plus_times",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.5,
  sci_color_alpha = 1,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::violin_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html](https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::violin_plot
```

#### 3.2.3 survival_plot
Survival plot for analyzing and visualizing survival data.
```{r survival_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load survival_plot example datasets
data(survival_data)

# 2. Run survival_plot plot function
survival_plot(
  survival_data,
  curve_function = "pct",
  conf_inter = TRUE,
  interval_style = "ribbon",
  risk_table = TRUE,
  num_censor = TRUE,
  sci_palette = "aaas",
  ggTheme = "theme_light",
  x_start = 0,
  y_start = 0,
  y_end = 100,
  x_break = 100,
  y_break = 25
)
```

Get help using command `?TOmicsVis::survival_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/survival_plot.html](https://benben-miao.github.io/TOmicsVis/reference/survival_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::survival_plot
```

#### 3.2.4 tsne_plot
TSNE plot for analyzing and visualizing TSNE algorithm.
```{r tsne_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load tsne_plot example datasets
data(tsne_data)

# 2. Run tsne_plot plot function
tsne_plot(
  tsne_data,
  seed = 5,
  point_size = 4,
  point_alpha = 0.8,
  text_size = 2,
  text_alpha = 0.8,
  ci_level = 0.95,
  ellipse_alpha = 0.3,
  sci_fill_color = "Sci_JAMA",
  sci_color_alpha = 0.9,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::tsne_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/tsne_plot.html](https://benben-miao.github.io/TOmicsVis/reference/tsne_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::tsne_plot
```

### 3.3 Differential Expression Analyais
#### 3.3.1 venn_plot
Venn plot for stat common and unique gene among multiple sets.
```{r venn_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load venn_data example datasets
data(venn_data)

# 2. Run venn_plot plot function
venn_plot(
  venn_data,
  line_type = "blank",
  ellipse_shape = "circle",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.65
)
```

Get help using command `?TOmicsVis::venn_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html](https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::venn_plot
```

#### 3.3.2 circos_heatmap
Circos heatmap plot for visualizing gene expressing in multiple samples.
```{r circos_heatmap, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load circos_heatmap_data example datasets
data(circos_heatmap_data)

# 2. Run circos_heatmap plot function
circos_heatmap(
  circos_heatmap_data,
  low_color = "#0000ff",
  mid_color = "#ffffff",
  high_color = "#ff0000",
  gap_size = 10,
  cluster_method = "complete",
  distance_method = "euclidean",
  dend_height = 0.2,
  rowname_size = 0.8
)
```

Get help using command `?TOmicsVis::circos_heatmap` or reference page [https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html](https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::circos_heatmap
```

#### 3.3.3 volcano_plot
Volcano plot for visualizing differentailly expressed genes.
```{r volcano_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load deg_data example datasets
data(deg_data)

# 2. Run volcano_plot plot function
volcano_plot(
  deg_data,
  log2fc_cutoff = 1,
  pq_value = "pvalue",
  pq_cutoff = 0.005,
  cutoff_line = "longdash",
  point_shape = "large_circle",
  point_size = 1,
  point_alpha = 0.5,
  color_normal = "#888888",
  color_log2fc = "#008000",
  color_pvalue = "#0088ee",
  color_Log2fc_p = "#ff0000",
  label_size = 3,
  boxed_labels = FALSE,
  draw_connectors = FALSE,
  legend_pos = "right"
)
```

Get help using command `?TOmicsVis::volcano_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/volcano_plot.html](https://benben-miao.github.io/TOmicsVis/reference/volcano_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::volcano_plot
```

#### 3.3.4 ma_plot
MversusA plot for visualizing differentially expressed genes.
```{r ma_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load deg_data example datasets
data(deg_data2)

# 2. Run volcano_plot plot function
ma_plot(
  deg_data2,
  foldchange = 2,
  fdr_value = 0.05,
  point_size = 0.5,
  color_up = "#FF0000",
  color_down = "#008800",
  color_alpha = 0.5,
  top_method = "fc",
  top_num = 20,
  label_size = 8,
  label_box = TRUE,
  title = "Group1 -versus- Group2",
  xlab = "Log2 mean expression",
  ylab = "Log2 fold change",
  ggTheme = "theme_minimal"
)
```

Get help using command `?TOmicsVis::ma_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/ma_plot.html](https://benben-miao.github.io/TOmicsVis/reference/ma_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::ma_plot
```

#### 3.3.5 heatmap_group
Heatmap group for visualizing grouped gene expression data.
```{r heatmap_group, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(heatmap_group_data)
head(heatmap_group_data)

# 2. Run heatmap_group plot function
ma_plot(
  deg_data2,
  foldchange = 2,
  fdr_value = 0.05,
  point_size = 0.5,
  color_up = "#FF0000",
  color_down = "#008800",
  color_alpha = 0.5,
  top_method = "fc",
  top_num = 20,
  label_size = 8,
  label_box = TRUE,
  title = "Group1 -versus- Group2",
  xlab = "Log2 mean expression",
  ylab = "Log2 fold change",
  ggTheme = "theme_minimal"
)
```

Get help using command `?TOmicsVis::heatmap_group` or reference page [https://benben-miao.github.io/TOmicsVis/reference/heatmap_group.html](https://benben-miao.github.io/TOmicsVis/reference/heatmap_group.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::heatmap_group
```

### 3.4 Advanced Analysis
#### 3.4.1 trend_plot
Trend plot for visualizing gene expression trend profile in multiple traits.
```{r trend_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load chord_data example datasets
data(trend_data)

# 2. Run trend_plot plot function
trend_plot(
  trend_data,
  scale_method = "globalminmax",
  miss_value = "exclude",
  line_alpha = 0.5,
  show_points = TRUE,
  show_boxplot = TRUE,
  num_column = 2,
  xlab = "Traits",
  ylab = "Genes Expression",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.8,
  sci_color_alpha = 0.8,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::trend_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/trend_plot.html](https://benben-miao.github.io/TOmicsVis/reference/trend_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::trend_plot
```

#### 3.4.2 wgcna_pipeline
WGCNA analysis pipeline for RNA-Seq.
```{r wgcna_pipeline, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load wgcna_pipeline example datasets
data(wgcna_gene_exp)
data(wgcna_sample_group)

# 2. Run wgcna_pipeline plot function
# wgcna_pipeline(wgcna_gene_exp, wgcna_sample_group)
```

Get help using command `?TOmicsVis::wgcna_pipeline` or reference page [https://benben-miao.github.io/TOmicsVis/reference/wgcna_pipeline.html](https://benben-miao.github.io/TOmicsVis/reference/wgcna_pipeline.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::wgcna_pipeline
```

#### 3.4.3 network_plot
Network plot for analyzing and visualizing relationship of genes.
```{r network_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(network_data)
head(network_data)

# 2. Run network_plot plot function
network_plot(
  network_data,
  calcBy = "degree",
  degreeValue = 0.05,
  nodeColorNormal = "#00888888",
  nodeBorderColor = "#FFFFFF",
  nodeColorFrom = "#FF000088",
  nodeColorTo = "#00880088",
  nodeShapeNormal = "circle",
  nodeShapeSpatial = "csquare",
  nodeSize = 10,
  labelSize = 0.5,
  edgeCurved = TRUE,
  netLayout = "layout_on_sphere"
)
```

Get help using command `?TOmicsVis::network_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/network_plot.html](https://benben-miao.github.io/TOmicsVis/reference/network_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::network_plot
```

### 3.5 GO and KEGG Enrichment
#### 3.5.1 chord_plot
Chord plot for visualizing the relationships of pathways and genes.
```{r chord_plot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load chord_data example datasets
data(chord_data)

# 2. Run chord_plot plot function
chord_plot(
  chord_data,
  multi_colors = "RainbowColors",
  color_alpha = 0.5,
  link_visible = TRUE,
  link_dir = -1,
  link_type = "diffHeight",
  sector_scale = "Origin",
  width_circle = 3,
  dist_name = 3,
  label_dir = "Vertical",
  dist_label = 0.3
)
```

Get help using command `?TOmicsVis::chord_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html](https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::chord_plot
```

#### 3.5.2 go_enrich
GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).
```{r go_enrich, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(go_anno)
head(go_anno)

data(go_deg_fc)
head(go_deg_fc)

# 2. Run go_enrich analysis function
res <- go_enrich(
  go_anno,
  go_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 0.5,
  qvalue_cutoff = 0.5
)
head(res)
```

Get help using command `?TOmicsVis::go_enrich` or reference page [https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html](https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::go_enrich
```

#### 3.5.3 go_enrich_stat
GO enrichment analysis and stat plot based on GO annotation results (None/Exist Reference Genome).
```{r go_enrich_stat, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(go_anno)
# head(go_anno)

data(go_deg_fc)
# head(go_deg_fc)

# 2. Run go_enrich_stat analysis function
go_enrich_stat(
  go_anno,
  go_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 0.5,
  qvalue_cutoff = 0.5,
  max_go_item = 15,
  strip_fill = "#CDCDCD",
  xtext_angle = 45,
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.8,
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::go_enrich_stat` or reference page [https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html](https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::go_enrich_stat
```

#### 3.5.4 go_enrich_bar
GO enrichment analysis and bar plot based on GO annotation results (None/Exist Reference Genome).
```{r go_enrich_bar, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(go_anno)
# head(go_anno)

data(go_deg_fc)
# head(go_deg_fc)

# 2. Run go_enrich_bar analysis function
go_enrich_bar(
  go_anno,
  go_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 0.5,
  qvalue_cutoff = 0.5,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008888aa",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::go_enrich_bar` or reference page [https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html](https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::go_enrich_bar
```

#### 3.5.5 go_enrich_dot
GO enrichment analysis and dot plot based on GO annotation results (None/Exist Reference Genome).
```{r go_enrich_dot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(go_anno)
# head(go_anno)

data(go_deg_fc)
# head(go_deg_fc)

# 2. Run go_enrich_dot analysis function
go_enrich_dot(
  go_anno,
  go_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 0.5,
  qvalue_cutoff = 0.5,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008888aa",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::go_enrich_dot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html](https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::go_enrich_dot
```

#### 3.5.6 go_enrich_tree
GO enrichment analysis and tree plot based on GO annotation results (None/Exist Reference Genome).
```{r go_enrich_tree, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(go_anno)
# head(go_anno)

data(go_deg_fc)
# head(go_deg_fc)

# 2. Run go_enrich_tree analysis function
go_enrich_tree(
  go_anno,
  go_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 0.5,
  qvalue_cutoff = 0.5,
  sign_by = "p.adjust",
  category_num = 20,
  font_size = 4,
  low_color = "#ff0000aa",
  high_color = "#008888aa",
  hclust_method = "complete",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::go_enrich_tree` or reference page [https://benben-miao.github.io/TOmicsVis/reference/go_enrich_tree.html](https://benben-miao.github.io/TOmicsVis/reference/go_enrich_tree.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::go_enrich_tree
```

#### 3.5.7 go_enrich_net
GO enrichment analysis and net plot based on GO annotation results (None/Exist Reference Genome).
```{r go_enrich_net, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(go_anno)
# head(go_anno)

data(go_deg_fc)
# head(go_deg_fc)

# 2. Run go_enrich_net analysis function
go_enrich_net(
  go_anno,
  go_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 0.5,
  qvalue_cutoff = 0.5,
  category_num = 20,
  net_layout = "circle",
  net_circular = TRUE,
  low_color = "#ff0000aa",
  high_color = "#008888aa"
)
```

Get help using command `?TOmicsVis::go_enrich_net` or reference page [https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html](https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::go_enrich_net
```

#### 3.5.8 kegg_enrich
KEGG enrichment analysis based on KEGG annotation results (None/Exist Reference Genome).
```{r kegg_enrich, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(kegg_anno)
head(kegg_anno)

data(kegg_deg_fc)
head(kegg_deg_fc)

# 2. Run go_enrich analysis function
res <- kegg_enrich(
  kegg_anno,
  kegg_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 1,
  qvalue_cutoff = 1
)
head(res)
```

Get help using command `?TOmicsVis::kegg_enrich` or reference page [https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html](https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::kegg_enrich
```

#### 3.5.9 kegg_enrich_bar
KEGG enrichment analysis and bar plot based on KEGG annotation results (None/Exist Reference Genome).
```{r kegg_enrich_bar, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(kegg_anno)
# head(kegg_anno)

data(kegg_deg_fc)
# head(kegg_deg_fc)

# 2. Run kegg_enrich_bar analysis function
kegg_enrich_bar(
  kegg_anno,
  kegg_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 1,
  qvalue_cutoff = 1,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008888aa",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::kegg_enrich_bar` or reference page [https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html](https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::kegg_enrich_bar
```

#### 3.5.10 kegg_enrich_dot
KEGG enrichment analysis and dot plot based on KEGG annotation results (None/Exist Reference Genome).
```{r kegg_enrich_dot, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(kegg_anno)
# head(kegg_anno)

data(kegg_deg_fc)
# head(kegg_deg_fc)

# 2. Run kegg_enrich_dot analysis function
kegg_enrich_dot(
  kegg_anno,
  kegg_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 1,
  qvalue_cutoff = 1,
  sign_by = "p.adjust",
  category_num = 30,
  font_size = 12,
  low_color = "#ff0000aa",
  high_color = "#008888aa",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::kegg_enrich_dot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html](https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::kegg_enrich_dot
```

#### 3.5.11 kegg_enrich_tree
KEGG enrichment analysis and tree plot based on KEGG annotation results (None/Exist Reference Genome).
```{r kegg_enrich_tree, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(kegg_anno)
# head(kegg_anno)

data(kegg_deg_fc)
# head(kegg_deg_fc)

# 2. Run kegg_enrich_tree analysis function
kegg_enrich_tree(
  kegg_anno,
  kegg_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 1,
  qvalue_cutoff = 1,
  sign_by = "p.adjust",
  category_num = 20,
  font_size = 4,
  low_color = "#ff0000aa",
  high_color = "#008888aa",
  hclust_method = "complete",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::kegg_enrich_tree` or reference page [https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_tree.html](https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_tree.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::kegg_enrich_tree
```

#### 3.5.12 kegg_enrich_net
KEGG enrichment analysis and net plot based on KEGG annotation results (None/Exist Reference Genome).
```{r kegg_enrich_net, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(kegg_anno)
# head(kegg_anno)

data(kegg_deg_fc)
# head(kegg_deg_fc)

# 2. Run kegg_enrich_net analysis function
kegg_enrich_net(
  kegg_anno,
  kegg_deg_fc,
  padjust_method = "fdr",
  pvalue_cutoff = 1,
  qvalue_cutoff = 1,
  category_num = 20,
  net_layout = "circle",
  net_circular = TRUE,
  low_color = "#ff0000aa",
  high_color = "#008888aa"
)
```

Get help using command `?TOmicsVis::kegg_enrich_net` or reference page [https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html](https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::kegg_enrich_net
```

### 3.6 Tables Operations
#### 3.6.1 table_split
Table split used for splitting a grouped column to multiple columns.
```{r table_split, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load table_split_data example datasets
data(table_split_data)
head(table_split_data)

# 2. Run table_split plot function
res <- table_split(table_split_data, 
                  grouped_var = "variable", 
                  miss_drop = TRUE
                  )
head(res)
```

Get help using command `?TOmicsVis::table_split` or reference page [https://benben-miao.github.io/TOmicsVis/reference/table_split.html](https://benben-miao.github.io/TOmicsVis/reference/table_split.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::table_split
```

#### 3.6.2 table_merge
Table merge used to merge multiple variables to on variable.
```{r table_merge, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(table_merge_data)
head(table_merge_data)

# 2. Run function
res <- table_merge(
  table_merge_data,
  merge_vars = c("Ozone", "Solar.R", "Wind", "Temp"),
  new_var = "Variable",
  new_value = "Value",
  na_remove = FALSE
)
head(res)
```

Get help using command `?TOmicsVis::table_merge` or reference page [https://benben-miao.github.io/TOmicsVis/reference/table_merge.html](https://benben-miao.github.io/TOmicsVis/reference/table_merge.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::table_merge
```

#### 3.6.3 table_filter
Table filter used to filter row by column condition.
```{r table_filter, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(table_filter_data)
head(table_filter_data)

# 2. Run function
res <- table_filter(table_filter_data, 
                    height > 100 & eye_color == "black"
                    )
head(res)
```

Get help using command `?TOmicsVis::table_filter` or reference page [https://benben-miao.github.io/TOmicsVis/reference/table_filter.html](https://benben-miao.github.io/TOmicsVis/reference/table_filter.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::table_filter
```

#### 3.6.4 table_cross
Table cross used to cross search and merge results in two tables.
```{r table_cross, warning=FALSE, fig.width=10.00, fig.height=6.18}
# 1. Load example datasets
data(table_cross_data1)
head(table_cross_data1)

data(table_cross_data2)
head(table_cross_data2)

# 2. Run function
res <- table_cross(
  table_cross_data1,
  table_cross_data2,
  inter_var = "geneID",
  left_index = TRUE,
  right_index = FALSE
)
head(res)
```

Get help using command `?TOmicsVis::table_cross` or reference page [https://benben-miao.github.io/TOmicsVis/reference/table_cross.html](https://benben-miao.github.io/TOmicsVis/reference/table_cross.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::table_cross
```
