---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-"
)
```

# TOmicsVis
## 1. Introduction
**TOmicsVis**: TranscriptOmics Visualization.

**Github**: [https://github.com/benben-miao/TOmicsVis/](https://github.com/benben-miao/TOmicsVis/)

**Website**: [https://benben-miao.github.io/TOmicsVis/](https://benben-miao.github.io/TOmicsVis/)

**Author**: [benben-miao github](https://github.com/benben-miao)

**BioSciTools**: [Biology Science Tools](https://github.com/BioSciTools/BioSciTools.github.io)

## 2. Install package
```{r}
# 1. TOmicsVis
# install.packages("devtools")
# devtools::install_github("benben-miao/TOmicsVis")
library(TOmicsVis)

# 2. Extra package
# install.packages("ggplot2")
library(ggplot2)
```

## 3. Usage cases
### 3.1 corr_heatmap
Correlation Heatmap for samples/groups based on Pearson algorithm.
```{r corr_heatmap}
# 1. Load gene_exp example dataset
data(gene_exp)
head(gene_exp)

# 2. Run corr_heatmap plot function
corr_heatmap(
  gene_exp,
  corr_method = "pearson",
  cell_shape = "square",
  fill_type = "full",
  lable_size = 3,
  lable_digits = 3,
  color_low = "blue",
  color_mid = "white",
  color_high = "red",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::corr_heatmap` or reference page [https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html](https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::corr_heatmap
```

### 3.2 pca_plot
PCA dimensional reduction visualization for RNA-Seq.
```{r pca_plot}
# 1. Load pca_sample_gene and pca_group_sample example datasets
data(pca_sample_gene)
data(pca_group_sample)

# 2. Run pca_plot plot function
pca_plot(
  pca_sample_gene,
  pca_group_sample,
  point_size = 5,
  text_size = 5,
  ellipse_alpha = 0.3,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::pca_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html](https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::pca_plot
```

### 3.3 box_plot
Box plot support two levels and multiple groups with P value.
```{r box_plot, warning=FALSE}
# 1. Load box_data example datasets
data(box_data)

# 2. Run box_plot plot function
box_plot(
  box_data,
  test_method = "wilcox.test",
  test_label = "p.format",
  notch = TRUE,
  group_level = "Three_Column",
  add_element = "dotplot",
  my_shape = "fill_circle",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.5,
  sci_color_alpha = 1,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::box_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/box_plot.html](https://benben-miao.github.io/TOmicsVis/reference/box_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::box_plot
```

### 3.4 violin_plot
Violin plot support two levels and multiple groups with P value.
```{r violin_plot, warning=FALSE}
# 1. Load box_data example datasets
data(box_data)

# 2. Run violin_plot plot function
violin_plot(
  box_data,
  test_method = "wilcox.test",
  test_label = "p.format",
  group_level = "Three_Column",
  violin_orientation = "vertical",
  add_element = "boxplot",
  element_alpha = 0.5,
  my_shape = "plus_times",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.5,
  sci_color_alpha = 1,
  legend_pos = "right",
  legend_dir = "vertical",
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::violin_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html](https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::violin_plot
```

### 3.5 quantile_plot
Quantile plot for visualizing data distribution.
```{r quantile_plot, warning=FALSE}
# 1. Load box_data example datasets
data(quantile_data)

# 2. Run quantile_plot plot function
quantile_plot(
  quantile_data,
  my_shape = "fill_circle",
  point_size = 1.5,
  conf_int = TRUE,
  conf_level = 0.95,
  split_panel = "One_Panel",
  legend_pos = "right",
  legend_dir = "vertical",
  sci_fill_color = "Sci_AAAS",
  sci_color_alpha = 0.75,
  ggTheme = "theme_light"
)
```

Get help using command `?TOmicsVis::quantile_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html](https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::quantile_plot
```

### 3.6 venn_plot
Venn plot for stat common and unique gene among multiple sets.
```{r venn_plot, warning=FALSE}
# 1. Load venn_data example datasets
data(venn_data)

# 2. Run venn_plot plot function
venn_plot(
  venn_data,
  line_type = "blank",
  ellipse_shape = "circle",
  sci_fill_color = "Sci_AAAS",
  sci_fill_alpha = 0.65
)
```

Get help using command `?TOmicsVis::venn_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html](https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::venn_plot
```

### 3.7 chord_plot
Chord plot for visualizing the relationships of pathways and genes.
```{r chord_plot, warning=FALSE}
# 1. Load chord_data example datasets
data(chord_data)

# 2. Run chord_plot plot function
chord_plot(
  chord_data,
  multi_colors = "RainbowColors",
  color_alpha = 0.5,
  link_visible = TRUE,
  link_dir = -1,
  link_type = "diffHeight",
  sector_scale = "Origin",
  width_circle = 3,
  dist_name = 3,
  label_dir = "Vertical",
  dist_label = 0.3
)
```

Get help using command `?TOmicsVis::chord_plot` or reference page [https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html](https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::chord_plot
```

### 3.8 circos_heatmap
Circos heatmap plot for visualizing gene expressing in multiple samples.
```{r circos_heatmap, warning=FALSE}
# 1. Load circos_heatmap_data example datasets
data(circos_heatmap_data)

# 2. Run circos_heatmap plot function
circos_heatmap(
  circos_heatmap_data,
  low_color = "#0000ff",
  mid_color = "#ffffff",
  high_color = "#ff0000",
  gap_size = 10,
  cluster_method = "complete",
  distance_method = "euclidean",
  dend_height = 0.2,
  rowname_size = 0.8
)
```

Get help using command `?TOmicsVis::circos_heatmap` or reference page [https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html](https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html).
```{r}
# Get help with command in R console.
# ?TOmicsVis::circos_heatmap
```
