<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content='Transcriptome visualization from sample trait statistics to gene expression analysis. Six categories include "Samples Statistics", "Traits Analysis", "Differential Expression Analysis", "Advanced Analysis", "GO and KEGG Enrichment", "Tables Operations", with complete sample data.'>
<title>Transcriptome Visualization Process Scheme • TOmicsVis</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="deps/JetBrains_Mono-0.4.7/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Transcriptome Visualization Process Scheme">
<meta property="og:description" content='Transcriptome visualization from sample trait statistics to gene expression analysis. Six categories include "Samples Statistics", "Traits Analysis", "Differential Expression Analysis", "Advanced Analysis", "GO and KEGG Enrichment", "Tables Operations", with complete sample data.'>
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">TOmicsVis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">
    <span class="fa fa-passport"></span>
     
    Reference
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--articles">
    <span class="fa fa-file-code"></span>
     
    Articles
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--articles">
    <h6 class="dropdown-header" data-toc-skip>Get Start</h6>
    <a class="external-link dropdown-item" href="https://mp.weixin.qq.com/s/g8sRcK_ExlsOFniMWEJnVQ">
      <span class="fa fa-weixin"></span>
       
      LongArticle
    </a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Tutorials</h6>
    <a class="dropdown-item" href="articles/Tutorials.html">
      <span class="fa fa-file-code"></span>
       
      Tutorials
    </a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Blog</h6>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">
    <span class="fa fa-code-branch"></span>
     
    Versions
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--courses">
    <span class="fa fa-film"></span>
     
    Courses
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--courses">
    <h6 class="dropdown-header" data-toc-skip>Platforms</h6>
    <a class="external-link dropdown-item" href="https://space.bilibili.com/34105515/channel/series">
      <span class="fa fa-youtube"></span>
       
      BiliBili
    </a>
    <a class="external-link dropdown-item" href="https://mp.weixin.qq.com/s/mVXJmHPAnC9J1-zMj7eG7g">
      <span class="fa fa-weixin"></span>
       
      WeChat
    </a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/OmicsSuite/">
    <span class="fa fa-dna"></span>
     
    OmicsSuite
  </a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="https://github.com/benben-miao/TOmicsVis/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="tomicsvis">TOmicsVis<a class="anchor" aria-label="anchor" href="#tomicsvis"></a>
</h1></div>
<div class="section level2">
<h2 id="id_1-introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#id_1-introduction"></a>
</h2>
<p><strong>TOmicsVis</strong>: TranscriptOmics Visualization.</p>
<p><strong>Website</strong>: <a href="https://benben-miao.github.io/TOmicsVis/" class="external-link uri">https://benben-miao.github.io/TOmicsVis/</a></p>
<div class="section level4">
<h4 id="id_11-tomicsvis-shinyapp">1.1 TOmicsVis Shinyapp<a class="anchor" aria-label="anchor" href="#id_11-tomicsvis-shinyapp"></a>
</h4>
<p><strong>1.1.1 Local start funcion:</strong></p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Start shiny application.</span></span>
<span><span class="fu">TOmicsVis</span><span class="fu">::</span><span class="fu"><a href="reference/tomicsvis.html">tomicsvis</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<figure><img src="https://benben-miao.github.io/TOmicsVis/TOmicsVis-Shinyapp.png" alt="TOmicsVis Shinyapp"><figcaption aria-hidden="true">
TOmicsVis Shinyapp
</figcaption></figure><p><strong>1.1.2 Online cloud platform:</strong> <a href="https://shiny.hiplot.cn/tomicsvis-shiny/" class="external-link uri">https://shiny.hiplot.cn/tomicsvis-shiny/</a></p>
</div>
<div class="section level4">
<h4 id="id_12-github-and-cran-install">1.2 Github and CRAN Install<a class="anchor" aria-label="anchor" href="#id_12-github-and-cran-install"></a>
</h4>
<p><a href="https://cran.r-project.org/package=TOmicsVis" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/grand-total/TOmicsVis?color=orange"></a></p>
<p><strong>1.2.1 Install required packages from Bioconductor:</strong></p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install required packages from Bioconductor</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ComplexHeatmap"</span>, <span class="st">"EnhancedVolcano"</span>, <span class="st">"clusterProfiler"</span>, <span class="st">"enrichplot"</span>, <span class="st">"impute"</span>, <span class="st">"preprocessCore"</span>, <span class="st">"Mfuzz"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>1.2.2 Github</strong>: <a href="https://github.com/benben-miao/TOmicsVis/" class="uri">https://github.com/benben-miao/TOmicsVis/</a></p>
<p><strong>Install from Github:</strong></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"benben-miao/TOmicsVis"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Resolve network by GitClone</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_git.html" class="external-link">install_git</a></span><span class="op">(</span><span class="st">"https://gitclone.com/github.com/benben-miao/TOmicsVis.git"</span><span class="op">)</span></span></code></pre></div>
<p><strong>1.2.3 CRAN</strong>: <a href="https://cran.r-project.org/package=TOmicsVis" class="external-link uri">https://cran.r-project.org/package=TOmicsVis</a></p>
<p><strong>Install from CRAN:</strong></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Install from CRAN</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"TOmicsVis"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_13-articles-and-courses">1.3 Articles and Courses<a class="anchor" aria-label="anchor" href="#id_13-articles-and-courses"></a>
</h4>
<p><strong>Videos Courses:</strong> <a href="https://space.bilibili.com/34105515/channel/series" class="external-link uri">https://space.bilibili.com/34105515/channel/series</a></p>
<p><strong>Article Introduction:</strong> <a href="https://mp.weixin.qq.com/s/g8sRcK_ExlsOFniMWEJnVQ" class="external-link">全解TOmicsVis完美应用于转录组可视化R包</a></p>
<p><strong>Article Courses:</strong> <a href="https://mp.weixin.qq.com/s/mVXJmHPAnC9J1-zMj7eG7g" class="external-link">TOmicsVis 转录组学R代码分析及可视化视频</a></p>
</div>
<div class="section level4">
<h4 id="id_14-about-and-authors">1.4 About and Authors<a class="anchor" aria-label="anchor" href="#id_14-about-and-authors"></a>
</h4>
<p><strong>OmicsSuite</strong>: <a href="https://github.com/omicssuite/" class="external-link">Omics Suite Github: https://github.com/omicssuite/</a></p>
<p><strong>Authors</strong>:</p>
<ul>
<li><p><a href="https://github.com/benben-miao/" class="external-link">benben-miao Github: https://github.com/benben-miao/</a></p></li>
<li><p><a href="https://github.com/dongwei1220/" class="external-link">dongwei1220 Github: https://github.com/dongwei1220/</a></p></li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="id_2-libary-packages">2. Libary packages<a class="anchor" aria-label="anchor" href="#id_2-libary-packages"></a>
</h2>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Library TOmicsVis package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://benben-miao.github.io/TOmicsVis/" class="external-link">TOmicsVis</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; 载入需要的程辑包：Biobase</span></span>
<span><span class="co">#&gt; 载入需要的程辑包：BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 载入程辑包：'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; 载入需要的程辑包：e1071</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span></span>
<span><span class="co">#&gt;   method from   </span></span>
<span><span class="co">#&gt;   +.gg   ggplot2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 载入程辑包：'DynDoc'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     path</span></span>
<span></span>
<span><span class="co"># 2. Extra package</span></span>
<span><span class="co"># install.packages("ggplot2")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="id_3-usage-cases">3. Usage cases<a class="anchor" aria-label="anchor" href="#id_3-usage-cases"></a>
</h2>
<div class="section level3">
<h3 id="id_31-samples-statistics">3.1 Samples Statistics<a class="anchor" aria-label="anchor" href="#id_31-samples-statistics"></a>
</h3>
<div class="section level4">
<h4 id="id_311-quantile_plot">3.1.1 quantile_plot<a class="anchor" aria-label="anchor" href="#id_311-quantile_plot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: Weight and Sex traits dataframe (1st-col: Weight, 2nd-col: Sex).</p>
<p><strong>Output Plot:</strong> Quantile plot for visualizing data distribution.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">weight_sex</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">weight_sex</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Weight    Sex</span></span>
<span><span class="co">#&gt; 1  36.74 Female</span></span>
<span><span class="co">#&gt; 2  38.54 Female</span></span>
<span><span class="co">#&gt; 3  44.91 Female</span></span>
<span><span class="co">#&gt; 4  43.53 Female</span></span>
<span><span class="co">#&gt; 5  39.03 Female</span></span>
<span><span class="co">#&gt; 6  26.01 Female</span></span>
<span></span>
<span><span class="co"># 2. Run quantile_plot plot function</span></span>
<span><span class="fu"><a href="reference/quantile_plot.html">quantile_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">weight_sex</span>,</span>
<span>  my_shape <span class="op">=</span> <span class="st">"fill_circle"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  conf_int <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  conf_level <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  split_panel <span class="op">=</span> <span class="st">"Split_Panel"</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_NPG"</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">0.75</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-quantile_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/quantile_plot.html">?TOmicsVis::quantile_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html</a>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::quantile_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_312-box_plot">3.1.2 box_plot<a class="anchor" aria-label="anchor" href="#id_312-box_plot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: Length, Width, Weight, and Sex traits dataframe (1st-col: Value, 2nd-col: Traits, 3rd-col: Sex).</p>
<p><strong>Output Plot:</strong> Plot: Box plot support two levels and multiple groups with P value.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">traits_sex</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">traits_sex</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Value Traits    Sex</span></span>
<span><span class="co">#&gt; 1 36.74 Weight Female</span></span>
<span><span class="co">#&gt; 2 38.54 Weight Female</span></span>
<span><span class="co">#&gt; 3 44.91 Weight Female</span></span>
<span><span class="co">#&gt; 4 43.53 Weight Female</span></span>
<span><span class="co">#&gt; 5 39.03 Weight Female</span></span>
<span><span class="co">#&gt; 6 26.01 Weight Female</span></span>
<span></span>
<span><span class="co"># 2. Run box_plot plot function</span></span>
<span><span class="fu"><a href="reference/box_plot.html">box_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">traits_sex</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"t.test"</span>,</span>
<span>  test_label <span class="op">=</span> <span class="st">"p.format"</span>,</span>
<span>  notch <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  group_level <span class="op">=</span> <span class="st">"Three_Column"</span>,</span>
<span>  add_element <span class="op">=</span> <span class="st">"jitter"</span>,</span>
<span>  my_shape <span class="op">=</span> <span class="st">"fill_circle"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-box_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/box_plot.html">?TOmicsVis::box_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/box_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/box_plot.html</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::box_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_313-violin_plot">3.1.3 violin_plot<a class="anchor" aria-label="anchor" href="#id_313-violin_plot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: Length, Width, Weight, and Sex traits dataframe (1st-col: Value, 2nd-col: Traits, 3rd-col: Sex).</p>
<p><strong>Output Plot:</strong> Plot: Violin plot support two levels and multiple groups with P value.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">traits_sex</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run violin_plot plot function</span></span>
<span><span class="fu"><a href="reference/violin_plot.html">violin_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">traits_sex</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"t.test"</span>,</span>
<span>  test_label <span class="op">=</span> <span class="st">"p.format"</span>,</span>
<span>  group_level <span class="op">=</span> <span class="st">"Three_Column"</span>,</span>
<span>  violin_orientation <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  add_element <span class="op">=</span> <span class="st">"boxplot"</span>,</span>
<span>  element_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  my_shape <span class="op">=</span> <span class="st">"plus_times"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-violin_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/violin_plot.html">?TOmicsVis::violin_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::violin_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_314-survival_plot">3.1.4 survival_plot<a class="anchor" aria-label="anchor" href="#id_314-survival_plot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: survival record data (1st-col: Time, 2nd-col: Status, 3rd-col: Group).</p>
<p><strong>Output Plot:</strong> Survival plot for analyzing and visualizing survival data.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">survival_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">survival_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Time Status Group</span></span>
<span><span class="co">#&gt; 1   48      0    CT</span></span>
<span><span class="co">#&gt; 2   48      0    CT</span></span>
<span><span class="co">#&gt; 3   48      0    CT</span></span>
<span><span class="co">#&gt; 4   48      0    CT</span></span>
<span><span class="co">#&gt; 5   48      0    CT</span></span>
<span><span class="co">#&gt; 6   48      0    CT</span></span>
<span></span>
<span><span class="co"># 2. Run survival_plot plot function</span></span>
<span><span class="fu"><a href="reference/survival_plot.html">survival_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">survival_data</span>,</span>
<span>  curve_function <span class="op">=</span> <span class="st">"pct"</span>,</span>
<span>  conf_inter <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  interval_style <span class="op">=</span> <span class="st">"ribbon"</span>,</span>
<span>  risk_table <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num_censor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sci_palette <span class="op">=</span> <span class="st">"aaas"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span>,</span>
<span>  x_start <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  y_start <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  y_end <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  x_break <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  y_break <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-survival_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/survival_plot.html">?TOmicsVis::survival_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/survival_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/survival_plot.html</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::survival_plot</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_32-traits-analysis">3.2 Traits Analysis<a class="anchor" aria-label="anchor" href="#id_32-traits-analysis"></a>
</h3>
<div class="section level4">
<h4 id="id_321-corr_heatmap">3.2.1 corr_heatmap<a class="anchor" aria-label="anchor" href="#id_321-corr_heatmap"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Output Plot:</strong> Plot: heatmap plot filled with Pearson correlation values and P values.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="co">#&gt;              Genes   CT_1   CT_2   CT_3 LT20_1 LT20_2 LT20_3 LT15_1 LT15_2</span></span>
<span><span class="co">#&gt; 1     transcript_0 655.78 631.08 669.89 654.21 402.56 447.09 510.08 442.22</span></span>
<span><span class="co">#&gt; 2     transcript_1  92.72 112.26 150.30  88.35  76.35  94.55 120.24  80.89</span></span>
<span><span class="co">#&gt; 3    transcript_10  21.74  31.11  22.58  15.09  13.67  13.24  12.48   7.53</span></span>
<span><span class="co">#&gt; 4   transcript_100   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00</span></span>
<span><span class="co">#&gt; 5  transcript_1000   0.00  14.15  36.01   0.00   0.00 193.59 208.45   0.00</span></span>
<span><span class="co">#&gt; 6 transcript_10000  89.18 158.04  86.28  82.97 117.78 102.24 129.61 112.73</span></span>
<span><span class="co">#&gt;   LT15_3 LT12_1 LT12_2 LT12_3 LT12_6_1 LT12_6_2 LT12_6_3</span></span>
<span><span class="co">#&gt; 1 399.82 483.30 437.89 444.06   405.43   416.63   464.75</span></span>
<span><span class="co">#&gt; 2  73.94  96.25  82.62  85.48    65.12    61.94    73.44</span></span>
<span><span class="co">#&gt; 3  13.35  11.16  11.36   6.96     7.82     4.01    10.02</span></span>
<span><span class="co">#&gt; 4   0.00   0.00   0.00   0.00     0.00     0.00     0.00</span></span>
<span><span class="co">#&gt; 5 232.40 148.58   0.00 181.61     0.02    12.18     0.00</span></span>
<span><span class="co">#&gt; 6  85.70  80.89 124.11 115.25   113.87   107.69   119.83</span></span>
<span></span>
<span><span class="co"># 2. Run corr_heatmap plot function</span></span>
<span><span class="fu"><a href="reference/corr_heatmap.html">corr_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_expression</span>,</span>
<span>  corr_method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  cell_shape <span class="op">=</span> <span class="st">"square"</span>,</span>
<span>  fill_type <span class="op">=</span> <span class="st">"full"</span>,</span>
<span>  lable_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  axis_angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  axis_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  lable_digits <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  color_low <span class="op">=</span> <span class="st">"blue"</span>,</span>
<span>  color_mid <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  color_high <span class="op">=</span> <span class="st">"red"</span>,</span>
<span>  outline_color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for fill is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for fill, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-corr_heatmap-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/corr_heatmap.html">?TOmicsVis::corr_heatmap</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html</a>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::corr_heatmap</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_322-pca_analysis">3.2.2 pca_analysis<a class="anchor" aria-label="anchor" href="#id_322-pca_analysis"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Table:</strong> PCA dimensional reduction analysis for RNA-Seq.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Samples Groups</span></span>
<span><span class="co">#&gt; 1    CT_1     CT</span></span>
<span><span class="co">#&gt; 2    CT_2     CT</span></span>
<span><span class="co">#&gt; 3    CT_3     CT</span></span>
<span><span class="co">#&gt; 4  LT20_1   LT20</span></span>
<span><span class="co">#&gt; 5  LT20_2   LT20</span></span>
<span><span class="co">#&gt; 6  LT20_3   LT20</span></span>
<span></span>
<span><span class="co"># 2. Run pca_analysis plot function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pca_analysis.html">pca_analysis</a></span><span class="op">(</span><span class="va">gene_expression</span>, <span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;               PC1         PC2         PC3        PC4         PC5       PC6</span></span>
<span><span class="co">#&gt; CT_1   -27010.536 -18328.2803   5955.2569 46547.7319  11394.1043 -7197.285</span></span>
<span><span class="co">#&gt; CT_2    16248.651  29132.9251   -824.1857 20747.9618 -18798.8755 21096.088</span></span>
<span><span class="co">#&gt; CT_3    22421.017 -26832.3964   6789.4490  5864.1171 -15375.3418 17424.861</span></span>
<span><span class="co">#&gt; LT20_1 -18587.073   -472.9036 -21638.7836  7765.9575    114.1225 -3943.968</span></span>
<span><span class="co">#&gt; LT20_2  33275.933  -9874.9959 -14991.3942 -7443.9250  -4600.8302 -8072.298</span></span>
<span><span class="co">#&gt; LT20_3  -1596.255  11683.5426 -10892.8493   381.0795  11080.3560 -8994.187</span></span>
<span><span class="co">#&gt;                PC7        PC8        PC9        PC10        PC11       PC12</span></span>
<span><span class="co">#&gt; CT_1     2150.6739   4850.320   4051.745   7666.9445  -3141.9327  -2487.939</span></span>
<span><span class="co">#&gt; CT_2   -12329.1138  -3353.734   4805.659   1503.8533  11184.0296  -4865.436</span></span>
<span><span class="co">#&gt; CT_3    12744.2255 -10037.516 -11468.842    202.4016 -11001.6260  -3847.291</span></span>
<span><span class="co">#&gt; LT20_1   8864.7482 -14171.127  -1968.082  -3562.1899   7446.2105  14831.486</span></span>
<span><span class="co">#&gt; LT20_2   -941.3943  -5072.401   5345.106   6494.1383  -3954.2153   9351.346</span></span>
<span><span class="co">#&gt; LT20_3   7263.9321  -7774.725  -1853.546 -21427.2641    -46.1503 -12507.011</span></span>
<span><span class="co">#&gt;              PC13       PC14          PC15</span></span>
<span><span class="co">#&gt; CT_1    -2704.613  2396.7383  2.528517e-11</span></span>
<span><span class="co">#&gt; CT_2    -2633.057 -1375.3352  6.825657e-11</span></span>
<span><span class="co">#&gt; CT_3     5193.978   188.5601  2.255671e-11</span></span>
<span><span class="co">#&gt; LT20_1   3937.457 -7871.8062  4.864246e-11</span></span>
<span><span class="co">#&gt; LT20_2 -12904.673  6071.6618 -2.020696e-10</span></span>
<span><span class="co">#&gt; LT20_3  -5369.380  2606.1762  1.903509e-11</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/pca_analysis.html">?TOmicsVis::pca_analysis</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/pca_analysis.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/pca_analysis.html</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::pca_analysis</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_323-pca_plot">3.2.3 pca_plot<a class="anchor" aria-label="anchor" href="#id_323-pca_plot"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Plot:</strong> Plot: PCA dimensional reduction visualization for RNA-Seq.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Samples Groups</span></span>
<span><span class="co">#&gt; 1    CT_1     CT</span></span>
<span><span class="co">#&gt; 2    CT_2     CT</span></span>
<span><span class="co">#&gt; 3    CT_3     CT</span></span>
<span><span class="co">#&gt; 4  LT20_1   LT20</span></span>
<span><span class="co">#&gt; 5  LT20_2   LT20</span></span>
<span><span class="co">#&gt; 6  LT20_3   LT20</span></span>
<span></span>
<span><span class="co"># 2. Run pca_plot plot function</span></span>
<span><span class="fu"><a href="reference/pca_plot.html">pca_plot</a></span><span class="op">(</span></span>
<span>  sample_gene <span class="op">=</span> <span class="va">gene_expression</span>,</span>
<span>  group_sample <span class="op">=</span> <span class="va">samples_groups</span>,</span>
<span>  xPC <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  yPC <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  text_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  fill_alpha <span class="op">=</span> <span class="fl">0.10</span>,</span>
<span>  border_alpha <span class="op">=</span> <span class="fl">0.00</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-pca_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/pca_plot.html">?TOmicsVis::pca_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html</a>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::pca_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_324-tsne_analysis">3.2.4 tsne_analysis<a class="anchor" aria-label="anchor" href="#id_324-tsne_analysis"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Table:</strong> TSNE analysis for analyzing and visualizing TSNE algorithm.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run tsne_analysis plot function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/tsne_analysis.html">tsne_analysis</a></span><span class="op">(</span><span class="va">gene_expression</span>, <span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;       TSNE1     TSNE2</span></span>
<span><span class="co">#&gt; 1 -67.41252 -16.61397</span></span>
<span><span class="co">#&gt; 2  43.08349 -34.02654</span></span>
<span><span class="co">#&gt; 3 123.32273  54.14358</span></span>
<span><span class="co">#&gt; 4 -42.52065 -31.30027</span></span>
<span><span class="co">#&gt; 5  94.98790  48.97986</span></span>
<span><span class="co">#&gt; 6 -23.90637 -22.26434</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/tsne_analysis.html">?TOmicsVis::tsne_analysis</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/tsne_analysis.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/tsne_analysis.html</a>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::tsne_analysis</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_325-tsne_plot">3.2.5 tsne_plot<a class="anchor" aria-label="anchor" href="#id_325-tsne_plot"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Plot:</strong> TSNE plot for analyzing and visualizing TSNE algorithm.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run tsne_plot plot function</span></span>
<span><span class="fu"><a href="reference/tsne_plot.html">tsne_plot</a></span><span class="op">(</span></span>
<span>  sample_gene <span class="op">=</span> <span class="va">gene_expression</span>,</span>
<span>  group_sample <span class="op">=</span> <span class="va">samples_groups</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  multi_shape <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  text_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  text_alpha <span class="op">=</span> <span class="fl">0.80</span>,</span>
<span>  fill_alpha <span class="op">=</span> <span class="fl">0.10</span>,</span>
<span>  border_alpha <span class="op">=</span> <span class="fl">0.00</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-tsne_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/tsne_plot.html">?TOmicsVis::tsne_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/tsne_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/tsne_plot.html</a>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::tsne_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_326-umap_analysis">3.2.6 umap_analysis<a class="anchor" aria-label="anchor" href="#id_326-umap_analysis"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Table:</strong> UMAP analysis for analyzing RNA-Seq data.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run tsne_plot plot function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/umap_analysis.html">umap_analysis</a></span><span class="op">(</span><span class="va">gene_expression</span>, <span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;             UMAP1       UMAP2</span></span>
<span><span class="co">#&gt; CT_1   -0.6752746  0.49425898</span></span>
<span><span class="co">#&gt; CT_2    1.0232441  0.03062202</span></span>
<span><span class="co">#&gt; CT_3   -0.4722297 -1.32183550</span></span>
<span><span class="co">#&gt; LT20_1 -0.2414214  0.13870703</span></span>
<span><span class="co">#&gt; LT20_2  0.1991701 -1.23434000</span></span>
<span><span class="co">#&gt; LT20_3  0.6431577  1.11879669</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/umap_analysis.html">?TOmicsVis::umap_analysis</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/umap_analysis.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/umap_analysis.html</a>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::umap_analysis</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_327-umap_plot">3.2.7 umap_plot<a class="anchor" aria-label="anchor" href="#id_327-umap_plot"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Plot:</strong> UMAP plot for analyzing and visualizing UMAP algorithm.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run tsne_plot plot function</span></span>
<span><span class="fu"><a href="reference/umap_plot.html">umap_plot</a></span><span class="op">(</span></span>
<span>  sample_gene <span class="op">=</span> <span class="va">gene_expression</span>,</span>
<span>  group_sample <span class="op">=</span> <span class="va">samples_groups</span>,</span>
<span>  seed <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  multi_shape <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  text_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  text_alpha <span class="op">=</span> <span class="fl">0.80</span>,</span>
<span>  fill_alpha <span class="op">=</span> <span class="fl">0.00</span>,</span>
<span>  border_alpha <span class="op">=</span> <span class="fl">0.00</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-umap_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/umap_plot.html">?TOmicsVis::umap_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/umap_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/umap_plot.html</a>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::umap_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_328-dendro_plot">3.2.8 dendro_plot<a class="anchor" aria-label="anchor" href="#id_328-dendro_plot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Output Plot:</strong> Plot: dendrogram for multiple samples clustering.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run plot function</span></span>
<span><span class="fu"><a href="reference/dendro_plot.html">dendro_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_expression</span>,</span>
<span>  dist_method <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  hc_method <span class="op">=</span> <span class="st">"ward.D2"</span>,</span>
<span>  tree_type <span class="op">=</span> <span class="st">"rectangle"</span>,</span>
<span>  k_num <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"npg"</span>,</span>
<span>  color_labels_by_k <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  horiz <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  line_width <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  rect <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  rect_fill <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Samples"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Height"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'dendextend':</span></span>
<span><span class="co">#&gt;   method     from </span></span>
<span><span class="co">#&gt;   rev.hclust vegan</span></span></code></pre></div>
<p><img src="reference/figures/README-dendro_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/dendro_plot.html">?TOmicsVis::dendro_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/dendro_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/dendro_plot.html</a>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::dendro_plot</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_33-differential-expression-analyais">3.3 Differential Expression Analyais<a class="anchor" aria-label="anchor" href="#id_33-differential-expression-analyais"></a>
</h3>
<div class="section level4">
<h4 id="id_331-venn_plot">3.3.1 venn_plot<a class="anchor" aria-label="anchor" href="#id_331-venn_plot"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: Paired comparisons differentially expressed genes (degs) among groups (1st-col~: degs of paired comparisons).</p>
<p><strong>Output Plot:</strong> Venn plot for stat common and unique gene among multiple sets.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">degs_lists</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">degs_lists</span><span class="op">)</span></span>
<span><span class="co">#&gt;        CT.vs.LT20      CT.vs.LT15       CT.vs.LT12     CT.vs.LT12_6</span></span>
<span><span class="co">#&gt; 1 transcript_9024 transcript_4738  transcript_9956 transcript_10354</span></span>
<span><span class="co">#&gt; 2  transcript_604 transcript_6050  transcript_7601  transcript_2959</span></span>
<span><span class="co">#&gt; 3 transcript_3912 transcript_1039  transcript_5960  transcript_5919</span></span>
<span><span class="co">#&gt; 4 transcript_8676 transcript_1344  transcript_3240  transcript_2395</span></span>
<span><span class="co">#&gt; 5 transcript_8832 transcript_3069 transcript_10224  transcript_9881</span></span>
<span><span class="co">#&gt; 6   transcript_74 transcript_9809  transcript_3151  transcript_8836</span></span>
<span></span>
<span><span class="co"># 2. Run venn_plot plot function</span></span>
<span><span class="fu"><a href="reference/venn_plot.html">venn_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">degs_lists</span>,</span>
<span>    title_size <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    label_show <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    label_size <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>    border_show <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    line_type <span class="op">=</span> <span class="st">"longdash"</span>,</span>
<span>    ellipse_shape <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>    sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>    sci_fill_alpha <span class="op">=</span> <span class="fl">0.65</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-venn_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/venn_plot.html">?TOmicsVis::venn_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html</a>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::venn_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_332-upsetr_plot">3.3.2 upsetr_plot<a class="anchor" aria-label="anchor" href="#id_332-upsetr_plot"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: Paired comparisons differentially expressed genes (degs) among groups (1st-col~: degs of paired comparisons).</p>
<p><strong>Output Plot:</strong> UpSet plot for stat common and unique gene among multiple sets.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">degs_lists</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">degs_lists</span><span class="op">)</span></span>
<span><span class="co">#&gt;        CT.vs.LT20      CT.vs.LT15       CT.vs.LT12     CT.vs.LT12_6</span></span>
<span><span class="co">#&gt; 1 transcript_9024 transcript_4738  transcript_9956 transcript_10354</span></span>
<span><span class="co">#&gt; 2  transcript_604 transcript_6050  transcript_7601  transcript_2959</span></span>
<span><span class="co">#&gt; 3 transcript_3912 transcript_1039  transcript_5960  transcript_5919</span></span>
<span><span class="co">#&gt; 4 transcript_8676 transcript_1344  transcript_3240  transcript_2395</span></span>
<span><span class="co">#&gt; 5 transcript_8832 transcript_3069 transcript_10224  transcript_9881</span></span>
<span><span class="co">#&gt; 6   transcript_74 transcript_9809  transcript_3151  transcript_8836</span></span>
<span></span>
<span><span class="co"># 2. Run upsetr_plot plot function</span></span>
<span><span class="fu"><a href="reference/upsetr_plot.html">upsetr_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">degs_lists</span>,</span>
<span>  sets_num <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  keep_order <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  order_by <span class="op">=</span> <span class="st">"freq"</span>,</span>
<span>  decrease <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  mainbar_color <span class="op">=</span> <span class="st">"#006600"</span>,</span>
<span>  number_angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  matrix_color <span class="op">=</span> <span class="st">"#cc0000"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">4.5</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  line_size <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  shade_color <span class="op">=</span> <span class="st">"#cdcdcd"</span>,</span>
<span>  shade_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  setsbar_color <span class="op">=</span> <span class="st">"#000066"</span>,</span>
<span>  setsnum_size <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  text_scale <span class="op">=</span> <span class="fl">1.2</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-upsetr_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/upsetr_plot.html">?TOmicsVis::upsetr_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/upsetr_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/upsetr_plot.html</a>.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::upsetr_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_333-flower_plot">3.3.3 flower_plot<a class="anchor" aria-label="anchor" href="#id_333-flower_plot"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: Paired comparisons differentially expressed genes (degs) among groups (1st-col~: degs of paired comparisons).</p>
<p><strong>Output Plot:</strong> Flower plot for stat common and unique gene among multiple sets.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">degs_lists</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run plot function</span></span>
<span><span class="fu"><a href="reference/flower_plot.html">flower_plot</a></span><span class="op">(</span></span>
<span>  flower_dat <span class="op">=</span> <span class="va">degs_lists</span>,</span>
<span>  angle <span class="op">=</span> <span class="fl">90</span>,</span>
<span>  a <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  b <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  r <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  ellipse_col_pal <span class="op">=</span> <span class="st">"Spectral"</span>,</span>
<span>  circle_col <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  label_text_cex <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-flower_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/flower_plot.html">?TOmicsVis::flower_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/flower_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/flower_plot.html</a>.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::flower_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_334-volcano_plot">3.3.4 volcano_plot<a class="anchor" aria-label="anchor" href="#id_334-volcano_plot"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: All DEGs of paired comparison CT-vs-LT12 stats dataframe (1st-col: Genes, 2nd-col: log2FoldChange, 3rd-col: Pvalue, 4th-col: FDR).</p>
<p><strong>Output Plot:</strong> Volcano plot for visualizing differentailly expressed genes.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">degs_stats</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">degs_stats</span><span class="op">)</span></span>
<span><span class="co">#&gt;    Gene log2FoldChange      Pvalue         FDR</span></span>
<span><span class="co">#&gt; 1  A1I3    -1.13855748 0.000111040 0.000862478</span></span>
<span><span class="co">#&gt; 2   A1M     0.59076131 0.070988041 0.192551708</span></span>
<span><span class="co">#&gt; 3   A2M     0.09297827 0.819706797 0.913893947</span></span>
<span><span class="co">#&gt; 4 A2ML1    -0.26940689 0.745374782 0.874295125</span></span>
<span><span class="co">#&gt; 5  ABAT     1.24811621 0.000001440 0.000016800</span></span>
<span><span class="co">#&gt; 6 ABCC3    -0.72947545 0.005171574 0.024228298</span></span>
<span></span>
<span><span class="co"># 2. Run volcano_plot plot function</span></span>
<span><span class="fu"><a href="reference/volcano_plot.html">volcano_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">degs_stats</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"CT-vs-LT12"</span>,</span>
<span>  log2fc_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  pq_value <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>  pq_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  cutoff_line <span class="op">=</span> <span class="st">"longdash"</span>,</span>
<span>  point_shape <span class="op">=</span> <span class="st">"large_circle"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  color_normal <span class="op">=</span> <span class="st">"#888888"</span>,</span>
<span>  color_log2fc <span class="op">=</span> <span class="st">"#008000"</span>,</span>
<span>  color_pvalue <span class="op">=</span> <span class="st">"#0088ee"</span>,</span>
<span>  color_Log2fc_p <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  boxed_labels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  draw_connectors <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-volcano_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/volcano_plot.html">?TOmicsVis::volcano_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/volcano_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/volcano_plot.html</a>.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::volcano_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_335-ma_plot">3.3.5 ma_plot<a class="anchor" aria-label="anchor" href="#id_335-ma_plot"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: All DEGs of paired comparison CT-vs-LT12 stats2 dataframe (1st-col: Gene, 2nd-col: baseMean, 3rd-col: Log2FoldChange, 4th-col: FDR).</p>
<p><strong>Output Plot:</strong> MversusA plot for visualizing differentially expressed genes.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">degs_stats2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">degs_stats2</span><span class="op">)</span></span>
<span><span class="co">#&gt;    name     baseMean log2FoldChange         padj</span></span>
<span><span class="co">#&gt; 1  A1I3    0.1184475      0.0000000           NA</span></span>
<span><span class="co">#&gt; 2   A1M 1654.4618140      0.6789538 5.280802e-02</span></span>
<span><span class="co">#&gt; 3   A2M  681.0463277      1.5263838 3.920000e-07</span></span>
<span><span class="co">#&gt; 4 A2ML1  389.7226640      3.8933573 1.180000e-14</span></span>
<span><span class="co">#&gt; 5  ABAT  364.7810090     -2.3554014 1.559230e-04</span></span>
<span><span class="co">#&gt; 6 ABCC3    1.1346239      1.2932740 4.491812e-01</span></span>
<span></span>
<span><span class="co"># 2. Run volcano_plot plot function</span></span>
<span><span class="fu"><a href="reference/ma_plot.html">ma_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">degs_stats2</span>,</span>
<span>  foldchange <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  fdr_value <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">3.0</span>,</span>
<span>  color_up <span class="op">=</span> <span class="st">"#FF0000"</span>,</span>
<span>  color_down <span class="op">=</span> <span class="st">"#008800"</span>,</span>
<span>  color_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  top_method <span class="op">=</span> <span class="st">"fc"</span>,</span>
<span>  top_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  label_box <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"CT-vs-LT12"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Log2 mean expression"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Log2 fold change"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-ma_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/ma_plot.html">?TOmicsVis::ma_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/ma_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/ma_plot.html</a>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::ma_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_336-heatmap_group">3.3.6 heatmap_group<a class="anchor" aria-label="anchor" href="#id_336-heatmap_group"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Plot:</strong> Heatmap group for visualizing grouped gene expression data.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run heatmap_group plot function</span></span>
<span><span class="fu"><a href="reference/heatmap_group.html">heatmap_group</a></span><span class="op">(</span></span>
<span>  sample_gene <span class="op">=</span> <span class="va">gene_expression2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,<span class="op">]</span>,</span>
<span>  group_sample <span class="op">=</span> <span class="va">samples_groups</span>,</span>
<span>  scale_data <span class="op">=</span> <span class="st">"row"</span>,</span>
<span>  clust_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  border_show <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  border_color <span class="op">=</span> <span class="st">"#ffffff"</span>,</span>
<span>  value_show <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  value_decimal <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  value_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  axis_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  cell_height <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#00880055"</span>,</span>
<span>  mid_color <span class="op">=</span> <span class="st">"#ffffff"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#ff000055"</span>,</span>
<span>  na_color <span class="op">=</span> <span class="st">"#ff8800"</span>,</span>
<span>  x_angle <span class="op">=</span> <span class="fl">45</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmap_group-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/heatmap_group.html">?TOmicsVis::heatmap_group</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/heatmap_group.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/heatmap_group.html</a>.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::heatmap_group</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_337-circos_heatmap">3.3.7 circos_heatmap<a class="anchor" aria-label="anchor" href="#id_337-circos_heatmap"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Output Plot:</strong> Circos heatmap plot for visualizing gene expressing in multiple samples.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Genes  CT_1    CT_2  CT_3 LT20_1 LT20_2 LT20_3 LT15_1 LT15_2 LT15_3 LT12_1</span></span>
<span><span class="co">#&gt; 1 ACAA2 24.50   39.83 55.38 114.11 159.32  96.88 169.56 464.84 182.66 116.08</span></span>
<span><span class="co">#&gt; 2  ACAN 14.97   18.71 10.30  71.23 142.67 213.54 253.15 320.80 104.15 174.02</span></span>
<span><span class="co">#&gt; 3  ADH1  1.54    1.56  2.04  14.95  13.60  15.87  12.80  17.74   6.06  10.97</span></span>
<span><span class="co">#&gt; 4  AHSG  0.00 1911.99  0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00</span></span>
<span><span class="co">#&gt; 5 ALDH2  2.07    2.86  2.54   0.85   0.49   0.47   0.42   0.13   0.26   0.00</span></span>
<span><span class="co">#&gt; 6 AP1S3  6.62   14.59  9.30  24.90  33.94  23.19  24.00  36.08  27.40  24.06</span></span>
<span><span class="co">#&gt;   LT12_2 LT12_3 LT12_6_1 LT12_6_2 LT12_6_3</span></span>
<span><span class="co">#&gt; 1 497.29 464.48   471.43   693.62   229.77</span></span>
<span><span class="co">#&gt; 2 305.81 469.48  1291.90   991.90   966.77</span></span>
<span><span class="co">#&gt; 3  10.71  30.95     9.84    10.91     7.28</span></span>
<span><span class="co">#&gt; 4   0.00   0.00     0.00     0.00     0.00</span></span>
<span><span class="co">#&gt; 5   0.28   0.11     0.37     0.15     0.11</span></span>
<span><span class="co">#&gt; 6  38.74  34.54    62.72    41.36    28.75</span></span>
<span></span>
<span><span class="co"># 2. Run circos_heatmap plot function</span></span>
<span><span class="fu"><a href="reference/circos_heatmap.html">circos_heatmap</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_expression2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">50</span>,<span class="op">]</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#0000ff"</span>,</span>
<span>  mid_color <span class="op">=</span> <span class="st">"#ffffff"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  gap_size <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  cluster_run <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  cluster_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  distance_method <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  dend_show <span class="op">=</span> <span class="st">"inside"</span>,</span>
<span>  dend_height <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  track_height <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  rowname_show <span class="op">=</span> <span class="st">"outside"</span>,</span>
<span>  rowname_size <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Note: 15 points are out of plotting region in sector 'group', track</span></span>
<span><span class="co">#&gt; '3'.</span></span>
<span><span class="co">#&gt; Note: 15 points are out of plotting region in sector 'group', track</span></span>
<span><span class="co">#&gt; '3'.</span></span></code></pre></div>
<p><img src="reference/figures/README-circos_heatmap-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/circos_heatmap.html">?TOmicsVis::circos_heatmap</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html</a>.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::circos_heatmap</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_338-chord_plot">3.3.8 chord_plot<a class="anchor" aria-label="anchor" href="#id_338-chord_plot"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Output Plot:</strong> Chord plot for visualizing the relationships of pathways and genes.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load chord_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Genes  CT_1    CT_2  CT_3 LT20_1 LT20_2 LT20_3 LT15_1 LT15_2 LT15_3 LT12_1</span></span>
<span><span class="co">#&gt; 1 ACAA2 24.50   39.83 55.38 114.11 159.32  96.88 169.56 464.84 182.66 116.08</span></span>
<span><span class="co">#&gt; 2  ACAN 14.97   18.71 10.30  71.23 142.67 213.54 253.15 320.80 104.15 174.02</span></span>
<span><span class="co">#&gt; 3  ADH1  1.54    1.56  2.04  14.95  13.60  15.87  12.80  17.74   6.06  10.97</span></span>
<span><span class="co">#&gt; 4  AHSG  0.00 1911.99  0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00</span></span>
<span><span class="co">#&gt; 5 ALDH2  2.07    2.86  2.54   0.85   0.49   0.47   0.42   0.13   0.26   0.00</span></span>
<span><span class="co">#&gt; 6 AP1S3  6.62   14.59  9.30  24.90  33.94  23.19  24.00  36.08  27.40  24.06</span></span>
<span><span class="co">#&gt;   LT12_2 LT12_3 LT12_6_1 LT12_6_2 LT12_6_3</span></span>
<span><span class="co">#&gt; 1 497.29 464.48   471.43   693.62   229.77</span></span>
<span><span class="co">#&gt; 2 305.81 469.48  1291.90   991.90   966.77</span></span>
<span><span class="co">#&gt; 3  10.71  30.95     9.84    10.91     7.28</span></span>
<span><span class="co">#&gt; 4   0.00   0.00     0.00     0.00     0.00</span></span>
<span><span class="co">#&gt; 5   0.28   0.11     0.37     0.15     0.11</span></span>
<span><span class="co">#&gt; 6  38.74  34.54    62.72    41.36    28.75</span></span>
<span></span>
<span><span class="co"># 2. Run chord_plot plot function</span></span>
<span><span class="fu"><a href="reference/chord_plot.html">chord_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_expression2</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>,<span class="op">]</span>,</span>
<span>  multi_colors <span class="op">=</span> <span class="st">"VividColors"</span>,</span>
<span>  color_seed <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  color_alpha <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  link_visible <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  link_dir <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,</span>
<span>  link_type <span class="op">=</span> <span class="st">"diffHeight"</span>,</span>
<span>  sector_scale <span class="op">=</span> <span class="st">"Origin"</span>,</span>
<span>  width_circle <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  dist_name <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  label_dir <span class="op">=</span> <span class="st">"Vertical"</span>,</span>
<span>  dist_label <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  label_scale <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-chord_plot-1.png"><!-- --></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="co">#&gt;      rn   cn value1 value2 o1 o2      x1     x2       col</span></span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a><span class="co">#&gt; 1 ACAA2 CT_1  24.50  24.50 15 30 3779.75 394.66 #89A2E4B2</span></span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a><span class="co">#&gt; 2  ACAN CT_1  14.97  14.97 15 29 5349.40 370.16 #E5B07FB2</span></span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a><span class="co">#&gt; 3  ADH1 CT_1   1.54   1.54 15 28  166.82 355.19 #CB6280B2</span></span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a><span class="co">#&gt; 4  AHSG CT_1   0.00   0.00 15 27 1911.99 353.65 #DEBD62B2</span></span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a><span class="co">#&gt; 5 ALDH2 CT_1   2.07   2.07 15 26   11.11 353.65 #56CCD8B2</span></span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a><span class="co">#&gt; 6 AP1S3 CT_1   6.62   6.62 15 25  430.19 351.58 #6ABA8AB2</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/chord_plot.html">?TOmicsVis::chord_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html</a>.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::chord_plot</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_34-advanced-analysis">3.4 Advanced Analysis<a class="anchor" aria-label="anchor" href="#id_34-advanced-analysis"></a>
</h3>
<div class="section level4">
<h4 id="id_341-gene_rank_plot">3.4.1 gene_rank_plot<a class="anchor" aria-label="anchor" href="#id_341-gene_rank_plot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: All DEGs of paired comparison CT-vs-LT12 stats dataframe (1st-col: Genes, 2nd-col: log2FoldChange, 3rd-col: Pvalue, 4th-col: FDR).</p>
<p><strong>Output Plot:</strong> Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">degs_stats</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run plot function</span></span>
<span><span class="fu"><a href="reference/gene_rank_plot.html">gene_rank_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">degs_stats</span>,</span>
<span>  log2fc <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"Spectral"</span>,</span>
<span>  top_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  genes_to_label <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  base_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Gene ranking dotplot"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Ranking of differentially expressed genes"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Log2FoldChange"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-gene_rank_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/gene_rank_plot.html">?TOmicsVis::gene_rank_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/gene_rank_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/gene_rank_plot.html</a>.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::gene_rank_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_342-gene_cluster_trend">3.4.2 gene_cluster_trend<a class="anchor" aria-label="anchor" href="#id_342-gene_cluster_trend"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~n-1-col: Groups, n-col: Pathways).</p>
<p><strong>Output Plot:</strong> Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run plot function</span></span>
<span><span class="fu"><a href="reference/gene_cluster_trend.html">gene_cluster_trend</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_expression3</span><span class="op">[</span>,<span class="op">-</span><span class="fl">7</span><span class="op">]</span>,</span>
<span>  thres <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  min_std <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"PiYG"</span>,</span>
<span>  cluster_num <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 0 genes excluded.</span></span>
<span><span class="co">#&gt; 0 genes excluded.</span></span></code></pre></div>
<p><img src="reference/figures/README-gene_cluster_trend-1.png"><!-- --></p>
<div class="sourceCode" id="cb50"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" tabindex="-1"></a><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/gene_cluster_trend.html">?TOmicsVis::gene_cluster_trend</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/gene_cluster_trend.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/gene_cluster_trend.html</a>.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::gene_cluster_trend</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_343-trend_plot">3.4.3 trend_plot<a class="anchor" aria-label="anchor" href="#id_343-trend_plot"></a>
</h4>
<p><strong>Input Data2:</strong> Dataframe: Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~n-1-col: Groups, n-col: Pathways).</p>
<p><strong>Output Plot:</strong> Trend plot for visualizing gene expression trend profile in multiple traits.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression3</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_expression3</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Genes         CT        LT20     LT15      LT12      LT12_6</span></span>
<span><span class="co">#&gt; 1 ACAA2  39.903333 123.4366667 272.3533 359.28333  464.940000</span></span>
<span><span class="co">#&gt; 2  ACAN  14.660000 142.4800000 226.0333 316.43667 1083.523333</span></span>
<span><span class="co">#&gt; 3  ADH1   1.713333  14.8066667  12.2000  17.54333    9.343333</span></span>
<span><span class="co">#&gt; 4  AHSG 637.330000   0.0000000   0.0000   0.00000    0.000000</span></span>
<span><span class="co">#&gt; 5 ALDH2   2.490000   0.6033333   0.2700   0.13000    0.210000</span></span>
<span><span class="co">#&gt; 6 AP1S3  10.170000  27.3433333  29.1600  32.44667   44.276667</span></span>
<span><span class="co">#&gt;                 Pathways</span></span>
<span><span class="co">#&gt; 1 PPAR signaling pathway</span></span>
<span><span class="co">#&gt; 2 PPAR signaling pathway</span></span>
<span><span class="co">#&gt; 3 PPAR signaling pathway</span></span>
<span><span class="co">#&gt; 4 PPAR signaling pathway</span></span>
<span><span class="co">#&gt; 5 PPAR signaling pathway</span></span>
<span><span class="co">#&gt; 6 PPAR signaling pathway</span></span>
<span></span>
<span><span class="co"># 2. Run trend_plot plot function</span></span>
<span><span class="fu"><a href="reference/trend_plot.html">trend_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_expression3</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span>,<span class="op">]</span>,</span>
<span>  scale_method <span class="op">=</span> <span class="st">"centerObs"</span>,</span>
<span>  miss_value <span class="op">=</span> <span class="st">"exclude"</span>,</span>
<span>  line_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  show_points <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_boxplot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num_column <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Traits"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Genes Expression"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-trend_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/trend_plot.html">?TOmicsVis::trend_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/trend_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/trend_plot.html</a>.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::trend_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_344-wgcna_pipeline">3.4.4 wgcna_pipeline<a class="anchor" aria-label="anchor" href="#id_344-wgcna_pipeline"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: All genes in all samples expression dataframe of RNA-Seq (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: Samples and groups for gene expression (1st-col: Samples, 2nd-col: Groups).</p>
<p><strong>Output Plot:</strong> WGCNA analysis pipeline for RNA-Seq.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load wgcna_pipeline example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_expression</span><span class="op">)</span></span>
<span><span class="co">#&gt;              Genes   CT_1   CT_2   CT_3 LT20_1 LT20_2 LT20_3 LT15_1 LT15_2</span></span>
<span><span class="co">#&gt; 1     transcript_0 655.78 631.08 669.89 654.21 402.56 447.09 510.08 442.22</span></span>
<span><span class="co">#&gt; 2     transcript_1  92.72 112.26 150.30  88.35  76.35  94.55 120.24  80.89</span></span>
<span><span class="co">#&gt; 3    transcript_10  21.74  31.11  22.58  15.09  13.67  13.24  12.48   7.53</span></span>
<span><span class="co">#&gt; 4   transcript_100   0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00</span></span>
<span><span class="co">#&gt; 5  transcript_1000   0.00  14.15  36.01   0.00   0.00 193.59 208.45   0.00</span></span>
<span><span class="co">#&gt; 6 transcript_10000  89.18 158.04  86.28  82.97 117.78 102.24 129.61 112.73</span></span>
<span><span class="co">#&gt;   LT15_3 LT12_1 LT12_2 LT12_3 LT12_6_1 LT12_6_2 LT12_6_3</span></span>
<span><span class="co">#&gt; 1 399.82 483.30 437.89 444.06   405.43   416.63   464.75</span></span>
<span><span class="co">#&gt; 2  73.94  96.25  82.62  85.48    65.12    61.94    73.44</span></span>
<span><span class="co">#&gt; 3  13.35  11.16  11.36   6.96     7.82     4.01    10.02</span></span>
<span><span class="co">#&gt; 4   0.00   0.00   0.00   0.00     0.00     0.00     0.00</span></span>
<span><span class="co">#&gt; 5 232.40 148.58   0.00 181.61     0.02    12.18     0.00</span></span>
<span><span class="co">#&gt; 6  85.70  80.89 124.11 115.25   113.87   107.69   119.83</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples_groups</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Samples Groups</span></span>
<span><span class="co">#&gt; 1    CT_1     CT</span></span>
<span><span class="co">#&gt; 2    CT_2     CT</span></span>
<span><span class="co">#&gt; 3    CT_3     CT</span></span>
<span><span class="co">#&gt; 4  LT20_1   LT20</span></span>
<span><span class="co">#&gt; 5  LT20_2   LT20</span></span>
<span><span class="co">#&gt; 6  LT20_3   LT20</span></span>
<span></span>
<span><span class="co"># 2. Run wgcna_pipeline plot function</span></span>
<span><span class="co"># wgcna_pipeline(gene_expression[1:3000,], samples_groups)</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/wgcna_pipeline.html">?TOmicsVis::wgcna_pipeline</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/wgcna_pipeline.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/wgcna_pipeline.html</a>.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::wgcna_pipeline</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_345-network_plot">3.4.5 network_plot<a class="anchor" aria-label="anchor" href="#id_345-network_plot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: Network data from WGCNA tan module top-200 dataframe (1st-col: Source, 2nd-col: Target).</p>
<p><strong>Output Plot:</strong> Network plot for analyzing and visualizing relationship of genes.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">network_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">network_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Source Target</span></span>
<span><span class="co">#&gt; 1  Cebpd  Cebpd</span></span>
<span><span class="co">#&gt; 2  CYR61  Cebpd</span></span>
<span><span class="co">#&gt; 3  Cebpd CDKN1B</span></span>
<span><span class="co">#&gt; 4  CYR61 CDKN1B</span></span>
<span><span class="co">#&gt; 5   junb  Cebpd</span></span>
<span><span class="co">#&gt; 6 IGFBP1  Cebpd</span></span>
<span></span>
<span><span class="co"># 2. Run network_plot plot function</span></span>
<span><span class="fu"><a href="reference/network_plot.html">network_plot</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">network_data</span>,</span>
<span>  calc_by <span class="op">=</span> <span class="st">"degree"</span>,</span>
<span>  degree_value <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  normal_color <span class="op">=</span> <span class="st">"#008888cc"</span>,</span>
<span>  border_color <span class="op">=</span> <span class="st">"#FFFFFF"</span>,</span>
<span>  from_color <span class="op">=</span> <span class="st">"#FF0000cc"</span>,</span>
<span>  to_color <span class="op">=</span> <span class="st">"#008800cc"</span>,</span>
<span>  normal_shape <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  spatial_shape <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  node_size <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  lable_color <span class="op">=</span> <span class="st">"#FFFFFF"</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  edge_color <span class="op">=</span> <span class="st">"#888888"</span>,</span>
<span>  edge_width <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  edge_curved <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  net_layout <span class="op">=</span> <span class="st">"layout_on_sphere"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-network_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/network_plot.html">?TOmicsVis::network_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/network_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/network_plot.html</a>.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::network_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_346-heatmap_cluster">3.4.6 heatmap_cluster<a class="anchor" aria-label="anchor" href="#id_346-heatmap_cluster"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Output Plot:</strong> Heatmap cluster plot for visualizing clustered gene expression data.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Genes  CT_1    CT_2  CT_3 LT20_1 LT20_2 LT20_3 LT15_1 LT15_2 LT15_3 LT12_1</span></span>
<span><span class="co">#&gt; 1 ACAA2 24.50   39.83 55.38 114.11 159.32  96.88 169.56 464.84 182.66 116.08</span></span>
<span><span class="co">#&gt; 2  ACAN 14.97   18.71 10.30  71.23 142.67 213.54 253.15 320.80 104.15 174.02</span></span>
<span><span class="co">#&gt; 3  ADH1  1.54    1.56  2.04  14.95  13.60  15.87  12.80  17.74   6.06  10.97</span></span>
<span><span class="co">#&gt; 4  AHSG  0.00 1911.99  0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00</span></span>
<span><span class="co">#&gt; 5 ALDH2  2.07    2.86  2.54   0.85   0.49   0.47   0.42   0.13   0.26   0.00</span></span>
<span><span class="co">#&gt; 6 AP1S3  6.62   14.59  9.30  24.90  33.94  23.19  24.00  36.08  27.40  24.06</span></span>
<span><span class="co">#&gt;   LT12_2 LT12_3 LT12_6_1 LT12_6_2 LT12_6_3</span></span>
<span><span class="co">#&gt; 1 497.29 464.48   471.43   693.62   229.77</span></span>
<span><span class="co">#&gt; 2 305.81 469.48  1291.90   991.90   966.77</span></span>
<span><span class="co">#&gt; 3  10.71  30.95     9.84    10.91     7.28</span></span>
<span><span class="co">#&gt; 4   0.00   0.00     0.00     0.00     0.00</span></span>
<span><span class="co">#&gt; 5   0.28   0.11     0.37     0.15     0.11</span></span>
<span><span class="co">#&gt; 6  38.74  34.54    62.72    41.36    28.75</span></span>
<span></span>
<span><span class="co"># 2. Run network_plot plot function</span></span>
<span><span class="fu"><a href="reference/heatmap_cluster.html">heatmap_cluster</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_expression2</span>,</span>
<span>  dist_method <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  hc_method <span class="op">=</span> <span class="st">"average"</span>,</span>
<span>  k_num <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  show_rownames <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"RdBu"</span>,</span>
<span>  cluster_pal <span class="op">=</span> <span class="st">"Set1"</span>,</span>
<span>  border_color <span class="op">=</span> <span class="st">"#ffffff"</span>,</span>
<span>  angle_col <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  base_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  line_color <span class="op">=</span> <span class="st">"#0000cd"</span>,</span>
<span>  line_alpha <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  summary_color <span class="op">=</span> <span class="st">"#0000cd"</span>,</span>
<span>  summary_alpha <span class="op">=</span> <span class="fl">0.8</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmap_cluster-1.png"><!-- --></p>
<div class="sourceCode" id="cb59"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" tabindex="-1"></a><span class="co">#&gt; Using Cluster, gene as id variables</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmap_cluster-2.png"><!-- --><img src="reference/figures/README-heatmap_cluster-3.png"><!-- --></p>
<p>Get help using command <code><a href="reference/heatmap_cluster.html">?TOmicsVis::heatmap_cluster</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/heatmap_cluster.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/heatmap_cluster.html</a>.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::heatmap_cluster</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_35-go-and-kegg-enrichment">3.5 GO and KEGG Enrichment<a class="anchor" aria-label="anchor" href="#id_35-go-and-kegg-enrichment"></a>
</h3>
<div class="section level4">
<h4 id="id_351-go_enrich">3.5.1 go_enrich<a class="anchor" aria-label="anchor" href="#id_351-go_enrich"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Table:</strong> GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes</span></span>
<span><span class="co">#&gt; 1        FN1</span></span>
<span><span class="co">#&gt; 2 14-3-3ZETA</span></span>
<span><span class="co">#&gt; 3       A1I3</span></span>
<span><span class="co">#&gt; 4        A2M</span></span>
<span><span class="co">#&gt; 5       AARS</span></span>
<span><span class="co">#&gt; 6       ABAT</span></span>
<span><span class="co">#&gt;                                                                                                 biological_process</span></span>
<span><span class="co">#&gt; 1 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 2                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5                                                                           GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 6                                                            GO:0009448(gamma-aminobutyric acid metabolic process)</span></span>
<span><span class="co">#&gt;                 cellular_component</span></span>
<span><span class="co">#&gt; 1 GO:0005576(extracellular region)</span></span>
<span><span class="co">#&gt; 2                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 4  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 5            GO:0005737(cytoplasm)</span></span>
<span><span class="co">#&gt; 6                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                                                                                                       molecular_function</span></span>
<span><span class="co">#&gt; 1                                                                                                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2                                                                            GO:0019904(protein domain specific binding)</span></span>
<span><span class="co">#&gt; 3                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 4                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 5 GO:0004813(alanine-tRNA ligase activity);GO:0005524(ATP binding);GO:0000049(tRNA binding);GO:0008270(zinc ion binding)</span></span>
<span><span class="co">#&gt; 6                              GO:0003867(4-aminobutyrate transaminase activity);GO:0030170(pyridoxal phosphate binding)</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1                                                                                                   ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 6         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich analysis function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/go_enrich.html">go_enrich</a></span><span class="op">(</span></span>
<span>  go_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="op">-</span><span class="fl">5</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;           ID           ontology</span></span>
<span><span class="co">#&gt; 1 GO:0000221 cellular component</span></span>
<span><span class="co">#&gt; 2 GO:0000275 cellular component</span></span>
<span><span class="co">#&gt; 3 GO:0000276 cellular component</span></span>
<span><span class="co">#&gt; 4 GO:0000398 biological process</span></span>
<span><span class="co">#&gt; 5 GO:0000774 molecular function</span></span>
<span><span class="co">#&gt; 6 GO:0001671 molecular function</span></span>
<span><span class="co">#&gt;                                                                 Description</span></span>
<span><span class="co">#&gt; 1                     vacuolar proton-transporting V-type ATPase, V1 domain</span></span>
<span><span class="co">#&gt; 2  mitochondrial proton-transporting ATP synthase complex, catalytic core F</span></span>
<span><span class="co">#&gt; 3 mitochondrial proton-transporting ATP synthase complex, coupling factor F</span></span>
<span><span class="co">#&gt; 4                                            mRNA splicing, via spliceosome</span></span>
<span><span class="co">#&gt; 5                                adenyl-nucleotide exchange factor activity</span></span>
<span><span class="co">#&gt; 6                                                 ATPase activator activity</span></span>
<span><span class="co">#&gt;   GeneRatio BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">#&gt; 1     1/101  1/1279 7.896794e-02 1.110997e-01 9.458955e-02</span></span>
<span><span class="co">#&gt; 2     1/101  1/1279 7.896794e-02 1.110997e-01 9.458955e-02</span></span>
<span><span class="co">#&gt; 3     6/101  6/1279 2.109128e-07 1.075656e-05 9.158058e-06</span></span>
<span><span class="co">#&gt; 4     1/101 14/1279 6.858207e-01 7.363549e-01 6.269275e-01</span></span>
<span><span class="co">#&gt; 5     1/101  1/1279 7.896794e-02 1.110997e-01 9.458955e-02</span></span>
<span><span class="co">#&gt; 6     1/101  1/1279 7.896794e-02 1.110997e-01 9.458955e-02</span></span>
<span><span class="co">#&gt;                                       geneID Count</span></span>
<span><span class="co">#&gt; 1                                    ATP6V1H     1</span></span>
<span><span class="co">#&gt; 2                                    ATP5F1E     1</span></span>
<span><span class="co">#&gt; 3 ATP5MC1/ATP5ME/ATP5MG/ATP5PB/ATP5PD/ATP5PF     6</span></span>
<span><span class="co">#&gt; 4                                      CDC40     1</span></span>
<span><span class="co">#&gt; 5                                       BAG2     1</span></span>
<span><span class="co">#&gt; 6                                     ATP1B1     1</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/go_enrich.html">?TOmicsVis::go_enrich</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html</a>.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_352-go_enrich_stat">3.5.2 go_enrich_stat<a class="anchor" aria-label="anchor" href="#id_352-go_enrich_stat"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> GO enrichment analysis and stat plot (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_stat analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_stat.html">go_enrich_stat</a></span><span class="op">(</span></span>
<span>  go_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="op">-</span><span class="fl">5</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  max_go_item <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  strip_fill <span class="op">=</span> <span class="st">"#CDCDCD"</span>,</span>
<span>  xtext_angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_stat-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_stat.html">?TOmicsVis::go_enrich_stat</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html</a>.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_stat</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_353-go_enrich_bar">3.5.3 go_enrich_bar<a class="anchor" aria-label="anchor" href="#id_353-go_enrich_bar"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> GO enrichment analysis and bar plot (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_bar analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_bar.html">go_enrich_bar</a></span><span class="op">(</span></span>
<span>  go_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="op">-</span><span class="fl">5</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008800aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for fill is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for fill, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_bar-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_bar.html">?TOmicsVis::go_enrich_bar</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html</a>.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_bar</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_354-go_enrich_dot">3.5.4 go_enrich_dot<a class="anchor" aria-label="anchor" href="#id_354-go_enrich_dot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> GO enrichment analysis and dot plot (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_dot analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_dot.html">go_enrich_dot</a></span><span class="op">(</span></span>
<span>  go_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="op">-</span><span class="fl">5</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008800aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_dot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_dot.html">?TOmicsVis::go_enrich_dot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html</a>.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_dot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_355-go_enrich_net">3.5.5 go_enrich_net<a class="anchor" aria-label="anchor" href="#id_355-go_enrich_net"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> GO enrichment analysis and net plot (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_net analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_net.html">go_enrich_net</a></span><span class="op">(</span></span>
<span>  go_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="op">-</span><span class="fl">5</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  net_layout <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  net_circular <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008800aa"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_net-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_net.html">?TOmicsVis::go_enrich_net</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html</a>.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_net</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_356-kegg_enrich">3.5.6 kegg_enrich<a class="anchor" aria-label="anchor" href="#id_356-kegg_enrich"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes</span></span>
<span><span class="co">#&gt; 1        FN1</span></span>
<span><span class="co">#&gt; 2 14-3-3ZETA</span></span>
<span><span class="co">#&gt; 3       A1I3</span></span>
<span><span class="co">#&gt; 4        A2M</span></span>
<span><span class="co">#&gt; 5       AARS</span></span>
<span><span class="co">#&gt; 6       ABAT</span></span>
<span><span class="co">#&gt;                                                                                                 biological_process</span></span>
<span><span class="co">#&gt; 1 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 2                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5                                                                           GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 6                                                            GO:0009448(gamma-aminobutyric acid metabolic process)</span></span>
<span><span class="co">#&gt;                 cellular_component</span></span>
<span><span class="co">#&gt; 1 GO:0005576(extracellular region)</span></span>
<span><span class="co">#&gt; 2                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 4  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 5            GO:0005737(cytoplasm)</span></span>
<span><span class="co">#&gt; 6                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                                                                                                       molecular_function</span></span>
<span><span class="co">#&gt; 1                                                                                                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2                                                                            GO:0019904(protein domain specific binding)</span></span>
<span><span class="co">#&gt; 3                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 4                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 5 GO:0004813(alanine-tRNA ligase activity);GO:0005524(ATP binding);GO:0000049(tRNA binding);GO:0008270(zinc ion binding)</span></span>
<span><span class="co">#&gt; 6                              GO:0003867(4-aminobutyrate transaminase activity);GO:0030170(pyridoxal phosphate binding)</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1                                                                                                   ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 6         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich analysis function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/kegg_enrich.html">kegg_enrich</a></span><span class="op">(</span></span>
<span>  kegg_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              ID                         Description GeneRatio BgRatio</span></span>
<span><span class="co">#&gt; ko04966 ko04966      Collecting duct acid secretion     7/101  7/1279</span></span>
<span><span class="co">#&gt; ko00190 ko00190           Oxidative phosphorylation    23/101 88/1279</span></span>
<span><span class="co">#&gt; ko04721 ko04721              Synaptic vesicle cycle     8/101 13/1279</span></span>
<span><span class="co">#&gt; ko04610 ko04610 Complement and coagulation cascades    13/101 43/1279</span></span>
<span><span class="co">#&gt; ko04145 ko04145                           Phagosome    11/101 33/1279</span></span>
<span><span class="co">#&gt; ko04971 ko04971              Gastric acid secretion     4/101  4/1279</span></span>
<span><span class="co">#&gt;               pvalue     p.adjust       qvalue</span></span>
<span><span class="co">#&gt; ko04966 1.573976e-08 2.030430e-06 1.723090e-06</span></span>
<span><span class="co">#&gt; ko00190 5.232645e-08 3.375056e-06 2.864185e-06</span></span>
<span><span class="co">#&gt; ko04721 1.069634e-06 4.599427e-05 3.903227e-05</span></span>
<span><span class="co">#&gt; ko04610 1.078094e-05 3.476853e-04 2.950573e-04</span></span>
<span><span class="co">#&gt; ko04145 1.941460e-05 5.008968e-04 4.250776e-04</span></span>
<span><span class="co">#&gt; ko04971 3.679084e-05 7.910030e-04 6.712714e-04</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                        geneID</span></span>
<span><span class="co">#&gt; ko04966                                                                                                                              ATP6V0C/ATP6V0E1/ATP6V1B2/ATP6V1C1A/ATP6V1F/ATP6V1G1/CA1</span></span>
<span><span class="co">#&gt; ko00190 ATP5F1A/ATP5F1B/ATP5F1C/ATP5F1D/ATP5F1E/ATP5MC1/ATP5MC2/ATP5MC3/ATP5ME/ATP5MF/ATP5MG/ATP5PB/ATP5PD/ATP5PF/ATP5PO/ATP6V0B/ATP6V0C/ATP6V0E1/ATP6V1B2/ATP6V1C1A/ATP6V1F/ATP6V1G1/ATP6V1H</span></span>
<span><span class="co">#&gt; ko04721                                                                                                                  ATP6V0B/ATP6V0C/ATP6V0E1/ATP6V1B2/ATP6V1C1A/ATP6V1F/ATP6V1G1/ATP6V1H</span></span>
<span><span class="co">#&gt; ko04610                                                                                                                                       C1QC/C1S/C3/C4/C4A/C5/C6/C7/C8A/C8B/C8G/C9/CD59</span></span>
<span><span class="co">#&gt; ko04145                                                                                                     ATP6V0B/ATP6V0C/ATP6V0E1/ATP6V1B2/ATP6V1C1A/ATP6V1F/ATP6V1G1/ATP6V1H/C3/CALR/CANX</span></span>
<span><span class="co">#&gt; ko04971                                                                                                                                                               ATP1B1/CA1/CALM1/CAMK2D</span></span>
<span><span class="co">#&gt;         Count</span></span>
<span><span class="co">#&gt; ko04966     7</span></span>
<span><span class="co">#&gt; ko00190    23</span></span>
<span><span class="co">#&gt; ko04721     8</span></span>
<span><span class="co">#&gt; ko04610    13</span></span>
<span><span class="co">#&gt; ko04145    11</span></span>
<span><span class="co">#&gt; ko04971     4</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/kegg_enrich.html">?TOmicsVis::kegg_enrich</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html</a>.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_357-kegg_enrich_bar">3.5.7 kegg_enrich_bar<a class="anchor" aria-label="anchor" href="#id_357-kegg_enrich_bar"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> KEGG enrichment analysis and bar plot (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_bar analysis function</span></span>
<span><span class="fu"><a href="reference/kegg_enrich_bar.html">kegg_enrich_bar</a></span><span class="op">(</span></span>
<span>  kegg_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008800aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for fill is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for fill, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-kegg_enrich_bar-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/kegg_enrich_bar.html">?TOmicsVis::kegg_enrich_bar</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html</a>.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_bar</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_358-kegg_enrich_dot">3.5.8 kegg_enrich_dot<a class="anchor" aria-label="anchor" href="#id_358-kegg_enrich_dot"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> KEGG enrichment analysis and dot plot (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_dot analysis function</span></span>
<span><span class="fu"><a href="reference/kegg_enrich_dot.html">kegg_enrich_dot</a></span><span class="op">(</span></span>
<span>  kegg_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008800aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-kegg_enrich_dot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/kegg_enrich_dot.html">?TOmicsVis::kegg_enrich_dot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html</a>.</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_dot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_359-kegg_enrich_net">3.5.9 kegg_enrich_net<a class="anchor" aria-label="anchor" href="#id_359-kegg_enrich_net"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> KEGG enrichment analysis and net plot (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_net analysis function</span></span>
<span><span class="fu"><a href="reference/kegg_enrich_net.html">kegg_enrich_net</a></span><span class="op">(</span></span>
<span>  kegg_anno <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">5</span><span class="op">)</span><span class="op">]</span>,</span>
<span>  degs_list <span class="op">=</span> <span class="va">gene_go_kegg</span><span class="op">[</span><span class="fl">100</span><span class="op">:</span><span class="fl">200</span>,<span class="fl">1</span><span class="op">]</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  net_layout <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  net_circular <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008800aa"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-kegg_enrich_net-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/kegg_enrich_net.html">?TOmicsVis::kegg_enrich_net</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html</a>.</p>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_net</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_36-tables-operations">3.6 Tables Operations<a class="anchor" aria-label="anchor" href="#id_36-tables-operations"></a>
</h3>
<div class="section level4">
<h4 id="id_361-table_split">3.6.1 table_split<a class="anchor" aria-label="anchor" href="#id_361-table_split"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Table:</strong> Table split used for splitting a grouped column to multiple columns.</p>
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_go_kegg2</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes</span></span>
<span><span class="co">#&gt; 1        FN1</span></span>
<span><span class="co">#&gt; 2 14-3-3ZETA</span></span>
<span><span class="co">#&gt; 3       A1I3</span></span>
<span><span class="co">#&gt; 4        A2M</span></span>
<span><span class="co">#&gt; 5       AARS</span></span>
<span><span class="co">#&gt; 6       ABAT</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1                                                                                                   ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 6         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span><span class="co">#&gt;          go_category</span></span>
<span><span class="co">#&gt; 1 biological_process</span></span>
<span><span class="co">#&gt; 2 biological_process</span></span>
<span><span class="co">#&gt; 3 biological_process</span></span>
<span><span class="co">#&gt; 4 biological_process</span></span>
<span><span class="co">#&gt; 5 biological_process</span></span>
<span><span class="co">#&gt; 6 biological_process</span></span>
<span><span class="co">#&gt;                                                                                                            go_term</span></span>
<span><span class="co">#&gt; 1 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 2                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5                                                                           GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 6                                                            GO:0009448(gamma-aminobutyric acid metabolic process)</span></span>
<span></span>
<span><span class="co"># 2. Run table_split function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_split.html">table_split</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_go_kegg2</span>,</span>
<span>  grouped_var <span class="op">=</span> <span class="st">"go_category"</span>,</span>
<span>  value_var <span class="op">=</span> <span class="st">"go_term"</span>,</span>
<span>  miss_drop <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes</span></span>
<span><span class="co">#&gt; 1 14-3-3ZETA</span></span>
<span><span class="co">#&gt; 2       A1I3</span></span>
<span><span class="co">#&gt; 3        A2M</span></span>
<span><span class="co">#&gt; 4       AARS</span></span>
<span><span class="co">#&gt; 5       ABAT</span></span>
<span><span class="co">#&gt; 6      ABCB7</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 2                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 5         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span><span class="co">#&gt; 6                                                                                                                                                                                                                 ko02010(ABC transporters)</span></span>
<span><span class="co">#&gt;                                      biological_process</span></span>
<span><span class="co">#&gt; 1                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 5 GO:0009448(gamma-aminobutyric acid metabolic process)</span></span>
<span><span class="co">#&gt; 6                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                           cellular_component</span></span>
<span><span class="co">#&gt; 1                                       &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2            GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 3            GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 4                      GO:0005737(cytoplasm)</span></span>
<span><span class="co">#&gt; 5                                       &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6 GO:0016021(integral component of membrane)</span></span>
<span><span class="co">#&gt;                                                                                                       molecular_function</span></span>
<span><span class="co">#&gt; 1                                                                            GO:0019904(protein domain specific binding)</span></span>
<span><span class="co">#&gt; 2                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 3                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 4 GO:0004813(alanine-tRNA ligase activity);GO:0005524(ATP binding);GO:0000049(tRNA binding);GO:0008270(zinc ion binding)</span></span>
<span><span class="co">#&gt; 5                              GO:0003867(4-aminobutyrate transaminase activity);GO:0030170(pyridoxal phosphate binding)</span></span>
<span><span class="co">#&gt; 6      GO:0005524(ATP binding);GO:0016887(ATPase activity);GO:0042626(ATPase-coupled transmembrane transporter activity)</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_split.html">?TOmicsVis::table_split</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_split.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_split.html</a>.</p>
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_split</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_362-table_merge">3.6.2 table_merge<a class="anchor" aria-label="anchor" href="#id_362-table_merge"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Table:</strong> Table merge used to merge multiple variables to on variable.</p>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes</span></span>
<span><span class="co">#&gt; 1        FN1</span></span>
<span><span class="co">#&gt; 2 14-3-3ZETA</span></span>
<span><span class="co">#&gt; 3       A1I3</span></span>
<span><span class="co">#&gt; 4        A2M</span></span>
<span><span class="co">#&gt; 5       AARS</span></span>
<span><span class="co">#&gt; 6       ABAT</span></span>
<span><span class="co">#&gt;                                                                                                 biological_process</span></span>
<span><span class="co">#&gt; 1 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 2                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5                                                                           GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 6                                                            GO:0009448(gamma-aminobutyric acid metabolic process)</span></span>
<span><span class="co">#&gt;                 cellular_component</span></span>
<span><span class="co">#&gt; 1 GO:0005576(extracellular region)</span></span>
<span><span class="co">#&gt; 2                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 4  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 5            GO:0005737(cytoplasm)</span></span>
<span><span class="co">#&gt; 6                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                                                                                                       molecular_function</span></span>
<span><span class="co">#&gt; 1                                                                                                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2                                                                            GO:0019904(protein domain specific binding)</span></span>
<span><span class="co">#&gt; 3                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 4                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 5 GO:0004813(alanine-tRNA ligase activity);GO:0005524(ATP binding);GO:0000049(tRNA binding);GO:0008270(zinc ion binding)</span></span>
<span><span class="co">#&gt; 6                              GO:0003867(4-aminobutyrate transaminase activity);GO:0030170(pyridoxal phosphate binding)</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1                                                                                                   ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 6         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span></span>
<span><span class="co"># 2. Run function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_merge.html">table_merge</a></span><span class="op">(</span></span>
<span>  data <span class="op">=</span> <span class="va">gene_go_kegg</span>,</span>
<span>  merge_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"biological_process"</span>, <span class="st">"cellular_component"</span>, <span class="st">"molecular_function"</span><span class="op">)</span>,</span>
<span>  new_var <span class="op">=</span> <span class="st">"go_category"</span>,</span>
<span>  new_value <span class="op">=</span> <span class="st">"go_term"</span>,</span>
<span>  na_remove <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes</span></span>
<span><span class="co">#&gt; 1        FN1</span></span>
<span><span class="co">#&gt; 2 14-3-3ZETA</span></span>
<span><span class="co">#&gt; 3       A1I3</span></span>
<span><span class="co">#&gt; 4        A2M</span></span>
<span><span class="co">#&gt; 5       AARS</span></span>
<span><span class="co">#&gt; 6       ABAT</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1                                                                                                   ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 6         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span><span class="co">#&gt;          go_category</span></span>
<span><span class="co">#&gt; 1 biological_process</span></span>
<span><span class="co">#&gt; 2 biological_process</span></span>
<span><span class="co">#&gt; 3 biological_process</span></span>
<span><span class="co">#&gt; 4 biological_process</span></span>
<span><span class="co">#&gt; 5 biological_process</span></span>
<span><span class="co">#&gt; 6 biological_process</span></span>
<span><span class="co">#&gt;                                                                                                            go_term</span></span>
<span><span class="co">#&gt; 1 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 2                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5                                                                           GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 6                                                            GO:0009448(gamma-aminobutyric acid metabolic process)</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_merge.html">?TOmicsVis::table_merge</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_merge.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_merge.html</a>.</p>
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_merge</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_363-table_filter">3.6.3 table_filter<a class="anchor" aria-label="anchor" href="#id_363-table_filter"></a>
</h4>
<p><strong>Input Data:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Table:</strong> Table filter used to filter row by column condition.</p>
<div class="sourceCode" id="cb83"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">traits_sex</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">traits_sex</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Value Traits    Sex</span></span>
<span><span class="co">#&gt; 1 36.74 Weight Female</span></span>
<span><span class="co">#&gt; 2 38.54 Weight Female</span></span>
<span><span class="co">#&gt; 3 44.91 Weight Female</span></span>
<span><span class="co">#&gt; 4 43.53 Weight Female</span></span>
<span><span class="co">#&gt; 5 39.03 Weight Female</span></span>
<span><span class="co">#&gt; 6 26.01 Weight Female</span></span>
<span></span>
<span><span class="co"># 2. Run function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_filter.html">table_filter</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">traits_sex</span>, </span>
<span>    <span class="va">Sex</span> <span class="op">==</span> <span class="st">"Male"</span> <span class="op">&amp;</span> <span class="va">Traits</span> <span class="op">==</span> <span class="st">"Weight"</span> <span class="op">&amp;</span> <span class="va">Value</span> <span class="op">&gt;</span> <span class="fl">40</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Value Traits  Sex</span></span>
<span><span class="co">#&gt; 1 48.06 Weight Male</span></span>
<span><span class="co">#&gt; 2 42.74 Weight Male</span></span>
<span><span class="co">#&gt; 3 45.25 Weight Male</span></span>
<span><span class="co">#&gt; 4 44.95 Weight Male</span></span>
<span><span class="co">#&gt; 5 43.21 Weight Male</span></span>
<span><span class="co">#&gt; 6 40.02 Weight Male</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_filter.html">?TOmicsVis::table_filter</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_filter.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_filter.html</a>.</p>
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_filter</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_364-table_cross">3.6.4 table_cross<a class="anchor" aria-label="anchor" href="#id_364-table_cross"></a>
</h4>
<p><strong>Input Data1:</strong> Dataframe: Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq. (1st-col: Genes, 2nd-col~: Samples).</p>
<p><strong>Input Data2:</strong> Dataframe: GO and KEGG annotation of background genes (1st-col: Genes, 2nd-col: biological_process, 3rd-col: cellular_component, 4th-col: molecular_function, 5th-col: kegg_pathway).</p>
<p><strong>Output Plot:</strong> Table cross used to cross search and merge results in two tables.</p>
<div class="sourceCode" id="cb85"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_expression2</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Genes  CT_1    CT_2  CT_3 LT20_1 LT20_2 LT20_3 LT15_1 LT15_2 LT15_3 LT12_1</span></span>
<span><span class="co">#&gt; 1 ACAA2 24.50   39.83 55.38 114.11 159.32  96.88 169.56 464.84 182.66 116.08</span></span>
<span><span class="co">#&gt; 2  ACAN 14.97   18.71 10.30  71.23 142.67 213.54 253.15 320.80 104.15 174.02</span></span>
<span><span class="co">#&gt; 3  ADH1  1.54    1.56  2.04  14.95  13.60  15.87  12.80  17.74   6.06  10.97</span></span>
<span><span class="co">#&gt; 4  AHSG  0.00 1911.99  0.00   0.00   0.00   0.00   0.00   0.00   0.00   0.00</span></span>
<span><span class="co">#&gt; 5 ALDH2  2.07    2.86  2.54   0.85   0.49   0.47   0.42   0.13   0.26   0.00</span></span>
<span><span class="co">#&gt; 6 AP1S3  6.62   14.59  9.30  24.90  33.94  23.19  24.00  36.08  27.40  24.06</span></span>
<span><span class="co">#&gt;   LT12_2 LT12_3 LT12_6_1 LT12_6_2 LT12_6_3</span></span>
<span><span class="co">#&gt; 1 497.29 464.48   471.43   693.62   229.77</span></span>
<span><span class="co">#&gt; 2 305.81 469.48  1291.90   991.90   966.77</span></span>
<span><span class="co">#&gt; 3  10.71  30.95     9.84    10.91     7.28</span></span>
<span><span class="co">#&gt; 4   0.00   0.00     0.00     0.00     0.00</span></span>
<span><span class="co">#&gt; 5   0.28   0.11     0.37     0.15     0.11</span></span>
<span><span class="co">#&gt; 6  38.74  34.54    62.72    41.36    28.75</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_go_kegg</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes</span></span>
<span><span class="co">#&gt; 1        FN1</span></span>
<span><span class="co">#&gt; 2 14-3-3ZETA</span></span>
<span><span class="co">#&gt; 3       A1I3</span></span>
<span><span class="co">#&gt; 4        A2M</span></span>
<span><span class="co">#&gt; 5       AARS</span></span>
<span><span class="co">#&gt; 6       ABAT</span></span>
<span><span class="co">#&gt;                                                                                                 biological_process</span></span>
<span><span class="co">#&gt; 1 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 2                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5                                                                           GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 6                                                            GO:0009448(gamma-aminobutyric acid metabolic process)</span></span>
<span><span class="co">#&gt;                 cellular_component</span></span>
<span><span class="co">#&gt; 1 GO:0005576(extracellular region)</span></span>
<span><span class="co">#&gt; 2                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 4  GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 5            GO:0005737(cytoplasm)</span></span>
<span><span class="co">#&gt; 6                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                                                                                                       molecular_function</span></span>
<span><span class="co">#&gt; 1                                                                                                                   &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2                                                                            GO:0019904(protein domain specific binding)</span></span>
<span><span class="co">#&gt; 3                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 4                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 5 GO:0004813(alanine-tRNA ligase activity);GO:0005524(ATP binding);GO:0000049(tRNA binding);GO:0008270(zinc ion binding)</span></span>
<span><span class="co">#&gt; 6                              GO:0003867(4-aminobutyrate transaminase activity);GO:0030170(pyridoxal phosphate binding)</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1                                                                                                   ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 6         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span></span>
<span><span class="co"># 2. Run function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_cross.html">table_cross</a></span><span class="op">(</span></span>
<span>  data1 <span class="op">=</span> <span class="va">gene_expression2</span>,</span>
<span>  data2 <span class="op">=</span> <span class="va">gene_go_kegg</span>,</span>
<span>  inter_var <span class="op">=</span> <span class="st">"Genes"</span>,</span>
<span>  left_index <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  right_index <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;        Genes CT_1 CT_2 CT_3 LT20_1 LT20_2 LT20_3 LT15_1 LT15_2 LT15_3 LT12_1</span></span>
<span><span class="co">#&gt; 1 14-3-3ZETA   NA   NA   NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 2       A1I3   NA   NA   NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 3        A2M   NA   NA   NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 4       AARS   NA   NA   NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 5       ABAT   NA   NA   NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt; 6      ABCB7   NA   NA   NA     NA     NA     NA     NA     NA     NA     NA</span></span>
<span><span class="co">#&gt;   LT12_2 LT12_3 LT12_6_1 LT12_6_2 LT12_6_3</span></span>
<span><span class="co">#&gt; 1     NA     NA       NA       NA       NA</span></span>
<span><span class="co">#&gt; 2     NA     NA       NA       NA       NA</span></span>
<span><span class="co">#&gt; 3     NA     NA       NA       NA       NA</span></span>
<span><span class="co">#&gt; 4     NA     NA       NA       NA       NA</span></span>
<span><span class="co">#&gt; 5     NA     NA       NA       NA       NA</span></span>
<span><span class="co">#&gt; 6     NA     NA       NA       NA       NA</span></span>
<span><span class="co">#&gt;                                      biological_process</span></span>
<span><span class="co">#&gt; 1                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                GO:0006419(alanyl-tRNA aminoacylation)</span></span>
<span><span class="co">#&gt; 5 GO:0009448(gamma-aminobutyric acid metabolic process)</span></span>
<span><span class="co">#&gt; 6                                                  &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                           cellular_component</span></span>
<span><span class="co">#&gt; 1                                       &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2            GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 3            GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 4                      GO:0005737(cytoplasm)</span></span>
<span><span class="co">#&gt; 5                                       &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6 GO:0016021(integral component of membrane)</span></span>
<span><span class="co">#&gt;                                                                                                       molecular_function</span></span>
<span><span class="co">#&gt; 1                                                                            GO:0019904(protein domain specific binding)</span></span>
<span><span class="co">#&gt; 2                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 3                                                                           GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 4 GO:0004813(alanine-tRNA ligase activity);GO:0005524(ATP binding);GO:0000049(tRNA binding);GO:0008270(zinc ion binding)</span></span>
<span><span class="co">#&gt; 5                              GO:0003867(4-aminobutyrate transaminase activity);GO:0030170(pyridoxal phosphate binding)</span></span>
<span><span class="co">#&gt; 6      GO:0005524(ATP binding);GO:0016887(ATPase activity);GO:0042626(ATPase-coupled transmembrane transporter activity)</span></span>
<span><span class="co">#&gt;                                                                                                                                                                                                                                kegg_pathway</span></span>
<span><span class="co">#&gt; 1 ko04110(Cell cycle);ko04114(Oocyte meiosis);ko04390(Hippo signaling pathway);ko04391(Hippo signaling pathway -fly);ko04013(MAPK signaling pathway - fly);ko04151(PI3K-Akt signaling pathway);ko04212(Longevity regulating pathway - worm)</span></span>
<span><span class="co">#&gt; 2                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                                                              ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                                                      ko00970(Aminoacyl-tRNA biosynthesis)</span></span>
<span><span class="co">#&gt; 5         ko00250(Alanine, aspartate and glutamate metabolism);ko00280(Valine, leucine and isoleucine degradation);ko00650(Butanoate metabolism);ko00640(Propanoate metabolism);ko00410(beta-Alanine metabolism);ko04727(GABAergic synapse)</span></span>
<span><span class="co">#&gt; 6                                                                                                                                                                                                                 ko02010(ABC transporters)</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_cross.html">?TOmicsVis::table_cross</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_cross.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_cross.html</a>.</p>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_cross</span></span></code></pre></div>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=TOmicsVis" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/benben-miao/TOmicsVis/">Browse source code</a></li>
<li><a href="https://github.com/benben-miao/TOmicsVis/issues/">Report a bug</a></li>
<li><a href="https://github.com/OmicsSuite/" class="external-link">OmicsSuite</a></li>
<li><a href="https://github.com/OmicsSuite/OmicsSuite.github.io" class="external-link">OmicsSuite website</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing TOmicsVis</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Benben Miao <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7307-1331" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Wei Dong <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-2201-7164" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Benben Miao, Wei Dong.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
