<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content='Transcriptome visualization from sample trait statistics to gene expression analysis. Six categories include "Samples Statistics", "Traits Analysis", "Differential Expression Analysis", "Advanced Analysis", "GO and KEGG Enrichment", "Tables Operations", with complete sample data.'>
<title>Transcriptome Visualization Process Scheme • TOmicsVis</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="deps/JetBrains_Mono-0.4.2/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Transcriptome Visualization Process Scheme">
<meta property="og:description" content='Transcriptome visualization from sample trait statistics to gene expression analysis. Six categories include "Samples Statistics", "Traits Analysis", "Differential Expression Analysis", "Advanced Analysis", "GO and KEGG Enrichment", "Tables Operations", with complete sample data.'>
<meta property="og:image" content="https://github.com/benben-miao/TOmicsVis/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">TOmicsVis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">
    <span class="fa fa-file-code"></span>
     
    Reference
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--articles">
    <span class="fa fa-book"></span>
     
    Articles
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--articles">
    <h6 class="dropdown-header" data-toc-skip>Tutorial</h6>
    <a class="dropdown-item" href="articles/Introduction.html">Introduction</a>
    <a class="dropdown-item" href="articles/SamplesStatistics.html">SamplesStatistics</a>
    <a class="dropdown-item" href="articles/TraitsAnalysis.html">TraitsAnalysis</a>
    <a class="dropdown-item" href="articles/DifferentialExpression.html">DifferentialExpression</a>
    <a class="dropdown-item" href="articles/AdvancedAnalysis.html">AdvancedAnalysis</a>
    <a class="dropdown-item" href="articles/GOandKEGG.html">GOandKEGG</a>
    <a class="dropdown-item" href="articles/TablesOperations.html">TablesOperations</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Blog</h6>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">
    <span class="fa fa-code-branch"></span>
     
    Versions
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/OmicsSuite/">
    <span class="fa fa-desktop"></span>
     
    OmicsSuite
  </a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="https://github.com/benben-miao/TOmicsVis/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="tomicsvis">TOmicsVis<a class="anchor" aria-label="anchor" href="#tomicsvis"></a>
</h1>
</div>
<div class="section level2">
<h2 id="id_1-introduction">1. Introduction<a class="anchor" aria-label="anchor" href="#id_1-introduction"></a>
</h2>
<p><strong>TOmicsVis</strong>: TranscriptOmics Visualization.</p>
<p><strong>CRAN</strong>: <a href="https://cran.r-project.org/web/packages/TOmicsVis/index.html" class="external-link uri">https://cran.r-project.org/web/packages/TOmicsVis/index.html</a></p>
<p><strong>Github</strong>: <a href="https://github.com/benben-miao/TOmicsVis/" class="uri">https://github.com/benben-miao/TOmicsVis/</a></p>
<p><strong>Website</strong>: <a href="https://benben-miao.github.io/TOmicsVis/" class="external-link uri">https://benben-miao.github.io/TOmicsVis/</a></p>
<p><strong>README</strong>: <a href="https://cran.r-project.org/web/packages/TOmicsVis/readme/README.html" class="external-link uri">https://cran.r-project.org/web/packages/TOmicsVis/readme/README.html</a></p>
<p><strong>Manual</strong>: <a href="https://cran.r-project.org/web/packages/TOmicsVis/TOmicsVis.pdf" class="external-link uri">https://cran.r-project.org/web/packages/TOmicsVis/TOmicsVis.pdf</a></p>
<p><strong><a href="mailto:Authors@R" class="email">Authors@R</a></strong>:</p>
<ul>
<li><p><a href="https://github.com/benben-miao" class="external-link">Benben Miao: https://github.com/benben-miao</a></p></li>
<li><p><a href="https://github.com/dongwei1220" class="external-link">Wei Dong: https://github.com/dongwei1220</a></p></li>
</ul>
</div>
<div class="section level2">
<h2 id="id_2-install-package">2. Install package<a class="anchor" aria-label="anchor" href="#id_2-install-package"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Install from CRAN (Stable Version)</span></span>
<span><span class="co"># install.packages("TOmicsVis")</span></span>
<span></span>
<span><span class="co"># 2. Install from Github (Laste Version)</span></span>
<span><span class="co"># install.packages("devtools")</span></span>
<span><span class="co"># devtools::install_github("benben-miao/TOmicsVis")</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://benben-miao.github.io/TOmicsVis/" class="external-link">TOmicsVis</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; 载入需要的程辑包：e1071</span></span>
<span><span class="co">#&gt; Warning: 程辑包'e1071'是用R版本4.2.3 来建造的</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span></span>
<span><span class="co">#&gt;   method from   </span></span>
<span><span class="co">#&gt;   +.gg   ggplot2</span></span>
<span><span class="co">#&gt; 载入需要的程辑包：Biobase</span></span>
<span><span class="co">#&gt; 载入需要的程辑包：BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 载入程辑包：'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; 载入程辑包：'DynDoc'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     path</span></span>
<span></span>
<span><span class="co"># 3. Recommend package</span></span>
<span><span class="co"># install.packages("ggplot2")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="id_3-usage-cases">3. Usage cases<a class="anchor" aria-label="anchor" href="#id_3-usage-cases"></a>
</h2>
<div class="section level3">
<h3 id="id_31-samples-statistics">3.1 Samples Statistics<a class="anchor" aria-label="anchor" href="#id_31-samples-statistics"></a>
</h3>
<div class="section level4">
<h4 id="id_311-quantile_plot">3.1.1 quantile_plot<a class="anchor" aria-label="anchor" href="#id_311-quantile_plot"></a>
</h4>
<p>Quantile plot for visualizing data distribution.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load box_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">quantile_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run quantile_plot plot function</span></span>
<span><span class="fu"><a href="reference/quantile_plot.html">quantile_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">quantile_data</span>,</span>
<span>  my_shape <span class="op">=</span> <span class="st">"fill_circle"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">2.0</span>,</span>
<span>  conf_int <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  conf_level <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  split_panel <span class="op">=</span> <span class="st">"One_Panel"</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_Simpsons"</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">0.50</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-quantile_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/quantile_plot.html">?TOmicsVis::quantile_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/quantile_plot.html</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::quantile_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_312-corr_heatmap">3.1.2 corr_heatmap<a class="anchor" aria-label="anchor" href="#id_312-corr_heatmap"></a>
</h4>
<p>Correlation Heatmap for samples/groups based on Pearson algorithm.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load gene_exp example dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_exp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_exp</span><span class="op">)</span></span>
<span><span class="co">#&gt;              M1       M2       M3       M4       M5       M6        M7       M8</span></span>
<span><span class="co">#&gt; RGL4   8.454808 8.019389 8.990836 9.718631 7.908075 4.147051  4.985084 4.576711</span></span>
<span><span class="co">#&gt; MPP7   8.690520 8.630346 7.080873 9.838476 8.271824 5.179200  5.200868 3.266993</span></span>
<span><span class="co">#&gt; UGCG   8.648366 8.600555 9.431046 7.923021 8.309214 4.902510  5.750804 4.492856</span></span>
<span><span class="co">#&gt; CYSTM1 8.628884 9.238677 8.487243 8.958537 7.357109 4.541605  6.370533 4.246651</span></span>
<span><span class="co">#&gt; ANXA2  4.983769 6.748022 6.220791 4.719403 3.284346 8.089850 10.637472 7.214912</span></span>
<span><span class="co">#&gt; ENDOD1 5.551640 5.406465 4.663785 3.550765 4.103507 8.393991  9.538503 9.069923</span></span>
<span><span class="co">#&gt;              M9      M10</span></span>
<span><span class="co">#&gt; RGL4   4.930349 4.293700</span></span>
<span><span class="co">#&gt; MPP7   5.565226 4.300309</span></span>
<span><span class="co">#&gt; UGCG   4.659987 3.306275</span></span>
<span><span class="co">#&gt; CYSTM1 4.745769 3.449627</span></span>
<span><span class="co">#&gt; ANXA2  9.002710 5.123359</span></span>
<span><span class="co">#&gt; ENDOD1 8.639664 7.106392</span></span>
<span></span>
<span><span class="co"># 2. Run corr_heatmap plot function</span></span>
<span><span class="fu"><a href="reference/corr_heatmap.html">corr_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">gene_exp</span>,</span>
<span>  corr_method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>  cell_shape <span class="op">=</span> <span class="st">"square"</span>,</span>
<span>  fill_type <span class="op">=</span> <span class="st">"full"</span>,</span>
<span>  lable_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  lable_digits <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  color_low <span class="op">=</span> <span class="st">"#008800ee"</span>,</span>
<span>  color_mid <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  color_high <span class="op">=</span> <span class="st">"#ff0000ee"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for fill is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for fill, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-corr_heatmap-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/corr_heatmap.html">?TOmicsVis::corr_heatmap</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/corr_heatmap.html</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::corr_heatmap</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_313-pca_plot">3.1.3 pca_plot<a class="anchor" aria-label="anchor" href="#id_313-pca_plot"></a>
</h4>
<p>PCA dimensional reduction visualization for RNA-Seq.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load pca_sample_gene and pca_group_sample example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pca_sample_gene</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pca_group_sample</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run pca_plot plot function</span></span>
<span><span class="fu"><a href="reference/pca_plot.html">pca_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">pca_sample_gene</span>,</span>
<span>  <span class="va">pca_group_sample</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  text_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  ellipse_alpha <span class="op">=</span> <span class="fl">0.30</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-pca_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/pca_plot.html">?TOmicsVis::pca_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/pca_plot.html</a>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::pca_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_314-dendro_plot">3.1.4 dendro_plot<a class="anchor" aria-label="anchor" href="#id_314-dendro_plot"></a>
</h4>
<p>Dendrograms for multiple samples/groups clustering.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_exp</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run pca_plot plot function</span></span>
<span><span class="fu"><a href="reference/dendro_plot.html">dendro_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">gene_exp</span>,</span>
<span>  dist_method <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  hc_method <span class="op">=</span> <span class="st">"average"</span>,</span>
<span>  tree_type <span class="op">=</span> <span class="st">"rectangle"</span>,</span>
<span>  k_num <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"npg"</span>,</span>
<span>  color_labels_by_k <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  horiz <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  line_width <span class="op">=</span> <span class="fl">0.7</span>,</span>
<span>  rect <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  rect_fill <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Cluster Dendrogram"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">""</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Height"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'dendextend':</span></span>
<span><span class="co">#&gt;   method     from </span></span>
<span><span class="co">#&gt;   rev.hclust vegan</span></span>
<span><span class="co">#&gt; Warning: The `&lt;scale&gt;` argument of `guides()` cannot be `FALSE`. Use "none" instead as</span></span>
<span><span class="co">#&gt; of ggplot2 3.3.4.</span></span>
<span><span class="co">#&gt; ℹ The deprecated feature was likely used in the factoextra package.</span></span>
<span><span class="co">#&gt;   Please report the issue at &lt;]8;;https://github.com/kassambara/factoextra/issueshttps://github.com/kassambara/factoextra/issues]8;;&gt;.</span></span></code></pre></div>
<p><img src="reference/figures/README-dendro_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/dendro_plot.html">?TOmicsVis::dendro_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/dendro_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/dendro_plot.html</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::dendro_plot</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_32-traits-analysis">3.2 Traits Analysis<a class="anchor" aria-label="anchor" href="#id_32-traits-analysis"></a>
</h3>
<div class="section level4">
<h4 id="id_321-box_plot">3.2.1 box_plot<a class="anchor" aria-label="anchor" href="#id_321-box_plot"></a>
</h4>
<p>Box plot support two levels and multiple groups with P value.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load box_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">box_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run box_plot plot function</span></span>
<span><span class="fu"><a href="reference/box_plot.html">box_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">box_data</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"wilcox.test"</span>,</span>
<span>  test_label <span class="op">=</span> <span class="st">"p.format"</span>,</span>
<span>  notch <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  group_level <span class="op">=</span> <span class="st">"Three_Column"</span>,</span>
<span>  add_element <span class="op">=</span> <span class="st">"dotplot"</span>,</span>
<span>  my_shape <span class="op">=</span> <span class="st">"fill_circle"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Bin width defaults to 1/30 of the range of the data. Pick better value with</span></span>
<span><span class="co">#&gt; `binwidth`.</span></span>
<span><span class="co">#&gt; Notch went outside hinges</span></span>
<span><span class="co">#&gt; ℹ Do you want `notch = FALSE`?</span></span>
<span><span class="co">#&gt; Notch went outside hinges</span></span>
<span><span class="co">#&gt; ℹ Do you want `notch = FALSE`?</span></span>
<span><span class="co">#&gt; Notch went outside hinges</span></span>
<span><span class="co">#&gt; ℹ Do you want `notch = FALSE`?</span></span>
<span><span class="co">#&gt; Notch went outside hinges</span></span>
<span><span class="co">#&gt; ℹ Do you want `notch = FALSE`?</span></span>
<span><span class="co">#&gt; Notch went outside hinges</span></span>
<span><span class="co">#&gt; ℹ Do you want `notch = FALSE`?</span></span>
<span><span class="co">#&gt; Notch went outside hinges</span></span>
<span><span class="co">#&gt; ℹ Do you want `notch = FALSE`?</span></span></code></pre></div>
<p><img src="reference/figures/README-box_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/box_plot.html">?TOmicsVis::box_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/box_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/box_plot.html</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::box_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_322-violin_plot">3.2.2 violin_plot<a class="anchor" aria-label="anchor" href="#id_322-violin_plot"></a>
</h4>
<p>Violin plot support two levels and multiple groups with P value.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load box_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">box_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run violin_plot plot function</span></span>
<span><span class="fu"><a href="reference/violin_plot.html">violin_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">box_data</span>,</span>
<span>  test_method <span class="op">=</span> <span class="st">"wilcox.test"</span>,</span>
<span>  test_label <span class="op">=</span> <span class="st">"p.format"</span>,</span>
<span>  group_level <span class="op">=</span> <span class="st">"Three_Column"</span>,</span>
<span>  violin_orientation <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  add_element <span class="op">=</span> <span class="st">"boxplot"</span>,</span>
<span>  element_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  my_shape <span class="op">=</span> <span class="st">"plus_times"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_NPG"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-violin_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/violin_plot.html">?TOmicsVis::violin_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/violin_plot.html</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::violin_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_323-survival_plot">3.2.3 survival_plot<a class="anchor" aria-label="anchor" href="#id_323-survival_plot"></a>
</h4>
<p>Survival plot for analyzing and visualizing survival data.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load survival_plot example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">survival_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run survival_plot plot function</span></span>
<span><span class="fu"><a href="reference/survival_plot.html">survival_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">survival_data</span>,</span>
<span>  curve_function <span class="op">=</span> <span class="st">"pct"</span>,</span>
<span>  conf_inter <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  interval_style <span class="op">=</span> <span class="st">"ribbon"</span>,</span>
<span>  risk_table <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num_censor <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  sci_palette <span class="op">=</span> <span class="st">"npg"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span>,</span>
<span>  x_start <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  y_start <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  y_end <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  x_break <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  y_break <span class="op">=</span> <span class="fl">25</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-survival_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/survival_plot.html">?TOmicsVis::survival_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/survival_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/survival_plot.html</a>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::survival_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_324-tsne_plot">3.2.4 tsne_plot<a class="anchor" aria-label="anchor" href="#id_324-tsne_plot"></a>
</h4>
<p>TSNE plot for analyzing and visualizing TSNE algorithm.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load tsne_plot example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">tsne_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run tsne_plot plot function</span></span>
<span><span class="fu"><a href="reference/tsne_plot.html">tsne_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">tsne_data</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.80</span>,</span>
<span>  text_size <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  text_alpha <span class="op">=</span> <span class="fl">1.00</span>,</span>
<span>  ci_level <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  ellipse_alpha <span class="op">=</span> <span class="fl">0.10</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_NPG"</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">0.50</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-tsne_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/tsne_plot.html">?TOmicsVis::tsne_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/tsne_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/tsne_plot.html</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::tsne_plot</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_33-differential-expression-analyais">3.3 Differential Expression Analyais<a class="anchor" aria-label="anchor" href="#id_33-differential-expression-analyais"></a>
</h3>
<div class="section level4">
<h4 id="id_331-venn_plot">3.3.1 venn_plot<a class="anchor" aria-label="anchor" href="#id_331-venn_plot"></a>
</h4>
<p>Venn plot for stat common and unique gene among multiple sets.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load venn_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">venn_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run venn_plot plot function</span></span>
<span><span class="fu"><a href="reference/venn_plot.html">venn_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">venn_data</span>,</span>
<span>  line_type <span class="op">=</span> <span class="st">"blank"</span>,</span>
<span>  ellipse_shape <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_D3"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.60</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-venn_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/venn_plot.html">?TOmicsVis::venn_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/venn_plot.html</a>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::venn_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_332-flower_plot">3.3.2 flower_plot<a class="anchor" aria-label="anchor" href="#id_332-flower_plot"></a>
</h4>
<p>flower plot for stat common and unique gene among multiple sets.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">venn_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run flower_plot plot function</span></span>
<span><span class="fu"><a href="reference/flower_plot.html">flower_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">venn_data</span>,</span>
<span>  angle <span class="op">=</span> <span class="fl">90</span>,</span>
<span>  a <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  b <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  r <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  ellipse_col_pal <span class="op">=</span> <span class="st">"Spectral"</span>,</span>
<span>  circle_col <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>  label_text_cex <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-flower_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/flower_plot.html">?TOmicsVis::flower_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/flower_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/flower_plot.html</a>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::flower_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_333-circos_heatmap">3.3.3 circos_heatmap<a class="anchor" aria-label="anchor" href="#id_333-circos_heatmap"></a>
</h4>
<p>Circos heatmap plot for visualizing gene expressing in multiple samples.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load circos_heatmap_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">circos_heatmap_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run circos_heatmap plot function</span></span>
<span><span class="fu"><a href="reference/circos_heatmap.html">circos_heatmap</a></span><span class="op">(</span></span>
<span>  <span class="va">circos_heatmap_data</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#008800aa"</span>,</span>
<span>  mid_color <span class="op">=</span> <span class="st">"#ffffff"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  gap_size <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  cluster_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  distance_method <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  dend_height <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  rowname_size <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-circos_heatmap-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/circos_heatmap.html">?TOmicsVis::circos_heatmap</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/circos_heatmap.html</a>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::circos_heatmap</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_334-volcano_plot">3.3.4 volcano_plot<a class="anchor" aria-label="anchor" href="#id_334-volcano_plot"></a>
</h4>
<p>Volcano plot for visualizing differentailly expressed genes.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load deg_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">deg_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run volcano_plot plot function</span></span>
<span><span class="fu"><a href="reference/volcano_plot.html">volcano_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">deg_data</span>,</span>
<span>  log2fc_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  pq_value <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>  pq_cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  cutoff_line <span class="op">=</span> <span class="st">"longdash"</span>,</span>
<span>  point_shape <span class="op">=</span> <span class="st">"large_circle"</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  point_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  color_normal <span class="op">=</span> <span class="st">"#888888"</span>,</span>
<span>  color_log2fc <span class="op">=</span> <span class="st">"#008000"</span>,</span>
<span>  color_pvalue <span class="op">=</span> <span class="st">"#0088ee"</span>,</span>
<span>  color_Log2fc_p <span class="op">=</span> <span class="st">"#ff0000"</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  boxed_labels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  draw_connectors <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-volcano_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/volcano_plot.html">?TOmicsVis::volcano_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/volcano_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/volcano_plot.html</a>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::volcano_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_335-ma_plot">3.3.5 ma_plot<a class="anchor" aria-label="anchor" href="#id_335-ma_plot"></a>
</h4>
<p>MversusA plot for visualizing differentially expressed genes.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load deg_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">deg_data2</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run volcano_plot plot function</span></span>
<span><span class="fu"><a href="reference/ma_plot.html">ma_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">deg_data2</span>,</span>
<span>  foldchange <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  fdr_value <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  point_size <span class="op">=</span> <span class="fl">1.5</span>,</span>
<span>  color_up <span class="op">=</span> <span class="st">"#FF000088"</span>,</span>
<span>  color_down <span class="op">=</span> <span class="st">"#00880088"</span>,</span>
<span>  color_alpha <span class="op">=</span> <span class="fl">0.20</span>,</span>
<span>  top_method <span class="op">=</span> <span class="st">"fc"</span>,</span>
<span>  top_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">8</span>,</span>
<span>  label_box <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Group1 -versus- Group2"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Log2 mean expression"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Log2 fold change"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_minimal"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-ma_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/ma_plot.html">?TOmicsVis::ma_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/ma_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/ma_plot.html</a>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::ma_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_336-heatmap_group">3.3.6 heatmap_group<a class="anchor" aria-label="anchor" href="#id_336-heatmap_group"></a>
</h4>
<p>Heatmap group for visualizing grouped gene expression data.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">heatmap_group_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">heatmap_group_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 V2          V3          V4          V5          V6          V7</span></span>
<span><span class="co">#&gt; Groups     Control     Control     Control       Treat       Treat       Treat</span></span>
<span><span class="co">#&gt; GeneID    Control1    Control2    Control3      Treat1      Treat2      Treat3</span></span>
<span><span class="co">#&gt; Gene1   6.59934411 5.226266025 3.693287538 9.308119032 8.987864851        &lt;NA&gt;</span></span>
<span><span class="co">#&gt; Gene2  5.760380377 4.892783021 5.448923917  8.66208104 8.793319848 8.765914637</span></span>
<span><span class="co">#&gt; Gene3  9.561905115 4.549168157 3.998654922 9.790770004 7.133187551  7.37959102</span></span>
<span><span class="co">#&gt; Gene4  8.396409316  8.71705522  8.03906411 4.417013007 4.725269731 3.542216879</span></span>
<span></span>
<span><span class="co"># 2. Run heatmap_group plot function</span></span>
<span><span class="fu"><a href="reference/heatmap_group.html">heatmap_group</a></span><span class="op">(</span></span>
<span>  <span class="va">heatmap_group_data</span>,</span>
<span>  scale_data <span class="op">=</span> <span class="st">"none"</span>,</span>
<span>  clust_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  border_show <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  value_show <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#00880088"</span>,</span>
<span>  mid_color <span class="op">=</span> <span class="st">"#ffffff"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#ff000088"</span>,</span>
<span>  na_color <span class="op">=</span> <span class="st">"#ffaa00"</span>,</span>
<span>  x_angle <span class="op">=</span> <span class="fl">45</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmap_group-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/heatmap_group.html">?TOmicsVis::heatmap_group</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/heatmap_group.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/heatmap_group.html</a>.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::heatmap_group</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_34-advanced-analysis">3.4 Advanced Analysis<a class="anchor" aria-label="anchor" href="#id_34-advanced-analysis"></a>
</h3>
<div class="section level4">
<h4 id="id_341-trend_plot">3.4.1 trend_plot<a class="anchor" aria-label="anchor" href="#id_341-trend_plot"></a>
</h4>
<p>Trend plot for visualizing gene expression trend profile in multiple traits.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load chord_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">trend_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run trend_plot plot function</span></span>
<span><span class="fu"><a href="reference/trend_plot.html">trend_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">trend_data</span>,</span>
<span>  scale_method <span class="op">=</span> <span class="st">"globalminmax"</span>,</span>
<span>  miss_value <span class="op">=</span> <span class="st">"exclude"</span>,</span>
<span>  line_alpha <span class="op">=</span> <span class="fl">0.3</span>,</span>
<span>  show_points <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  show_boxplot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  num_column <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Traits"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Genes Expression"</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_Futurama"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sci_color_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  legend_pos <span class="op">=</span> <span class="st">"right"</span>,</span>
<span>  legend_dir <span class="op">=</span> <span class="st">"vertical"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-trend_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/trend_plot.html">?TOmicsVis::trend_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/trend_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/trend_plot.html</a>.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::trend_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_342-gene_cluster_trend">3.4.2 gene_cluster_trend<a class="anchor" aria-label="anchor" href="#id_342-gene_cluster_trend"></a>
</h4>
<p>Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_cluster_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_cluster_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;            Time1      Time2       Time3</span></span>
<span><span class="co">#&gt; Gene1  0.1774993  1.6563226 -1.15259948</span></span>
<span><span class="co">#&gt; Gene2 -0.5037254 -0.5207024  0.46416071</span></span>
<span><span class="co">#&gt; Gene3  0.1050310  0.6079246  0.72893247</span></span>
<span><span class="co">#&gt; Gene4 -1.1791537  0.4340085  0.41061745</span></span>
<span><span class="co">#&gt; Gene5  0.8368975 -0.7047414 -1.46114720</span></span>
<span><span class="co">#&gt; Gene6  0.2611762  0.1351524 -0.01890809</span></span>
<span></span>
<span><span class="co"># 2. Run plot function</span></span>
<span><span class="fu"><a href="reference/gene_cluster_trend.html">gene_cluster_trend</a></span><span class="op">(</span></span>
<span>  <span class="va">gene_cluster_data</span>,</span>
<span>  thres <span class="op">=</span> <span class="fl">0.25</span>,</span>
<span>  min_std <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"PiYG"</span>,</span>
<span>  cluster_num <span class="op">=</span> <span class="fl">4</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; 0 genes excluded.</span></span>
<span><span class="co">#&gt; 12 genes excluded.</span></span></code></pre></div>
<p><img src="reference/figures/README-gene_cluster_trend-1.png"><!-- --></p>
<div class="sourceCode" id="cb33"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a><span class="co">#&gt; NULL</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/gene_cluster_trend.html">?TOmicsVis::gene_cluster_trend</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/gene_cluster_trend.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/gene_cluster_trend.html</a>.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::gene_cluster_trend</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_343-gene_rank_plot">3.4.3 gene_rank_plot<a class="anchor" aria-label="anchor" href="#id_343-gene_rank_plot"></a>
</h4>
<p>Gene ranking dotplot for visualizing differentailly expressed genes.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">deg_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run gene_rank_plot plot function</span></span>
<span><span class="fu"><a href="reference/gene_rank_plot.html">gene_rank_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">deg_data</span>,</span>
<span>  log2fc <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"Spectral"</span>,</span>
<span>  top_n <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  genes_to_label <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  base_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  title <span class="op">=</span> <span class="st">"Gene ranking dotplot"</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Ranking of differentially expressed genes"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Log2FoldChange"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-gene_rank_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/gene_rank_plot.html">?TOmicsVis::gene_rank_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/gene_rank_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/gene_rank_plot.html</a>.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::gene_rank_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_344-wgcna_pipeline">3.4.4 wgcna_pipeline<a class="anchor" aria-label="anchor" href="#id_344-wgcna_pipeline"></a>
</h4>
<p>WGCNA analysis pipeline for RNA-Seq.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load wgcna_pipeline example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wgcna_gene_exp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">wgcna_sample_group</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run wgcna_pipeline plot function</span></span>
<span><span class="co"># wgcna_pipeline(wgcna_gene_exp, wgcna_sample_group)</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/wgcna_pipeline.html">?TOmicsVis::wgcna_pipeline</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/wgcna_pipeline.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/wgcna_pipeline.html</a>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::wgcna_pipeline</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_345-network_plot">3.4.5 network_plot<a class="anchor" aria-label="anchor" href="#id_345-network_plot"></a>
</h4>
<p>Network plot for analyzing and visualizing relationship of genes.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">network_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">network_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;   node1           node2</span></span>
<span><span class="co">#&gt; 1  ABL2  PC-3p-5622_465</span></span>
<span><span class="co">#&gt; 2  ABL2  PC-5p-33384_55</span></span>
<span><span class="co">#&gt; 3  ABL2  chi-miR-107-3p</span></span>
<span><span class="co">#&gt; 4  ABL2  chi-miR-15b-5p</span></span>
<span><span class="co">#&gt; 5 CASP2 PC-3p-10204_250</span></span>
<span><span class="co">#&gt; 6 CASP2    bta-miR-6123</span></span>
<span><span class="va">network_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">network_data</span>, <span class="fl">150</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run network_plot plot function</span></span>
<span><span class="fu"><a href="reference/network_plot.html">network_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">network_data</span>,</span>
<span>  calcBy <span class="op">=</span> <span class="st">"degree"</span>,</span>
<span>  degreeValue <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  nodeColorNormal <span class="op">=</span> <span class="st">"#ff880055"</span>,</span>
<span>  nodeBorderColor <span class="op">=</span> <span class="st">"#FFFFFF"</span>,</span>
<span>  nodeColorFrom <span class="op">=</span> <span class="st">"#ff000055"</span>,</span>
<span>  nodeColorTo <span class="op">=</span> <span class="st">"#00880055"</span>,</span>
<span>  nodeShapeNormal <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  nodeShapeSpatial <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  nodeSize <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  labelSize <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  edgeCurved <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  netLayout <span class="op">=</span> <span class="st">"layout_on_sphere"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-network_plot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/network_plot.html">?TOmicsVis::network_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/network_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/network_plot.html</a>.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::network_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_346-heatmap_cluster">3.4.6 heatmap_cluster<a class="anchor" aria-label="anchor" href="#id_346-heatmap_cluster"></a>
</h4>
<p>Heatmap cluster plot for visualizing clustered gene expression data.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_exp</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run plot function</span></span>
<span><span class="fu"><a href="reference/heatmap_cluster.html">heatmap_cluster</a></span><span class="op">(</span></span>
<span>  <span class="va">gene_exp</span>,</span>
<span>  dist_method <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  hc_method <span class="op">=</span> <span class="st">"average"</span>,</span>
<span>  k_num <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"Spectral"</span>,</span>
<span>  cluster_pal <span class="op">=</span> <span class="st">"Set1"</span>,</span>
<span>  gaps_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  angle_col <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  base_size <span class="op">=</span> <span class="fl">12</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmap_cluster-1.png"><!-- --></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" tabindex="-1"></a><span class="co">#&gt; Using Cluster, gene as id variables</span></span></code></pre></div>
<p><img src="reference/figures/README-heatmap_cluster-2.png"><!-- --><img src="reference/figures/README-heatmap_cluster-3.png"><!-- --></p>
<p>Get help using command <code><a href="reference/heatmap_cluster.html">?TOmicsVis::heatmap_cluster</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/heatmap_cluster.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/heatmap_cluster.html</a>.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::heatmap_cluster</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_35-go-and-kegg-enrichment">3.5 GO and KEGG Enrichment<a class="anchor" aria-label="anchor" href="#id_35-go-and-kegg-enrichment"></a>
</h3>
<div class="section level4">
<h4 id="id_351-chord_plot">3.5.1 chord_plot<a class="anchor" aria-label="anchor" href="#id_351-chord_plot"></a>
</h4>
<p>Chord plot for visualizing the relationships of pathways and genes.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load chord_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">chord_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">chord_data</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"G1"</span>,<span class="st">"G2"</span>,<span class="st">"G3"</span>,<span class="st">"G4"</span>,<span class="st">"G5"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run chord_plot plot function</span></span>
<span><span class="fu"><a href="reference/chord_plot.html">chord_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">chord_data</span>,</span>
<span>  multi_colors <span class="op">=</span> <span class="st">"VividColors"</span>,</span>
<span>  color_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  link_visible <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  link_dir <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,</span>
<span>  link_type <span class="op">=</span> <span class="st">"diffHeight"</span>,</span>
<span>  sector_scale <span class="op">=</span> <span class="st">"Scale"</span>,</span>
<span>  width_circle <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  dist_name <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  label_dir <span class="op">=</span> <span class="st">"Vertical"</span>,</span>
<span>  dist_label <span class="op">=</span> <span class="fl">0.3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-chord_plot-1.png"><!-- --></p>
<div class="sourceCode" id="cb45"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" tabindex="-1"></a><span class="co">#&gt;      rn cn     value1      value2 o1 o2         x1         x2       col</span></span>
<span id="cb45-2"><a href="#cb45-2" tabindex="-1"></a><span class="co">#&gt; 1    S1 G1 0.06557377 0.024844720  5 20 1.00000000 1.00000000 #DF4A6E7F</span></span>
<span id="cb45-3"><a href="#cb45-3" tabindex="-1"></a><span class="co">#&gt; 2    S2 G1 0.15555556 0.043478261  5 19 1.00000000 0.97515528 #D3AFDC7F</span></span>
<span id="cb45-4"><a href="#cb45-4" tabindex="-1"></a><span class="co">#&gt; 3    S3 G1 0.16363636 0.055900621  5 18 1.00000000 0.93167702 #73A1E57F</span></span>
<span id="cb45-5"><a href="#cb45-5" tabindex="-1"></a><span class="co">#&gt; 4    S4 G1 0.25454545 0.086956522  5 17 1.00000000 0.87577640 #DFCD647F</span></span>
<span id="cb45-6"><a href="#cb45-6" tabindex="-1"></a><span class="co">#&gt; 5    S5 G1 0.04545455 0.006211180  5 16 1.00000000 0.78881988 #68E6D67F</span></span>
<span id="cb45-7"><a href="#cb45-7" tabindex="-1"></a><span class="co">#&gt; 6    S6 G1 0.16949153 0.062111801  5 15 1.00000000 0.78260870 #67E8987F</span></span>
<span id="cb45-8"><a href="#cb45-8" tabindex="-1"></a><span class="co">#&gt; 7    S7 G1 0.09677419 0.018633540  5 14 1.00000000 0.72049689 #7C6ADD7F</span></span>
<span id="cb45-9"><a href="#cb45-9" tabindex="-1"></a><span class="co">#&gt; 8    S8 G1 0.28846154 0.093167702  5 13 1.00000000 0.70186335 #BEE4E07F</span></span>
<span id="cb45-10"><a href="#cb45-10" tabindex="-1"></a><span class="co">#&gt; 9    S9 G1 0.13513514 0.031055901  5 12 1.00000000 0.60869565 #6FAA787F</span></span>
<span id="cb45-11"><a href="#cb45-11" tabindex="-1"></a><span class="co">#&gt; 10  S10 G1 0.25925926 0.043478261  5 11 1.00000000 0.57763975 #E2CFDA7F</span></span>
<span id="cb45-12"><a href="#cb45-12" tabindex="-1"></a><span class="co">#&gt; 11  S11 G1 0.17142857 0.037267081  5 10 1.00000000 0.53416149 #D3B5907F</span></span>
<span id="cb45-13"><a href="#cb45-13" tabindex="-1"></a><span class="co">#&gt; 12  S12 G1 0.22448980 0.068322981  5  9 1.00000000 0.49689441 #6C54897F</span></span>
<span id="cb45-14"><a href="#cb45-14" tabindex="-1"></a><span class="co">#&gt; 13  S13 G1 0.26086957 0.111801242  5  8 1.00000000 0.42857143 #DF53CC7F</span></span>
<span id="cb45-15"><a href="#cb45-15" tabindex="-1"></a><span class="co">#&gt; 14  S14 G1 0.04545455 0.006211180  5  7 1.00000000 0.31677019 #ABE8BD7F</span></span>
<span id="cb45-16"><a href="#cb45-16" tabindex="-1"></a><span class="co">#&gt; 15  S15 G1 0.08620690 0.031055901  5  6 1.00000000 0.31055901 #A9D9747F</span></span>
<span id="cb45-17"><a href="#cb45-17" tabindex="-1"></a><span class="co">#&gt; 16  S16 G1 0.25000000 0.074534161  5  5 1.00000000 0.27950311 #74EC4F7F</span></span>
<span id="cb45-18"><a href="#cb45-18" tabindex="-1"></a><span class="co">#&gt; 17  S17 G1 0.39130435 0.055900621  5  4 1.00000000 0.20496894 #78CDE87F</span></span>
<span id="cb45-19"><a href="#cb45-19" tabindex="-1"></a><span class="co">#&gt; 18  S18 G1 0.10000000 0.031055901  5  3 1.00000000 0.14906832 #E095427F</span></span>
<span id="cb45-20"><a href="#cb45-20" tabindex="-1"></a><span class="co">#&gt; 19  S19 G1 0.10000000 0.024844720  5  2 1.00000000 0.11801242 #D987877F</span></span>
<span id="cb45-21"><a href="#cb45-21" tabindex="-1"></a><span class="co">#&gt; 20  S20 G1 0.25862069 0.093167702  5  1 1.00000000 0.09316770 #A83FE67F</span></span>
<span id="cb45-22"><a href="#cb45-22" tabindex="-1"></a><span class="co">#&gt; 21   S1 G2 0.26229508 0.096385542  4 20 0.93442623 1.00000000 #DF4A6E7F</span></span>
<span id="cb45-23"><a href="#cb45-23" tabindex="-1"></a><span class="co">#&gt; 22   S2 G2 0.24444444 0.066265060  4 19 0.84444444 0.90361446 #D3AFDC7F</span></span>
<span id="cb45-24"><a href="#cb45-24" tabindex="-1"></a><span class="co">#&gt; 23   S3 G2 0.03636364 0.012048193  4 18 0.83636364 0.83734940 #73A1E57F</span></span>
<span id="cb45-25"><a href="#cb45-25" tabindex="-1"></a><span class="co">#&gt; 24   S4 G2 0.16363636 0.054216867  4 17 0.74545455 0.82530120 #DFCD647F</span></span>
<span id="cb45-26"><a href="#cb45-26" tabindex="-1"></a><span class="co">#&gt; 25   S5 G2 0.04545455 0.006024096  4 16 0.95454545 0.77108434 #68E6D67F</span></span>
<span id="cb45-27"><a href="#cb45-27" tabindex="-1"></a><span class="co">#&gt; 26   S6 G2 0.30508475 0.108433735  4 15 0.83050847 0.76506024 #67E8987F</span></span>
<span id="cb45-28"><a href="#cb45-28" tabindex="-1"></a><span class="co">#&gt; 27   S7 G2 0.25806452 0.048192771  4 14 0.90322581 0.65662651 #7C6ADD7F</span></span>
<span id="cb45-29"><a href="#cb45-29" tabindex="-1"></a><span class="co">#&gt; 28   S8 G2 0.07692308 0.024096386  4 13 0.71153846 0.60843373 #BEE4E07F</span></span>
<span id="cb45-30"><a href="#cb45-30" tabindex="-1"></a><span class="co">#&gt; 29   S9 G2 0.10810811 0.024096386  4 12 0.86486486 0.58433735 #6FAA787F</span></span>
<span id="cb45-31"><a href="#cb45-31" tabindex="-1"></a><span class="co">#&gt; 30  S10 G2 0.33333333 0.054216867  4 11 0.74074074 0.56024096 #E2CFDA7F</span></span>
<span id="cb45-32"><a href="#cb45-32" tabindex="-1"></a><span class="co">#&gt; 31  S11 G2 0.20000000 0.042168675  4 10 0.82857143 0.50602410 #D3B5907F</span></span>
<span id="cb45-33"><a href="#cb45-33" tabindex="-1"></a><span class="co">#&gt; 32  S12 G2 0.04081633 0.012048193  4  9 0.77551020 0.46385542 #6C54897F</span></span>
<span id="cb45-34"><a href="#cb45-34" tabindex="-1"></a><span class="co">#&gt; 33  S13 G2 0.18840580 0.078313253  4  8 0.73913043 0.45180723 #DF53CC7F</span></span>
<span id="cb45-35"><a href="#cb45-35" tabindex="-1"></a><span class="co">#&gt; 34  S14 G2 0.09090909 0.012048193  4  7 0.95454545 0.37349398 #ABE8BD7F</span></span>
<span id="cb45-36"><a href="#cb45-36" tabindex="-1"></a><span class="co">#&gt; 35  S15 G2 0.22413793 0.078313253  4  6 0.91379310 0.36144578 #A9D9747F</span></span>
<span id="cb45-37"><a href="#cb45-37" tabindex="-1"></a><span class="co">#&gt; 36  S16 G2 0.14583333 0.042168675  4  5 0.75000000 0.28313253 #74EC4F7F</span></span>
<span id="cb45-38"><a href="#cb45-38" tabindex="-1"></a><span class="co">#&gt; 37  S17 G2 0.08695652 0.012048193  4  4 0.60869565 0.24096386 #78CDE87F</span></span>
<span id="cb45-39"><a href="#cb45-39" tabindex="-1"></a><span class="co">#&gt; 38  S18 G2 0.26000000 0.078313253  4  3 0.90000000 0.22891566 #E095427F</span></span>
<span id="cb45-40"><a href="#cb45-40" tabindex="-1"></a><span class="co">#&gt; 39  S19 G2 0.30000000 0.072289157  4  2 0.90000000 0.15060241 #D987877F</span></span>
<span id="cb45-41"><a href="#cb45-41" tabindex="-1"></a><span class="co">#&gt; 40  S20 G2 0.22413793 0.078313253  4  1 0.74137931 0.07831325 #A83FE67F</span></span>
<span id="cb45-42"><a href="#cb45-42" tabindex="-1"></a><span class="co">#&gt; 41   S1 G3 0.19672131 0.080000000  3 20 0.67213115 1.00000000 #DF4A6E7F</span></span>
<span id="cb45-43"><a href="#cb45-43" tabindex="-1"></a><span class="co">#&gt; 42   S2 G3 0.04444444 0.013333333  3 19 0.60000000 0.92000000 #D3AFDC7F</span></span>
<span id="cb45-44"><a href="#cb45-44" tabindex="-1"></a><span class="co">#&gt; 43   S3 G3 0.30909091 0.113333333  3 18 0.80000000 0.90666667 #73A1E57F</span></span>
<span id="cb45-45"><a href="#cb45-45" tabindex="-1"></a><span class="co">#&gt; 44   S4 G3 0.21818182 0.080000000  3 17 0.58181818 0.79333333 #DFCD647F</span></span>
<span id="cb45-46"><a href="#cb45-46" tabindex="-1"></a><span class="co">#&gt; 45   S5 G3 0.31818182 0.046666667  3 16 0.90909091 0.71333333 #68E6D67F</span></span>
<span id="cb45-47"><a href="#cb45-47" tabindex="-1"></a><span class="co">#&gt; 46   S6 G3 0.15254237 0.060000000  3 15 0.52542373 0.66666667 #67E8987F</span></span>
<span id="cb45-48"><a href="#cb45-48" tabindex="-1"></a><span class="co">#&gt; 47   S7 G3 0.12903226 0.026666667  3 14 0.64516129 0.60666667 #7C6ADD7F</span></span>
<span id="cb45-49"><a href="#cb45-49" tabindex="-1"></a><span class="co">#&gt; 48   S8 G3 0.17307692 0.060000000  3 13 0.63461538 0.58000000 #BEE4E07F</span></span>
<span id="cb45-50"><a href="#cb45-50" tabindex="-1"></a><span class="co">#&gt; 49   S9 G3 0.18918919 0.046666667  3 12 0.75675676 0.52000000 #6FAA787F</span></span>
<span id="cb45-51"><a href="#cb45-51" tabindex="-1"></a><span class="co">#&gt; 50  S10 G3 0.03703704 0.006666667  3 11 0.40740741 0.47333333 #E2CFDA7F</span></span>
<span id="cb45-52"><a href="#cb45-52" tabindex="-1"></a><span class="co">#&gt; 51  S11 G3 0.14285714 0.033333333  3 10 0.62857143 0.46666667 #D3B5907F</span></span>
<span id="cb45-53"><a href="#cb45-53" tabindex="-1"></a><span class="co">#&gt; 52  S12 G3 0.04081633 0.013333333  3  9 0.73469388 0.43333333 #6C54897F</span></span>
<span id="cb45-54"><a href="#cb45-54" tabindex="-1"></a><span class="co">#&gt; 53  S13 G3 0.11594203 0.053333333  3  8 0.55072464 0.42000000 #DF53CC7F</span></span>
<span id="cb45-55"><a href="#cb45-55" tabindex="-1"></a><span class="co">#&gt; 54  S14 G3 0.09090909 0.013333333  3  7 0.86363636 0.36666667 #ABE8BD7F</span></span>
<span id="cb45-56"><a href="#cb45-56" tabindex="-1"></a><span class="co">#&gt; 55  S15 G3 0.10344828 0.040000000  3  6 0.68965517 0.35333333 #A9D9747F</span></span>
<span id="cb45-57"><a href="#cb45-57" tabindex="-1"></a><span class="co">#&gt; 56  S16 G3 0.31250000 0.100000000  3  5 0.60416667 0.31333333 #74EC4F7F</span></span>
<span id="cb45-58"><a href="#cb45-58" tabindex="-1"></a><span class="co">#&gt; 57  S17 G3 0.04347826 0.006666667  3  4 0.52173913 0.21333333 #78CDE87F</span></span>
<span id="cb45-59"><a href="#cb45-59" tabindex="-1"></a><span class="co">#&gt; 58  S18 G3 0.26000000 0.086666667  3  3 0.64000000 0.20666667 #E095427F</span></span>
<span id="cb45-60"><a href="#cb45-60" tabindex="-1"></a><span class="co">#&gt; 59  S19 G3 0.10000000 0.026666667  3  2 0.60000000 0.12000000 #D987877F</span></span>
<span id="cb45-61"><a href="#cb45-61" tabindex="-1"></a><span class="co">#&gt; 60  S20 G3 0.24137931 0.093333333  3  1 0.51724138 0.09333333 #A83FE67F</span></span>
<span id="cb45-62"><a href="#cb45-62" tabindex="-1"></a><span class="co">#&gt; 61   S1 G4 0.29508197 0.079295154  2 20 0.47540984 1.00000000 #DF4A6E7F</span></span>
<span id="cb45-63"><a href="#cb45-63" tabindex="-1"></a><span class="co">#&gt; 62   S2 G4 0.33333333 0.066079295  2 19 0.55555556 0.92070485 #D3AFDC7F</span></span>
<span id="cb45-64"><a href="#cb45-64" tabindex="-1"></a><span class="co">#&gt; 63   S3 G4 0.29090909 0.070484581  2 18 0.49090909 0.85462555 #73A1E57F</span></span>
<span id="cb45-65"><a href="#cb45-65" tabindex="-1"></a><span class="co">#&gt; 64   S4 G4 0.05454545 0.013215859  2 17 0.36363636 0.78414097 #DFCD647F</span></span>
<span id="cb45-66"><a href="#cb45-66" tabindex="-1"></a><span class="co">#&gt; 65   S5 G4 0.04545455 0.004405286  2 16 0.59090909 0.77092511 #68E6D67F</span></span>
<span id="cb45-67"><a href="#cb45-67" tabindex="-1"></a><span class="co">#&gt; 66   S6 G4 0.22033898 0.057268722  2 15 0.37288136 0.76651982 #67E8987F</span></span>
<span id="cb45-68"><a href="#cb45-68" tabindex="-1"></a><span class="co">#&gt; 67   S7 G4 0.48387097 0.066079295  2 14 0.51612903 0.70925110 #7C6ADD7F</span></span>
<span id="cb45-69"><a href="#cb45-69" tabindex="-1"></a><span class="co">#&gt; 68   S8 G4 0.25000000 0.057268722  2 13 0.46153846 0.64317181 #BEE4E07F</span></span>
<span id="cb45-70"><a href="#cb45-70" tabindex="-1"></a><span class="co">#&gt; 69   S9 G4 0.08108108 0.013215859  2 12 0.56756757 0.58590308 #6FAA787F</span></span>
<span id="cb45-71"><a href="#cb45-71" tabindex="-1"></a><span class="co">#&gt; 70  S10 G4 0.14814815 0.017621145  2 11 0.37037037 0.57268722 #E2CFDA7F</span></span>
<span id="cb45-72"><a href="#cb45-72" tabindex="-1"></a><span class="co">#&gt; 71  S11 G4 0.22857143 0.035242291  2 10 0.48571429 0.55506608 #D3B5907F</span></span>
<span id="cb45-73"><a href="#cb45-73" tabindex="-1"></a><span class="co">#&gt; 72  S12 G4 0.32653061 0.070484581  2  9 0.69387755 0.51982379 #6C54897F</span></span>
<span id="cb45-74"><a href="#cb45-74" tabindex="-1"></a><span class="co">#&gt; 73  S13 G4 0.20289855 0.061674009  2  8 0.43478261 0.44933921 #DF53CC7F</span></span>
<span id="cb45-75"><a href="#cb45-75" tabindex="-1"></a><span class="co">#&gt; 74  S14 G4 0.63636364 0.061674009  2  7 0.77272727 0.38766520 #ABE8BD7F</span></span>
<span id="cb45-76"><a href="#cb45-76" tabindex="-1"></a><span class="co">#&gt; 75  S15 G4 0.27586207 0.070484581  2  6 0.58620690 0.32599119 #A9D9747F</span></span>
<span id="cb45-77"><a href="#cb45-77" tabindex="-1"></a><span class="co">#&gt; 76  S16 G4 0.25000000 0.052863436  2  5 0.29166667 0.25550661 #74EC4F7F</span></span>
<span id="cb45-78"><a href="#cb45-78" tabindex="-1"></a><span class="co">#&gt; 77  S17 G4 0.43478261 0.044052863  2  4 0.47826087 0.20264317 #78CDE87F</span></span>
<span id="cb45-79"><a href="#cb45-79" tabindex="-1"></a><span class="co">#&gt; 78  S18 G4 0.22000000 0.048458150  2  3 0.38000000 0.15859031 #E095427F</span></span>
<span id="cb45-80"><a href="#cb45-80" tabindex="-1"></a><span class="co">#&gt; 79  S19 G4 0.35000000 0.061674009  2  2 0.50000000 0.11013216 #D987877F</span></span>
<span id="cb45-81"><a href="#cb45-81" tabindex="-1"></a><span class="co">#&gt; 80  S20 G4 0.18965517 0.048458150  2  1 0.27586207 0.04845815 #A83FE67F</span></span>
<span id="cb45-82"><a href="#cb45-82" tabindex="-1"></a><span class="co">#&gt; 81   S1 G5 0.18032787 0.057291667  1 20 0.18032787 1.00000000 #DF4A6E7F</span></span>
<span id="cb45-83"><a href="#cb45-83" tabindex="-1"></a><span class="co">#&gt; 82   S2 G5 0.22222222 0.052083333  1 19 0.22222222 0.94270833 #D3AFDC7F</span></span>
<span id="cb45-84"><a href="#cb45-84" tabindex="-1"></a><span class="co">#&gt; 83   S3 G5 0.20000000 0.057291667  1 18 0.20000000 0.89062500 #73A1E57F</span></span>
<span id="cb45-85"><a href="#cb45-85" tabindex="-1"></a><span class="co">#&gt; 84   S4 G5 0.30909091 0.088541667  1 17 0.30909091 0.83333333 #DFCD647F</span></span>
<span id="cb45-86"><a href="#cb45-86" tabindex="-1"></a><span class="co">#&gt; 85   S5 G5 0.54545455 0.062500000  1 16 0.54545455 0.74479167 #68E6D67F</span></span>
<span id="cb45-87"><a href="#cb45-87" tabindex="-1"></a><span class="co">#&gt; 86   S6 G5 0.15254237 0.046875000  1 15 0.15254237 0.68229167 #67E8987F</span></span>
<span id="cb45-88"><a href="#cb45-88" tabindex="-1"></a><span class="co">#&gt; 87   S7 G5 0.03225806 0.005208333  1 14 0.03225806 0.63541667 #7C6ADD7F</span></span>
<span id="cb45-89"><a href="#cb45-89" tabindex="-1"></a><span class="co">#&gt; 88   S8 G5 0.21153846 0.057291667  1 13 0.21153846 0.63020833 #BEE4E07F</span></span>
<span id="cb45-90"><a href="#cb45-90" tabindex="-1"></a><span class="co">#&gt; 89   S9 G5 0.48648649 0.093750000  1 12 0.48648649 0.57291667 #6FAA787F</span></span>
<span id="cb45-91"><a href="#cb45-91" tabindex="-1"></a><span class="co">#&gt; 90  S10 G5 0.22222222 0.031250000  1 11 0.22222222 0.47916667 #E2CFDA7F</span></span>
<span id="cb45-92"><a href="#cb45-92" tabindex="-1"></a><span class="co">#&gt; 91  S11 G5 0.25714286 0.046875000  1 10 0.25714286 0.44791667 #D3B5907F</span></span>
<span id="cb45-93"><a href="#cb45-93" tabindex="-1"></a><span class="co">#&gt; 92  S12 G5 0.36734694 0.093750000  1  9 0.36734694 0.40104167 #6C54897F</span></span>
<span id="cb45-94"><a href="#cb45-94" tabindex="-1"></a><span class="co">#&gt; 93  S13 G5 0.23188406 0.083333333  1  8 0.23188406 0.30729167 #DF53CC7F</span></span>
<span id="cb45-95"><a href="#cb45-95" tabindex="-1"></a><span class="co">#&gt; 94  S14 G5 0.13636364 0.015625000  1  7 0.13636364 0.22395833 #ABE8BD7F</span></span>
<span id="cb45-96"><a href="#cb45-96" tabindex="-1"></a><span class="co">#&gt; 95  S15 G5 0.31034483 0.093750000  1  6 0.31034483 0.20833333 #A9D9747F</span></span>
<span id="cb45-97"><a href="#cb45-97" tabindex="-1"></a><span class="co">#&gt; 96  S16 G5 0.04166667 0.010416667  1  5 0.04166667 0.11458333 #74EC4F7F</span></span>
<span id="cb45-98"><a href="#cb45-98" tabindex="-1"></a><span class="co">#&gt; 97  S17 G5 0.04347826 0.005208333  1  4 0.04347826 0.10416667 #78CDE87F</span></span>
<span id="cb45-99"><a href="#cb45-99" tabindex="-1"></a><span class="co">#&gt; 98  S18 G5 0.16000000 0.041666667  1  3 0.16000000 0.09895833 #E095427F</span></span>
<span id="cb45-100"><a href="#cb45-100" tabindex="-1"></a><span class="co">#&gt; 99  S19 G5 0.15000000 0.031250000  1  2 0.15000000 0.05729167 #D987877F</span></span>
<span id="cb45-101"><a href="#cb45-101" tabindex="-1"></a><span class="co">#&gt; 100 S20 G5 0.08620690 0.026041667  1  1 0.08620690 0.02604167 #A83FE67F</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/chord_plot.html">?TOmicsVis::chord_plot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html</a>.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::chord_plot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_352-go_enrich">3.5.2 go_enrich<a class="anchor" aria-label="anchor" href="#id_352-go_enrich"></a>
</h4>
<p>GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co">#&gt;       id</span></span>
<span><span class="co">#&gt; 1 gene_1</span></span>
<span><span class="co">#&gt; 2 gene_2</span></span>
<span><span class="co">#&gt; 3 gene_3</span></span>
<span><span class="co">#&gt; 4 gene_4</span></span>
<span><span class="co">#&gt; 5 gene_5</span></span>
<span><span class="co">#&gt; 6 gene_6</span></span>
<span><span class="co">#&gt;                                                                                                 biological_process</span></span>
<span><span class="co">#&gt; 1 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 2                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 3                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4 GO:0003181(atrioventricular valve morphogenesis);GO:0003128(heart field specification);GO:0001756(somitogenesis)</span></span>
<span><span class="co">#&gt; 5                                                                                GO:0006956(complement activation)</span></span>
<span><span class="co">#&gt; 6                                                                                                             &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                    cellular_component</span></span>
<span><span class="co">#&gt; 1    GO:0005576(extracellular region)</span></span>
<span><span class="co">#&gt; 2     GO:0005615(extracellular space)</span></span>
<span><span class="co">#&gt; 3                                &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4    GO:0005576(extracellular region)</span></span>
<span><span class="co">#&gt; 5 GO:0005579(membrane attack complex)</span></span>
<span><span class="co">#&gt; 6                                &lt;NA&gt;</span></span>
<span><span class="co">#&gt;                             molecular_function</span></span>
<span><span class="co">#&gt; 1                                         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2 GO:0004866(endopeptidase inhibitor activity)</span></span>
<span><span class="co">#&gt; 3                                         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 5                                         &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6                                         &lt;NA&gt;</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co">#&gt;        id log2FC</span></span>
<span><span class="co">#&gt; 1  gene_5  -1.20</span></span>
<span><span class="co">#&gt; 2 gene_12  -1.25</span></span>
<span><span class="co">#&gt; 3 gene_15  -1.30</span></span>
<span><span class="co">#&gt; 4 gene_22  -1.35</span></span>
<span><span class="co">#&gt; 5 gene_28  -1.40</span></span>
<span><span class="co">#&gt; 6 gene_33  -1.45</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich analysis function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/go_enrich.html">go_enrich</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;           ID           ontology                                    Description</span></span>
<span><span class="co">#&gt; 1 GO:0000015 cellular component              phosphopyruvate hydratase complex</span></span>
<span><span class="co">#&gt; 2 GO:0000027 biological process               ribosomal large subunit assembly</span></span>
<span><span class="co">#&gt; 3 GO:0000028 biological process               ribosomal small subunit assembly</span></span>
<span><span class="co">#&gt; 4 GO:0000280 biological process                               nuclear division</span></span>
<span><span class="co">#&gt; 5 GO:0000287 molecular function                          magnesium ion binding</span></span>
<span><span class="co">#&gt; 6 GO:0000334 molecular function 3-hydroxyanthranilate 3,4-dioxygenase activity</span></span>
<span><span class="co">#&gt;   GeneRatio BgRatio      pvalue   p.adjust     qvalue</span></span>
<span><span class="co">#&gt; 1      1/49   1/431 0.113689095 0.21992317 0.21992317</span></span>
<span><span class="co">#&gt; 2      1/49   1/431 0.113689095 0.21992317 0.21992317</span></span>
<span><span class="co">#&gt; 3      2/49   5/431 0.101279745 0.21992317 0.21992317</span></span>
<span><span class="co">#&gt; 4      5/49   9/431 0.001384956 0.08171238 0.08171238</span></span>
<span><span class="co">#&gt; 5      1/49   1/431 0.113689095 0.21992317 0.21992317</span></span>
<span><span class="co">#&gt; 6      1/49   1/431 0.113689095 0.21992317 0.21992317</span></span>
<span><span class="co">#&gt;                                     geneID Count</span></span>
<span><span class="co">#&gt; 1                                  gene_89     1</span></span>
<span><span class="co">#&gt; 2                                 gene_276     1</span></span>
<span><span class="co">#&gt; 3                         gene_12/gene_557     2</span></span>
<span><span class="co">#&gt; 4 gene_35/gene_58/gene_73/gene_91/gene_158     5</span></span>
<span><span class="co">#&gt; 5                                  gene_89     1</span></span>
<span><span class="co">#&gt; 6                                 gene_459     1</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/go_enrich.html">?TOmicsVis::go_enrich</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html</a>.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_353-go_enrich_stat">3.5.3 go_enrich_stat<a class="anchor" aria-label="anchor" href="#id_353-go_enrich_stat"></a>
</h4>
<p>GO enrichment analysis and stat plot based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_stat analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_stat.html">go_enrich_stat</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max_go_item <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  strip_fill <span class="op">=</span> <span class="st">"#CDCDCD"</span>,</span>
<span>  xtext_angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_stat-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_stat.html">?TOmicsVis::go_enrich_stat</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html</a>.</p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_stat</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_354-go_enrich_bar">3.5.4 go_enrich_bar<a class="anchor" aria-label="anchor" href="#id_354-go_enrich_bar"></a>
</h4>
<p>GO enrichment analysis and bar plot based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_bar analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_bar.html">go_enrich_bar</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for fill is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for fill, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_bar-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_bar.html">?TOmicsVis::go_enrich_bar</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html</a>.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_bar</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_355-go_enrich_dot">3.5.5 go_enrich_dot<a class="anchor" aria-label="anchor" href="#id_355-go_enrich_dot"></a>
</h4>
<p>GO enrichment analysis and dot plot based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_dot analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_dot.html">go_enrich_dot</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_dot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_dot.html">?TOmicsVis::go_enrich_dot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html</a>.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_dot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_356-go_enrich_tree">3.5.6 go_enrich_tree<a class="anchor" aria-label="anchor" href="#id_356-go_enrich_tree"></a>
</h4>
<p>GO enrichment analysis and tree plot based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_tree analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_tree.html">go_enrich_tree</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  hclust_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_tree-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_tree.html">?TOmicsVis::go_enrich_tree</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_tree.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_tree.html</a>.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_tree</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_357-go_enrich_net">3.5.7 go_enrich_net<a class="anchor" aria-label="anchor" href="#id_357-go_enrich_net"></a>
</h4>
<p>GO enrichment analysis and net plot based on GO annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_net analysis function</span></span>
<span><span class="fu"><a href="reference/go_enrich_net.html">go_enrich_net</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  net_layout <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  net_circular <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-go_enrich_net-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/go_enrich_net.html">?TOmicsVis::go_enrich_net</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html</a>.</p>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_net</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_358-kegg_enrich">3.5.8 kegg_enrich<a class="anchor" aria-label="anchor" href="#id_358-kegg_enrich"></a>
</h4>
<p>KEGG enrichment analysis based on KEGG annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co">#&gt;       id</span></span>
<span><span class="co">#&gt; 1 gene_1</span></span>
<span><span class="co">#&gt; 2 gene_2</span></span>
<span><span class="co">#&gt; 3 gene_3</span></span>
<span><span class="co">#&gt; 4 gene_4</span></span>
<span><span class="co">#&gt; 5 gene_5</span></span>
<span><span class="co">#&gt; 6 gene_6</span></span>
<span><span class="co">#&gt;                                                                                                                              kegg_pathway</span></span>
<span><span class="co">#&gt; 1 ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2                                                                                            ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 3 ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 4 ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 5                                                                                            ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 6                                                                                                                       ko04142(Lysosome)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co">#&gt;        id log2FC</span></span>
<span><span class="co">#&gt; 1  gene_5  -1.20</span></span>
<span><span class="co">#&gt; 2 gene_12  -1.25</span></span>
<span><span class="co">#&gt; 3 gene_15  -1.30</span></span>
<span><span class="co">#&gt; 4 gene_22  -1.35</span></span>
<span><span class="co">#&gt; 5 gene_28  -1.40</span></span>
<span><span class="co">#&gt; 6 gene_33  -1.45</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich analysis function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/kegg_enrich.html">kegg_enrich</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;              ID                    Description GeneRatio BgRatio     pvalue</span></span>
<span><span class="co">#&gt; ko00240 ko00240          Pyrimidine metabolism      3/38  10/504 0.03283089</span></span>
<span><span class="co">#&gt; ko04016 ko04016 MAPK signaling pathway - plant      2/38   6/504 0.06848785</span></span>
<span><span class="co">#&gt; ko00052 ko00052           Galactose metabolism      1/38   1/504 0.07539683</span></span>
<span><span class="co">#&gt; ko04066 ko04066        HIF-1 signaling pathway      2/38   7/504 0.09139001</span></span>
<span><span class="co">#&gt; ko03010 ko03010                       Ribosome     10/38  87/504 0.09838857</span></span>
<span><span class="co">#&gt; ko01200 ko01200              Carbon metabolism      3/38  17/504 0.12885914</span></span>
<span><span class="co">#&gt;          p.adjust    qvalue</span></span>
<span><span class="co">#&gt; ko00240 0.6627232 0.6627232</span></span>
<span><span class="co">#&gt; ko04016 0.6627232 0.6627232</span></span>
<span><span class="co">#&gt; ko00052 0.6627232 0.6627232</span></span>
<span><span class="co">#&gt; ko04066 0.6627232 0.6627232</span></span>
<span><span class="co">#&gt; ko03010 0.6627232 0.6627232</span></span>
<span><span class="co">#&gt; ko01200 0.6627232 0.6627232</span></span>
<span><span class="co">#&gt;                                                                                          geneID</span></span>
<span><span class="co">#&gt; ko00240                                                                gene_56/gene_58/gene_112</span></span>
<span><span class="co">#&gt; ko04016                                                                         gene_56/gene_58</span></span>
<span><span class="co">#&gt; ko00052                                                                                 gene_22</span></span>
<span><span class="co">#&gt; ko04066                                                                         gene_15/gene_89</span></span>
<span><span class="co">#&gt; ko03010 gene_33/gene_72/gene_275/gene_276/gene_285/gene_286/gene_303/gene_383/gene_460/gene_557</span></span>
<span><span class="co">#&gt; ko01200                                                               gene_89/gene_113/gene_240</span></span>
<span><span class="co">#&gt;         Count</span></span>
<span><span class="co">#&gt; ko00240     3</span></span>
<span><span class="co">#&gt; ko04016     2</span></span>
<span><span class="co">#&gt; ko00052     1</span></span>
<span><span class="co">#&gt; ko04066     2</span></span>
<span><span class="co">#&gt; ko03010    10</span></span>
<span><span class="co">#&gt; ko01200     3</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/kegg_enrich.html">?TOmicsVis::kegg_enrich</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html</a>.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_359-kegg_enrich_bar">3.5.9 kegg_enrich_bar<a class="anchor" aria-label="anchor" href="#id_359-kegg_enrich_bar"></a>
</h4>
<p>KEGG enrichment analysis and bar plot based on KEGG annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_bar analysis function</span></span>
<span><span class="fu"><a href="reference/kegg_enrich_bar.html">kegg_enrich_bar</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for fill is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for fill, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-kegg_enrich_bar-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/kegg_enrich_bar.html">?TOmicsVis::kegg_enrich_bar</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html</a>.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_bar</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_3510-kegg_enrich_dot">3.5.10 kegg_enrich_dot<a class="anchor" aria-label="anchor" href="#id_3510-kegg_enrich_dot"></a>
</h4>
<p>KEGG enrichment analysis and dot plot based on KEGG annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_dot analysis function</span></span>
<span><span class="fu"><a href="reference/kegg_enrich_dot.html">kegg_enrich_dot</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-kegg_enrich_dot-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/kegg_enrich_dot.html">?TOmicsVis::kegg_enrich_dot</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html</a>.</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_dot</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_3511-kegg_enrich_tree">3.5.11 kegg_enrich_tree<a class="anchor" aria-label="anchor" href="#id_3511-kegg_enrich_tree"></a>
</h4>
<p>KEGG enrichment analysis and tree plot based on KEGG annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_tree analysis function</span></span>
<span><span class="fu"><a href="reference/kegg_enrich_tree.html">kegg_enrich_tree</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  hclust_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-kegg_enrich_tree-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/kegg_enrich_tree.html">?TOmicsVis::kegg_enrich_tree</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_tree.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_tree.html</a>.</p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_tree</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_3512-kegg_enrich_net">3.5.12 kegg_enrich_net<a class="anchor" aria-label="anchor" href="#id_3512-kegg_enrich_net"></a>
</h4>
<p>KEGG enrichment analysis and net plot based on KEGG annotation results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_net analysis function</span></span>
<span><span class="fu"><a href="reference/kegg_enrich_net.html">kegg_enrich_net</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  net_layout <span class="op">=</span> <span class="st">"circle"</span>,</span>
<span>  net_circular <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for size is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for size, which will replace the existing scale.</span></span>
<span><span class="co">#&gt; Scale for colour is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for colour, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="reference/figures/README-kegg_enrich_net-1.png"><!-- --></p>
<p>Get help using command <code><a href="reference/kegg_enrich_net.html">?TOmicsVis::kegg_enrich_net</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html</a>.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_net</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_36-tables-operations">3.6 Tables Operations<a class="anchor" aria-label="anchor" href="#id_36-tables-operations"></a>
</h3>
<div class="section level4">
<h4 id="id_361-table_split">3.6.1 table_split<a class="anchor" aria-label="anchor" href="#id_361-table_split"></a>
</h4>
<p>Table split used for splitting a grouped column to multiple columns.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load table_split_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">table_split_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table_split_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;   month day variable value</span></span>
<span><span class="co">#&gt; 1     5   1    ozone    41</span></span>
<span><span class="co">#&gt; 2     5   2    ozone    36</span></span>
<span><span class="co">#&gt; 3     5   3    ozone    12</span></span>
<span><span class="co">#&gt; 4     5   4    ozone    18</span></span>
<span><span class="co">#&gt; 5     5   5    ozone    NA</span></span>
<span><span class="co">#&gt; 6     5   6    ozone    28</span></span>
<span></span>
<span><span class="co"># 2. Run table_split plot function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_split.html">table_split</a></span><span class="op">(</span><span class="va">table_split_data</span>, </span>
<span>                  grouped_var <span class="op">=</span> <span class="st">"variable"</span>, </span>
<span>                  miss_drop <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>                  <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;   month day ozone solar.r temp wind</span></span>
<span><span class="co">#&gt; 1     5   1    41     190   67  7.4</span></span>
<span><span class="co">#&gt; 2     5   2    36     118   72  8.0</span></span>
<span><span class="co">#&gt; 3     5   3    12     149   74 12.6</span></span>
<span><span class="co">#&gt; 4     5   4    18     313   62 11.5</span></span>
<span><span class="co">#&gt; 5     5   5    NA      NA   56 14.3</span></span>
<span><span class="co">#&gt; 6     5   6    28      NA   66 14.9</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_split.html">?TOmicsVis::table_split</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_split.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_split.html</a>.</p>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_split</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_362-table_merge">3.6.2 table_merge<a class="anchor" aria-label="anchor" href="#id_362-table_merge"></a>
</h4>
<p>Table merge used to merge multiple variables to on variable.</p>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">table_merge_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table_merge_data</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Ozone Solar.R Wind Temp Month Day</span></span>
<span><span class="co">#&gt; 1    41     190  7.4   67     5   1</span></span>
<span><span class="co">#&gt; 2    36     118  8.0   72     5   2</span></span>
<span><span class="co">#&gt; 3    12     149 12.6   74     5   3</span></span>
<span><span class="co">#&gt; 4    18     313 11.5   62     5   4</span></span>
<span><span class="co">#&gt; 5    NA      NA 14.3   56     5   5</span></span>
<span><span class="co">#&gt; 6    28      NA 14.9   66     5   6</span></span>
<span></span>
<span><span class="co"># 2. Run function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_merge.html">table_merge</a></span><span class="op">(</span></span>
<span>  <span class="va">table_merge_data</span>,</span>
<span>  merge_vars <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Ozone"</span>, <span class="st">"Solar.R"</span>, <span class="st">"Wind"</span>, <span class="st">"Temp"</span><span class="op">)</span>,</span>
<span>  new_var <span class="op">=</span> <span class="st">"Variable"</span>,</span>
<span>  new_value <span class="op">=</span> <span class="st">"Value"</span>,</span>
<span>  na_remove <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;   Month Day Variable Value</span></span>
<span><span class="co">#&gt; 1     5   1    Ozone    41</span></span>
<span><span class="co">#&gt; 2     5   2    Ozone    36</span></span>
<span><span class="co">#&gt; 3     5   3    Ozone    12</span></span>
<span><span class="co">#&gt; 4     5   4    Ozone    18</span></span>
<span><span class="co">#&gt; 5     5   5    Ozone    NA</span></span>
<span><span class="co">#&gt; 6     5   6    Ozone    28</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_merge.html">?TOmicsVis::table_merge</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_merge.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_merge.html</a>.</p>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_merge</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_363-table_filter">3.6.3 table_filter<a class="anchor" aria-label="anchor" href="#id_363-table_filter"></a>
</h4>
<p>Table filter used to filter row by column condition.</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">table_filter_data</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table_filter_data</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 14</span></span>
<span><span class="co">#&gt;   name         height  mass hair_…¹ skin_…² eye_c…³ birth…⁴ sex   gender homew…⁵</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;         &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; 1 Luke Skywal…    172    77 blond   fair    blue       19   male  mascu… Tatooi…</span></span>
<span><span class="co">#&gt; 2 C-3PO           167    75 &lt;NA&gt;    gold    yellow    112   none  mascu… Tatooi…</span></span>
<span><span class="co">#&gt; 3 R2-D2            96    32 &lt;NA&gt;    white,… red        33   none  mascu… Naboo  </span></span>
<span><span class="co">#&gt; 4 Darth Vader     202   136 none    white   yellow     41.9 male  mascu… Tatooi…</span></span>
<span><span class="co">#&gt; 5 Leia Organa     150    49 brown   light   brown      19   fema… femin… Aldera…</span></span>
<span><span class="co">#&gt; 6 Owen Lars       178   120 brown,… light   blue       52   male  mascu… Tatooi…</span></span>
<span><span class="co">#&gt; # … with 4 more variables: species &lt;chr&gt;, films &lt;list&gt;, vehicles &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   starships &lt;list&gt;, and abbreviated variable names ¹​hair_color, ²​skin_color,</span></span>
<span><span class="co">#&gt; #   ³​eye_color, ⁴​birth_year, ⁵​homeworld</span></span>
<span></span>
<span><span class="co"># 2. Run function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_filter.html">table_filter</a></span><span class="op">(</span><span class="va">table_filter_data</span>, </span>
<span>                    <span class="va">height</span> <span class="op">&gt;</span> <span class="fl">100</span> <span class="op">&amp;</span> <span class="va">eye_color</span> <span class="op">==</span> <span class="st">"black"</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 14</span></span>
<span><span class="co">#&gt;   name      height  mass hair_color skin_…¹ eye_c…² birth…³ sex   gender homew…⁴</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;      &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;      &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; 1 Greedo       173    74 &lt;NA&gt;       green   black        44 male  mascu… Rodia  </span></span>
<span><span class="co">#&gt; 2 Nien Nunb    160    68 none       grey    black        NA male  mascu… Sullust</span></span>
<span><span class="co">#&gt; 3 Gasgano      122    NA none       white,… black        NA male  mascu… Troiken</span></span>
<span><span class="co">#&gt; 4 Kit Fisto    196    87 none       green   black        NA male  mascu… Glee A…</span></span>
<span><span class="co">#&gt; 5 Plo Koon     188    80 none       orange  black        22 male  mascu… Dorin  </span></span>
<span><span class="co">#&gt; 6 Lama Su      229    88 none       grey    black        NA male  mascu… Kamino </span></span>
<span><span class="co">#&gt; # … with 4 more variables: species &lt;chr&gt;, films &lt;list&gt;, vehicles &lt;list&gt;,</span></span>
<span><span class="co">#&gt; #   starships &lt;list&gt;, and abbreviated variable names ¹​skin_color, ²​eye_color,</span></span>
<span><span class="co">#&gt; #   ³​birth_year, ⁴​homeworld</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_filter.html">?TOmicsVis::table_filter</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_filter.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_filter.html</a>.</p>
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_filter</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="id_364-table_cross">3.6.4 table_cross<a class="anchor" aria-label="anchor" href="#id_364-table_cross"></a>
</h4>
<p>Table cross used to cross search and merge results in two tables.</p>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">table_cross_data1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table_cross_data1</span><span class="op">)</span></span>
<span><span class="co">#&gt;      geneID root_exp leave_exp</span></span>
<span><span class="co">#&gt; 1 Unigene01  16.4798    3.3122</span></span>
<span><span class="co">#&gt; 2 Unigene02  44.5027   24.1932</span></span>
<span><span class="co">#&gt; 3 Unigene03  86.9566   43.0663</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">table_cross_data2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">table_cross_data2</span><span class="op">)</span></span>
<span><span class="co">#&gt;      geneID  KO_id                     ko_definition</span></span>
<span><span class="co">#&gt; 1 Unigene02 K10592 E3 ubiquitin-protein ligase HUWE1</span></span>
<span><span class="co">#&gt; 2 Unigene03 K10592    NADH dehydrogenase I subunit 4</span></span>
<span><span class="co">#&gt; 3 Unigene04 K05579         dehydrogenase I subunit 7</span></span>
<span></span>
<span><span class="co"># 2. Run function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/table_cross.html">table_cross</a></span><span class="op">(</span></span>
<span>  <span class="va">table_cross_data1</span>,</span>
<span>  <span class="va">table_cross_data2</span>,</span>
<span>  inter_var <span class="op">=</span> <span class="st">"geneID"</span>,</span>
<span>  left_index <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  right_index <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;      geneID root_exp leave_exp  KO_id                     ko_definition</span></span>
<span><span class="co">#&gt; 1 Unigene01  16.4798    3.3122   &lt;NA&gt;                              &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 2 Unigene02  44.5027   24.1932 K10592 E3 ubiquitin-protein ligase HUWE1</span></span>
<span><span class="co">#&gt; 3 Unigene03  86.9566   43.0663 K10592    NADH dehydrogenase I subunit 4</span></span></code></pre></div>
<p>Get help using command <code><a href="reference/table_cross.html">?TOmicsVis::table_cross</a></code> or reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/table_cross.html" class="external-link uri">https://benben-miao.github.io/TOmicsVis/reference/table_cross.html</a>.</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::table_cross</span></span></code></pre></div>
</div>
</div>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=TOmicsVis" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/benben-miao/TOmicsVis/">Browse source code</a></li>
<li><a href="https://github.com/benben-miao/TOmicsVis/issues/">Report a bug</a></li>
<li><a href="https://github.com/OmicsSuite/" class="external-link">OmicsSuite</a></li>
<li><a href="https://github.com/OmicsSuite/OmicsSuite.github.io" class="external-link">BioSciTools website</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing TOmicsVis</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Benben Miao <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7307-1331" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Wei Dong <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-2201-7164" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Benben Miao, Wei Dong.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
