<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="TOmicsVis">
<title>GOandKEGG â€¢ TOmicsVis</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/JetBrains_Mono-0.4.2/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GOandKEGG">
<meta property="og:description" content="TOmicsVis">
<meta property="og:image" content="https://github.com/benben-miao/TOmicsVis/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TOmicsVis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.2.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">
    <span class="fa fa-file-code"></span>
     
    Reference
  </a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--articles">
    <span class="fa fa-book"></span>
     
    Articles
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--articles">
    <h6 class="dropdown-header" data-toc-skip>Get Start</h6>
    <a class="external-link dropdown-item" href="https://mp.weixin.qq.com/s/g8sRcK_ExlsOFniMWEJnVQ">LongArticle</a>
    <h6 class="dropdown-header" data-toc-skip>Tutorial</h6>
    <a class="dropdown-item" href="../articles/Introduction.html">Introduction</a>
    <a class="dropdown-item" href="../articles/SamplesStatistics.html">SamplesStatistics</a>
    <a class="dropdown-item" href="../articles/TraitsAnalysis.html">TraitsAnalysis</a>
    <a class="dropdown-item" href="../articles/DifferentialExpression.html">DifferentialExpression</a>
    <a class="dropdown-item" href="../articles/AdvancedAnalysis.html">AdvancedAnalysis</a>
    <a class="dropdown-item" href="../articles/GOandKEGG.html">GOandKEGG</a>
    <a class="dropdown-item" href="../articles/TablesOperations.html">TablesOperations</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Blog</h6>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">
    <span class="fa fa-code-branch"></span>
     
    Versions
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--courses">
    <span class="fa fa-film"></span>
     
    Courses
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--courses">
    <h6 class="dropdown-header" data-toc-skip>Platforms</h6>
    <a class="external-link dropdown-item" href="https://space.bilibili.com/34105515/channel/series">Bilibili</a>
    <a class="external-link dropdown-item" href="https://mp.weixin.qq.com/s/mVXJmHPAnC9J1-zMj7eG7g">WeChat</a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/OmicsSuite/">
    <span class="fa fa-desktop"></span>
     
    OmicsSuite
  </a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="nav-link" href="https://github.com/benben-miao/TOmicsVis/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>GOandKEGG</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/benben-miao/TOmicsVis/blob/HEAD/vignettes/GOandKEGG.Rmd"><code>vignettes/GOandKEGG.Rmd</code></a></small>
      <div class="d-none name"><code>GOandKEGG.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="go-and-kegg-enrichment">5. GO and KEGG Enrichment<a class="anchor" aria-label="anchor" href="#go-and-kegg-enrichment"></a>
</h2>
<div class="section level3">
<h3 id="chord_plot">5.1 chord_plot<a class="anchor" aria-label="anchor" href="#chord_plot"></a>
</h3>
<p>Chord plot for visualizing the relationships of pathways and
genes.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://benben-miao.github.io/TOmicsVis/" class="external-link">TOmicsVis</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Loading required package: e1071</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span></span>
<span><span class="co">#&gt;   method from   </span></span>
<span><span class="co">#&gt;   +.gg   ggplot2</span></span>
<span><span class="co">#&gt; Loading required package: Biobase</span></span>
<span><span class="co">#&gt; Loading required package: BiocGenerics</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'BiocGenerics'</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:stats':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     IQR, mad, sd, var, xtabs</span></span>
<span><span class="co">#&gt; The following objects are masked from 'package:base':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">#&gt;     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">#&gt;     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">#&gt;     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">#&gt;     Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,</span></span>
<span><span class="co">#&gt;     table, tapply, union, unique, unsplit, which.max, which.min</span></span>
<span><span class="co">#&gt; Welcome to Bioconductor</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     Vignettes contain introductory material; view with</span></span>
<span><span class="co">#&gt;     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">#&gt;     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'DynDoc'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:BiocGenerics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     path</span></span>
<span></span>
<span><span class="co"># 1. Load chord_data example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">chord_data</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 2. Run chord_plot plot function</span></span>
<span><span class="fu"><a href="../reference/chord_plot.html">chord_plot</a></span><span class="op">(</span></span>
<span>  <span class="va">chord_data</span>,</span>
<span>  multi_colors <span class="op">=</span> <span class="st">"RainbowColors"</span>,</span>
<span>  color_alpha <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  link_visible <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  link_dir <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>,</span>
<span>  link_type <span class="op">=</span> <span class="st">"diffHeight"</span>,</span>
<span>  sector_scale <span class="op">=</span> <span class="st">"Origin"</span>,</span>
<span>  width_circle <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  dist_name <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  label_dir <span class="op">=</span> <span class="st">"Vertical"</span>,</span>
<span>  dist_label <span class="op">=</span> <span class="fl">0.3</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/chord_plot-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt;      rn         cn value1 value2 o1 o2 x1  x2       col</span></span>
<span><span class="co">#&gt; 1    S1 Samples.E1      4      4  5 20 61 161 #FF00007F</span></span>
<span><span class="co">#&gt; 2    S2 Samples.E1      7      7  5 19 45 157 #FF3D007F</span></span>
<span><span class="co">#&gt; 3    S3 Samples.E1      9      9  5 18 55 150 #FF7A007F</span></span>
<span><span class="co">#&gt; 4    S4 Samples.E1     14     14  5 17 55 141 #FFB8007F</span></span>
<span><span class="co">#&gt; 5    S5 Samples.E1      1      1  5 16 22 127 #FFF5007F</span></span>
<span><span class="co">#&gt; 6    S6 Samples.E1     10     10  5 15 59 126 #CCFF007F</span></span>
<span><span class="co">#&gt; 7    S7 Samples.E1      3      3  5 14 31 116 #8FFF007F</span></span>
<span><span class="co">#&gt; 8    S8 Samples.E1     15     15  5 13 52 113 #52FF007F</span></span>
<span><span class="co">#&gt; 9    S9 Samples.E1      5      5  5 12 37  98 #14FF007F</span></span>
<span><span class="co">#&gt; 10  S10 Samples.E1      7      7  5 11 27  93 #00FF297F</span></span>
<span><span class="co">#&gt; 11  S11 Samples.E1      6      6  5 10 35  86 #00FF667F</span></span>
<span><span class="co">#&gt; 12  S12 Samples.E1     11     11  5  9 49  80 #00FFA37F</span></span>
<span><span class="co">#&gt; 13  S13 Samples.E1     18     18  5  8 69  69 #00FFE07F</span></span>
<span><span class="co">#&gt; 14  S14 Samples.E1      1      1  5  7 22  51 #00E0FF7F</span></span>
<span><span class="co">#&gt; 15  S15 Samples.E1      5      5  5  6 58  50 #00A3FF7F</span></span>
<span><span class="co">#&gt; 16  S16 Samples.E1     12     12  5  5 48  45 #0066FF7F</span></span>
<span><span class="co">#&gt; 17  S17 Samples.E1      9      9  5  4 23  33 #0029FF7F</span></span>
<span><span class="co">#&gt; 18  S18 Samples.E1      5      5  5  3 50  24 #1400FF7F</span></span>
<span><span class="co">#&gt; 19  S19 Samples.E1      4      4  5  2 40  19 #5200FF7F</span></span>
<span><span class="co">#&gt; 20  S20 Samples.E1     15     15  5  1 58  15 #8F00FF7F</span></span>
<span><span class="co">#&gt; 21   S1         E2     16     16  4 20 57 166 #FF00007F</span></span>
<span><span class="co">#&gt; 22   S2         E2     11     11  4 19 38 150 #FF3D007F</span></span>
<span><span class="co">#&gt; 23   S3         E2      2      2  4 18 46 139 #FF7A007F</span></span>
<span><span class="co">#&gt; 24   S4         E2      9      9  4 17 41 137 #FFB8007F</span></span>
<span><span class="co">#&gt; 25   S5         E2      1      1  4 16 21 128 #FFF5007F</span></span>
<span><span class="co">#&gt; 26   S6         E2     18     18  4 15 49 127 #CCFF007F</span></span>
<span><span class="co">#&gt; 27   S7         E2      8      8  4 14 28 109 #8FFF007F</span></span>
<span><span class="co">#&gt; 28   S8         E2      4      4  4 13 37 101 #52FF007F</span></span>
<span><span class="co">#&gt; 29   S9         E2      4      4  4 12 32  97 #14FF007F</span></span>
<span><span class="co">#&gt; 30  S10         E2      9      9  4 11 20  93 #00FF297F</span></span>
<span><span class="co">#&gt; 31  S11         E2      7      7  4 10 29  84 #00FF667F</span></span>
<span><span class="co">#&gt; 32  S12         E2      2      2  4  9 38  77 #00FFA37F</span></span>
<span><span class="co">#&gt; 33  S13         E2     13     13  4  8 51  75 #00FFE07F</span></span>
<span><span class="co">#&gt; 34  S14         E2      2      2  4  7 21  62 #00E0FF7F</span></span>
<span><span class="co">#&gt; 35  S15         E2     13     13  4  6 53  60 #00A3FF7F</span></span>
<span><span class="co">#&gt; 36  S16         E2      7      7  4  5 36  47 #0066FF7F</span></span>
<span><span class="co">#&gt; 37  S17         E2      2      2  4  4 14  40 #0029FF7F</span></span>
<span><span class="co">#&gt; 38  S18         E2     13     13  4  3 45  38 #1400FF7F</span></span>
<span><span class="co">#&gt; 39  S19         E2     12     12  4  2 36  25 #5200FF7F</span></span>
<span><span class="co">#&gt; 40  S20         E2     13     13  4  1 43  13 #8F00FF7F</span></span>
<span><span class="co">#&gt; 41   S1         E3     12     12  3 20 41 150 #FF00007F</span></span>
<span><span class="co">#&gt; 42   S2         E3      2      2  3 19 27 138 #FF3D007F</span></span>
<span><span class="co">#&gt; 43   S3         E3     17     17  3 18 44 136 #FF7A007F</span></span>
<span><span class="co">#&gt; 44   S4         E3     12     12  3 17 32 119 #FFB8007F</span></span>
<span><span class="co">#&gt; 45   S5         E3      7      7  3 16 20 107 #FFF5007F</span></span>
<span><span class="co">#&gt; 46   S6         E3      9      9  3 15 31 100 #CCFF007F</span></span>
<span><span class="co">#&gt; 47   S7         E3      4      4  3 14 20  91 #8FFF007F</span></span>
<span><span class="co">#&gt; 48   S8         E3      9      9  3 13 33  87 #52FF007F</span></span>
<span><span class="co">#&gt; 49   S9         E3      7      7  3 12 28  78 #14FF007F</span></span>
<span><span class="co">#&gt; 50  S10         E3      1      1  3 11 11  71 #00FF297F</span></span>
<span><span class="co">#&gt; 51  S11         E3      5      5  3 10 22  70 #00FF667F</span></span>
<span><span class="co">#&gt; 52  S12         E3      2      2  3  9 36  65 #00FFA37F</span></span>
<span><span class="co">#&gt; 53  S13         E3      8      8  3  8 38  63 #00FFE07F</span></span>
<span><span class="co">#&gt; 54  S14         E3      2      2  3  7 19  55 #00E0FF7F</span></span>
<span><span class="co">#&gt; 55  S15         E3      6      6  3  6 40  53 #00A3FF7F</span></span>
<span><span class="co">#&gt; 56  S16         E3     15     15  3  5 29  47 #0066FF7F</span></span>
<span><span class="co">#&gt; 57  S17         E3      1      1  3  4 12  32 #0029FF7F</span></span>
<span><span class="co">#&gt; 58  S18         E3     13     13  3  3 32  31 #1400FF7F</span></span>
<span><span class="co">#&gt; 59  S19         E3      4      4  3  2 24  18 #5200FF7F</span></span>
<span><span class="co">#&gt; 60  S20         E3     14     14  3  1 30  14 #8F00FF7F</span></span>
<span><span class="co">#&gt; 61   S1         E4     18     18  2 20 29 227 #FF00007F</span></span>
<span><span class="co">#&gt; 62   S2         E4     15     15  2 19 25 209 #FF3D007F</span></span>
<span><span class="co">#&gt; 63   S3         E4     16     16  2 18 27 194 #FF7A007F</span></span>
<span><span class="co">#&gt; 64   S4         E4      3      3  2 17 20 178 #FFB8007F</span></span>
<span><span class="co">#&gt; 65   S5         E4      1      1  2 16 13 175 #FFF5007F</span></span>
<span><span class="co">#&gt; 66   S6         E4     13     13  2 15 22 174 #CCFF007F</span></span>
<span><span class="co">#&gt; 67   S7         E4     15     15  2 14 16 161 #8FFF007F</span></span>
<span><span class="co">#&gt; 68   S8         E4     13     13  2 13 24 146 #52FF007F</span></span>
<span><span class="co">#&gt; 69   S9         E4      3      3  2 12 21 133 #14FF007F</span></span>
<span><span class="co">#&gt; 70  S10         E4      4      4  2 11 10 130 #00FF297F</span></span>
<span><span class="co">#&gt; 71  S11         E4      8      8  2 10 17 126 #00FF667F</span></span>
<span><span class="co">#&gt; 72  S12         E4     16     16  2  9 34 118 #00FFA37F</span></span>
<span><span class="co">#&gt; 73  S13         E4     14     14  2  8 30 102 #00FFE07F</span></span>
<span><span class="co">#&gt; 74  S14         E4     14     14  2  7 17  88 #00E0FF7F</span></span>
<span><span class="co">#&gt; 75  S15         E4     16     16  2  6 34  74 #00A3FF7F</span></span>
<span><span class="co">#&gt; 76  S16         E4     12     12  2  5 14  58 #0066FF7F</span></span>
<span><span class="co">#&gt; 77  S17         E4     10     10  2  4 11  46 #0029FF7F</span></span>
<span><span class="co">#&gt; 78  S18         E4     11     11  2  3 19  36 #1400FF7F</span></span>
<span><span class="co">#&gt; 79  S19         E4     14     14  2  2 20  25 #5200FF7F</span></span>
<span><span class="co">#&gt; 80  S20         E4     11     11  2  1 16  11 #8F00FF7F</span></span>
<span><span class="co">#&gt; 81   S1         E5     11     11  1 20 11 192 #FF00007F</span></span>
<span><span class="co">#&gt; 82   S2         E5     10     10  1 19 10 181 #FF3D007F</span></span>
<span><span class="co">#&gt; 83   S3         E5     11     11  1 18 11 171 #FF7A007F</span></span>
<span><span class="co">#&gt; 84   S4         E5     17     17  1 17 17 160 #FFB8007F</span></span>
<span><span class="co">#&gt; 85   S5         E5     12     12  1 16 12 143 #FFF5007F</span></span>
<span><span class="co">#&gt; 86   S6         E5      9      9  1 15  9 131 #CCFF007F</span></span>
<span><span class="co">#&gt; 87   S7         E5      1      1  1 14  1 122 #8FFF007F</span></span>
<span><span class="co">#&gt; 88   S8         E5     11     11  1 13 11 121 #52FF007F</span></span>
<span><span class="co">#&gt; 89   S9         E5     18     18  1 12 18 110 #14FF007F</span></span>
<span><span class="co">#&gt; 90  S10         E5      6      6  1 11  6  92 #00FF297F</span></span>
<span><span class="co">#&gt; 91  S11         E5      9      9  1 10  9  86 #00FF667F</span></span>
<span><span class="co">#&gt; 92  S12         E5     18     18  1  9 18  77 #00FFA37F</span></span>
<span><span class="co">#&gt; 93  S13         E5     16     16  1  8 16  59 #00FFE07F</span></span>
<span><span class="co">#&gt; 94  S14         E5      3      3  1  7  3  43 #00E0FF7F</span></span>
<span><span class="co">#&gt; 95  S15         E5     18     18  1  6 18  40 #00A3FF7F</span></span>
<span><span class="co">#&gt; 96  S16         E5      2      2  1  5  2  22 #0066FF7F</span></span>
<span><span class="co">#&gt; 97  S17         E5      1      1  1  4  1  20 #0029FF7F</span></span>
<span><span class="co">#&gt; 98  S18         E5      8      8  1  3  8  19 #1400FF7F</span></span>
<span><span class="co">#&gt; 99  S19         E5      6      6  1  2  6  11 #5200FF7F</span></span>
<span><span class="co">#&gt; 100 S20         E5      5      5  1  1  5   5 #8F00FF7F</span></span></code></pre>
<p>Get help using command <code><a href="../reference/chord_plot.html">?TOmicsVis::chord_plot</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/chord_plot.html</a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::chord_plot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="go_enrich">5.2 go_enrich<a class="anchor" aria-label="anchor" href="#go_enrich"></a>
</h3>
<p>GO enrichment analysis based on GO annotation results (None/Exist
Reference Genome).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co">#&gt;       id</span></span>
<span><span class="co">#&gt; 1 gene_1</span></span>
<span><span class="co">#&gt; 2 gene_2</span></span>
<span><span class="co">#&gt; 3 gene_3</span></span>
<span><span class="co">#&gt; 4 gene_4</span></span>
<span><span class="co">#&gt; 5 gene_5</span></span>
<span><span class="co">#&gt; 6 gene_6</span></span>
<span><span class="co">#&gt;                                                                                                                                                                               biological_process</span></span>
<span><span class="co">#&gt; 1                                                                                                                                             GO:0015986(ATP synthesis coupled proton transport)</span></span>
<span><span class="co">#&gt; 2 GO:0071236(cellular response to antibiotic);GO:0071493(cellular response to UV-B);GO:0008630(intrinsic apoptotic signaling pathway in response to DNA damage);GO:0010043(response to zinc ion)</span></span>
<span><span class="co">#&gt; 3                                                                                                                                                               GO:0009058(biosynthetic process)</span></span>
<span><span class="co">#&gt; 4                                                                                                                                                                 GO:0006096(glycolytic process)</span></span>
<span><span class="co">#&gt; 5                                                                                                                                                                                           &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6                                                                                                                                                                        GO:0006412(translation)</span></span>
<span><span class="co">#&gt;                                                                           cellular_component</span></span>
<span><span class="co">#&gt; 1 "GO:0000276(mitochondrial proton-transporting ATP synthase complex, coupling factor F(o))"</span></span>
<span><span class="co">#&gt; 2                                                                      GO:0005737(cytoplasm)</span></span>
<span><span class="co">#&gt; 3                                                                                       &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 4                                              GO:0000015(phosphopyruvate hydratase complex)</span></span>
<span><span class="co">#&gt; 5                                                                                       &lt;NA&gt;</span></span>
<span><span class="co">#&gt; 6                                              GO:0022625(cytosolic large ribosomal subunit)</span></span>
<span><span class="co">#&gt;                                                                                                                                     molecular_function</span></span>
<span><span class="co">#&gt; 1                    GO:0047624(adenosine-tetraphosphatase activity);GO:0015078(proton transmembrane transporter activity);GO:0016887(ATPase activity)</span></span>
<span><span class="co">#&gt; 2 "GO:0046872(metal ion binding);GO:0003680(AT DNA binding);GO:0008301(DNA binding, bending);GO:0042277(peptide binding);GO:0008270(zinc ion binding)"</span></span>
<span><span class="co">#&gt; 3                                                                                                                       GO:0003824(catalytic activity)</span></span>
<span><span class="co">#&gt; 4                                                                     GO:0000287(magnesium ion binding);GO:0004634(phosphopyruvate hydratase activity)</span></span>
<span><span class="co">#&gt; 5                                                                                                               GO:0005319(lipid transporter activity)</span></span>
<span><span class="co">#&gt; 6                                                                               GO:0003723(RNA binding);GO:0003735(structural constituent of ribosome)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co">#&gt;        id log2FC</span></span>
<span><span class="co">#&gt; 1 gene_14  -1.20</span></span>
<span><span class="co">#&gt; 2 gene_15   1.25</span></span>
<span><span class="co">#&gt; 3 gene_16   1.30</span></span>
<span><span class="co">#&gt; 4 gene_17   1.35</span></span>
<span><span class="co">#&gt; 5 gene_18  -1.50</span></span>
<span><span class="co">#&gt; 6 gene_20  -1.55</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich analysis function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/go_enrich.html">go_enrich</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;           ID           ontology</span></span>
<span><span class="co">#&gt; 1 GO:0000027 biological process</span></span>
<span><span class="co">#&gt; 2 GO:0001732 biological process</span></span>
<span><span class="co">#&gt; 3 GO:0001944 biological process</span></span>
<span><span class="co">#&gt; 4 GO:0003723 molecular function</span></span>
<span><span class="co">#&gt; 5 GO:0003735 molecular function</span></span>
<span><span class="co">#&gt; 6 GO:0003743 molecular function</span></span>
<span><span class="co">#&gt;                                               Description GeneRatio BgRatio</span></span>
<span><span class="co">#&gt; 1                        ribosomal large subunit assembly      1/11   1/100</span></span>
<span><span class="co">#&gt; 2 formation of cytoplasmic translation initiation complex      1/11   1/100</span></span>
<span><span class="co">#&gt; 3                                 vasculature development      1/11   1/100</span></span>
<span><span class="co">#&gt; 4                                             RNA binding      1/11  11/100</span></span>
<span><span class="co">#&gt; 5                      structural constituent of ribosome      3/11  26/100</span></span>
<span><span class="co">#&gt; 6                  translation initiation factor activity      1/11   1/100</span></span>
<span><span class="co">#&gt;      pvalue  p.adjust    qvalue                  geneID Count</span></span>
<span><span class="co">#&gt; 1 0.1100000 0.1850000 0.1578947                 gene_23     1</span></span>
<span><span class="co">#&gt; 2 0.1100000 0.1850000 0.1578947                 gene_21     1</span></span>
<span><span class="co">#&gt; 3 0.1100000 0.1850000 0.1578947                 gene_21     1</span></span>
<span><span class="co">#&gt; 4 0.7421470 0.7421470 0.6334113                 gene_23     1</span></span>
<span><span class="co">#&gt; 5 0.5849832 0.6365993 0.5433280 gene_22/gene_23/gene_24     3</span></span>
<span><span class="co">#&gt; 6 0.1100000 0.1850000 0.1578947                 gene_21     1</span></span></code></pre></div>
<p>Get help using command <code><a href="../reference/go_enrich.html">?TOmicsVis::go_enrich</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/go_enrich.html</a>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="go_enrich_stat">5.3 go_enrich_stat<a class="anchor" aria-label="anchor" href="#go_enrich_stat"></a>
</h3>
<p>GO enrichment analysis and stat plot based on GO annotation results
(None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_stat analysis function</span></span>
<span><span class="fu"><a href="../reference/go_enrich_stat.html">go_enrich_stat</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  max_go_item <span class="op">=</span> <span class="fl">15</span>,</span>
<span>  strip_fill <span class="op">=</span> <span class="st">"#CDCDCD"</span>,</span>
<span>  xtext_angle <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  sci_fill_color <span class="op">=</span> <span class="st">"Sci_AAAS"</span>,</span>
<span>  sci_fill_alpha <span class="op">=</span> <span class="fl">0.8</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/go_enrich_stat-1.png" width="700"></p>
<p>Get help using command <code><a href="../reference/go_enrich_stat.html">?TOmicsVis::go_enrich_stat</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_stat.html</a>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_stat</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="go_enrich_bar">5.4 go_enrich_bar<a class="anchor" aria-label="anchor" href="#go_enrich_bar"></a>
</h3>
<p>GO enrichment analysis and bar plot based on GO annotation results
(None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_bar analysis function</span></span>
<span><span class="fu"><a href="../reference/go_enrich_bar.html">go_enrich_bar</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">fill</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/go_enrich_bar-1.png" width="700"></p>
<p>Get help using command <code><a href="../reference/go_enrich_bar.html">?TOmicsVis::go_enrich_bar</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_bar.html</a>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_bar</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="go_enrich_dot">5.5 go_enrich_dot<a class="anchor" aria-label="anchor" href="#go_enrich_dot"></a>
</h3>
<p>GO enrichment analysis and dot plot based on GO annotation results
(None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_dot analysis function</span></span>
<span><span class="fu"><a href="../reference/go_enrich_dot.html">go_enrich_dot</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/go_enrich_dot-1.png" width="700"></p>
<p>Get help using command <code><a href="../reference/go_enrich_dot.html">?TOmicsVis::go_enrich_dot</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_dot.html</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_dot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="go_enrich_tree">5.6 go_enrich_tree<a class="anchor" aria-label="anchor" href="#go_enrich_tree"></a>
</h3>
<p>GO enrichment analysis and tree plot based on GO annotation results
(None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_tree analysis function</span></span>
<span><span class="fu"><a href="../reference/go_enrich_tree.html">go_enrich_tree</a></span><span class="op">(</span></span>
<span>  <span class="va">go_anno</span>,</span>
<span>  <span class="va">go_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  hclust_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/go_enrich_tree-1.png" width="700"></p>
<p>Get help using command <code><a href="../reference/go_enrich_tree.html">?TOmicsVis::go_enrich_tree</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_tree.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_tree.html</a>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_tree</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="go_enrich_net">5.7 go_enrich_net<a class="anchor" aria-label="anchor" href="#go_enrich_net"></a>
</h3>
<p>GO enrichment analysis and net plot based on GO annotation results
(None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_anno</span><span class="op">)</span></span>
<span><span class="co"># head(go_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">go_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(go_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich_net analysis function</span></span>
<span><span class="co"># go_enrich_net(</span></span>
<span><span class="co">#   go_anno,</span></span>
<span><span class="co">#   go_deg_fc,</span></span>
<span><span class="co">#   padjust_method = "fdr",</span></span>
<span><span class="co">#   pvalue_cutoff = 0.5,</span></span>
<span><span class="co">#   qvalue_cutoff = 0.5,</span></span>
<span><span class="co">#   category_num = 20,</span></span>
<span><span class="co">#   net_layout = "circle",</span></span>
<span><span class="co">#   net_circular = TRUE,</span></span>
<span><span class="co">#   low_color = "#ff0000aa",</span></span>
<span><span class="co">#   high_color = "#008888aa"</span></span>
<span><span class="co"># )</span></span></code></pre></div>
<p>Get help using command <code><a href="../reference/go_enrich_net.html">?TOmicsVis::go_enrich_net</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/go_enrich_net.html</a>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::go_enrich_net</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="kegg_enrich">5.8 kegg_enrich<a class="anchor" aria-label="anchor" href="#kegg_enrich"></a>
</h3>
<p>KEGG enrichment analysis based on KEGG annotation results (None/Exist
Reference Genome).</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co">#&gt;       id</span></span>
<span><span class="co">#&gt; 1 gene_1</span></span>
<span><span class="co">#&gt; 2 gene_2</span></span>
<span><span class="co">#&gt; 3 gene_3</span></span>
<span><span class="co">#&gt; 4 gene_4</span></span>
<span><span class="co">#&gt; 5 gene_5</span></span>
<span><span class="co">#&gt; 6 gene_6</span></span>
<span><span class="co">#&gt;                                                                                                                              kegg_pathway</span></span>
<span><span class="co">#&gt; 1 ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 2                                                                                               ko04977(Vitamin digestion and absorption)</span></span>
<span><span class="co">#&gt; 3 ko04810(Regulation of actin cytoskeleton);ko04510(Focal adhesion);ko04151(PI3K-Akt signaling pathway);ko04512(ECM-receptor interaction)</span></span>
<span><span class="co">#&gt; 4                                                                                               ko04977(Vitamin digestion and absorption)</span></span>
<span><span class="co">#&gt; 5                                                                                            ko04610(Complement and coagulation cascades)</span></span>
<span><span class="co">#&gt; 6                                                                                                                       ko04142(Lysosome)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co">#&gt;        id log2FC</span></span>
<span><span class="co">#&gt; 1  gene_1   1.20</span></span>
<span><span class="co">#&gt; 2  gene_2  -1.25</span></span>
<span><span class="co">#&gt; 3  gene_3   1.30</span></span>
<span><span class="co">#&gt; 4  gene_4  -1.35</span></span>
<span><span class="co">#&gt; 5  gene_5   1.40</span></span>
<span><span class="co">#&gt; 6 gene_30   1.45</span></span>
<span></span>
<span><span class="co"># 2. Run go_enrich analysis function</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/kegg_enrich.html">kegg_enrich</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt;                ID                              Description GeneRatio BgRatio</span></span>
<span><span class="co">#&gt; ko04977   ko04977         Vitamin digestion and absorption      5/15   12/93</span></span>
<span><span class="co">#&gt; ko04510   ko04510                           Focal adhesion      2/15    2/93</span></span>
<span><span class="co">#&gt; ko04512   ko04512                 ECM-receptor interaction      2/15    2/93</span></span>
<span><span class="co">#&gt; "ko00260 "ko00260 Glycine, serine and threonine metabolism      4/15   11/93</span></span>
<span><span class="co">#&gt; ko00460   ko00460               Cyanoamino acid metabolism      4/15   11/93</span></span>
<span><span class="co">#&gt; ko00670   ko00670                One carbon pool by folate      4/15   11/93</span></span>
<span><span class="co">#&gt;              pvalue  p.adjust    qvalue                                geneID</span></span>
<span><span class="co">#&gt; ko04977  0.02247971 0.1227209 0.1033439 gene_2/gene_4/gene_33/gene_34/gene_19</span></span>
<span><span class="co">#&gt; ko04510  0.02454418 0.1227209 0.1033439                         gene_1/gene_3</span></span>
<span><span class="co">#&gt; ko04512  0.02454418 0.1227209 0.1033439                         gene_1/gene_3</span></span>
<span><span class="co">#&gt; "ko00260 0.07369837 0.1842459 0.1551545       gene_30/gene_15/gene_17/gene_18</span></span>
<span><span class="co">#&gt; ko00460  0.07369837 0.1842459 0.1551545       gene_30/gene_15/gene_17/gene_18</span></span>
<span><span class="co">#&gt; ko00670  0.07369837 0.1842459 0.1551545       gene_30/gene_15/gene_17/gene_18</span></span>
<span><span class="co">#&gt;          Count</span></span>
<span><span class="co">#&gt; ko04977      5</span></span>
<span><span class="co">#&gt; ko04510      2</span></span>
<span><span class="co">#&gt; ko04512      2</span></span>
<span><span class="co">#&gt; "ko00260     4</span></span>
<span><span class="co">#&gt; ko00460      4</span></span>
<span><span class="co">#&gt; ko00670      4</span></span></code></pre></div>
<p>Get help using command <code><a href="../reference/kegg_enrich.html">?TOmicsVis::kegg_enrich</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich.html</a>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="kegg_enrich_bar">5.9 kegg_enrich_bar<a class="anchor" aria-label="anchor" href="#kegg_enrich_bar"></a>
</h3>
<p>KEGG enrichment analysis and bar plot based on KEGG annotation
results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_bar analysis function</span></span>
<span><span class="fu"><a href="../reference/kegg_enrich_bar.html">kegg_enrich_bar</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">fill</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/kegg_enrich_bar-1.png" width="700"></p>
<p>Get help using command <code><a href="../reference/kegg_enrich_bar.html">?TOmicsVis::kegg_enrich_bar</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_bar.html</a>.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_bar</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="kegg_enrich_dot">5.10 kegg_enrich_dot<a class="anchor" aria-label="anchor" href="#kegg_enrich_dot"></a>
</h3>
<p>KEGG enrichment analysis and dot plot based on KEGG annotation
results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_dot analysis function</span></span>
<span><span class="fu"><a href="../reference/kegg_enrich_dot.html">kegg_enrich_dot</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">12</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/kegg_enrich_dot-1.png" width="700"></p>
<p>Get help using command <code><a href="../reference/kegg_enrich_dot.html">?TOmicsVis::kegg_enrich_dot</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_dot.html</a>.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_dot</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="kegg_enrich_tree">5.11 kegg_enrich_tree<a class="anchor" aria-label="anchor" href="#kegg_enrich_tree"></a>
</h3>
<p>KEGG enrichment analysis and tree plot based on KEGG annotation
results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_tree analysis function</span></span>
<span><span class="fu"><a href="../reference/kegg_enrich_tree.html">kegg_enrich_tree</a></span><span class="op">(</span></span>
<span>  <span class="va">kegg_anno</span>,</span>
<span>  <span class="va">kegg_deg_fc</span>,</span>
<span>  padjust_method <span class="op">=</span> <span class="st">"fdr"</span>,</span>
<span>  pvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  qvalue_cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  sign_by <span class="op">=</span> <span class="st">"p.adjust"</span>,</span>
<span>  category_num <span class="op">=</span> <span class="fl">20</span>,</span>
<span>  font_size <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  low_color <span class="op">=</span> <span class="st">"#ff0000aa"</span>,</span>
<span>  high_color <span class="op">=</span> <span class="st">"#008888aa"</span>,</span>
<span>  hclust_method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>  ggTheme <span class="op">=</span> <span class="st">"theme_light"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Scale for <span style="color: #00BB00;">colour</span> is already present.</span></span>
<span><span class="co">#&gt; Adding another scale for <span style="color: #00BB00;">colour</span>, which will replace the existing scale.</span></span></code></pre></div>
<p><img src="GOandKEGG_files/figure-html/kegg_enrich_tree-1.png" width="700"></p>
<p>Get help using command <code><a href="../reference/kegg_enrich_tree.html">?TOmicsVis::kegg_enrich_tree</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_tree.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_tree.html</a>.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_tree</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="kegg_enrich_net">5.12 kegg_enrich_net<a class="anchor" aria-label="anchor" href="#kegg_enrich_net"></a>
</h3>
<p>KEGG enrichment analysis and net plot based on KEGG annotation
results (None/Exist Reference Genome).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># 1. Load example datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_anno</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_anno)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">kegg_deg_fc</span><span class="op">)</span></span>
<span><span class="co"># head(kegg_deg_fc)</span></span>
<span></span>
<span><span class="co"># 2. Run kegg_enrich_net analysis function</span></span>
<span><span class="co"># kegg_enrich_net(</span></span>
<span><span class="co">#   kegg_anno,</span></span>
<span><span class="co">#   kegg_deg_fc,</span></span>
<span><span class="co">#   padjust_method = "fdr",</span></span>
<span><span class="co">#   pvalue_cutoff = 1,</span></span>
<span><span class="co">#   qvalue_cutoff = 1,</span></span>
<span><span class="co">#   category_num = 20,</span></span>
<span><span class="co">#   net_layout = "circle",</span></span>
<span><span class="co">#   net_circular = TRUE,</span></span>
<span><span class="co">#   low_color = "#ff0000aa",</span></span>
<span><span class="co">#   high_color = "#008888aa"</span></span>
<span><span class="co"># )</span></span></code></pre></div>
<p>Get help using command <code><a href="../reference/kegg_enrich_net.html">?TOmicsVis::kegg_enrich_net</a></code> or
reference page <a href="https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html" class="external-link">https://benben-miao.github.io/TOmicsVis/reference/kegg_enrich_net.html</a>.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get help with command in R console.</span></span>
<span><span class="co"># ?TOmicsVis::kegg_enrich_net</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Benben Miao, Wei Dong.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
