<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Heatmap cluster plot for visualizing clustered gene expression data."><title>Heatmap cluster plot for visualizing clustered gene expression data. — heatmap_cluster • TOmicsVis</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/JetBrains_Mono-0.4.2/font.css" rel="stylesheet"><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Heatmap cluster plot for visualizing clustered gene expression data. — heatmap_cluster"><meta property="og:description" content="Heatmap cluster plot for visualizing clustered gene expression data."><meta property="og:image" content="https://github.com/benben-miao/TOmicsVis/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TOmicsVis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.2.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">
    <span class="fa fa-file-code"></span>
     
    Reference
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--articles">
    <span class="fa fa-book"></span>
     
    Articles
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--articles">
    <h6 class="dropdown-header" data-toc-skip>Tutorial</h6>
    <a class="dropdown-item" href="../articles/Introduction.html">Introduction</a>
    <a class="dropdown-item" href="../articles/SamplesStatistics.html">SamplesStatistics</a>
    <a class="dropdown-item" href="../articles/TraitsAnalysis.html">TraitsAnalysis</a>
    <a class="dropdown-item" href="../articles/DifferentialExpression.html">DifferentialExpression</a>
    <a class="dropdown-item" href="../articles/AdvancedAnalysis.html">AdvancedAnalysis</a>
    <a class="dropdown-item" href="../articles/GOandKEGG.html">GOandKEGG</a>
    <a class="dropdown-item" href="../articles/TablesOperations.html">TablesOperations</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Blog</h6>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">
    <span class="fa fa-code-branch"></span>
     
    Versions
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/OmicsSuite/">
    <span class="fa fa-desktop"></span>
     
    OmicsSuite
  </a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="nav-link" href="https://github.com/benben-miao/TOmicsVis/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Heatmap cluster plot for visualizing clustered gene expression data.</h1>
      <small class="dont-index">Source: <a href="https://github.com/benben-miao/TOmicsVis/blob/HEAD/R/heatmap_cluster.R"><code>R/heatmap_cluster.R</code></a></small>
      <div class="d-none name"><code>heatmap_cluster.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Heatmap cluster plot for visualizing clustered gene expression data.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">heatmap_cluster</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  dist_method <span class="op">=</span> <span class="st">"euclidean"</span>,</span>
<span>  hc_method <span class="op">=</span> <span class="st">"average"</span>,</span>
<span>  k_num <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  palette <span class="op">=</span> <span class="st">"Spectral"</span>,</span>
<span>  cluster_pal <span class="op">=</span> <span class="st">"Set1"</span>,</span>
<span>  gaps_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  angle_col <span class="op">=</span> <span class="fl">45</span>,</span>
<span>  label_size <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  base_size <span class="op">=</span> <span class="fl">12</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>data</dt>
<dd><p>Dataframe: gene expression dataframe with cols (samples) and rows (genes).</p></dd>


<dt>dist_method</dt>
<dd><p>Character: distance measure method. Default: "euclidean", options: "euclidean", "maximum", "manhattan", "canberra", "binary" or "minkowski".</p></dd>


<dt>hc_method</dt>
<dd><p>Character: hierarchical clustering method. Default: "average", options: "ward.D", "ward.D2", "single", "complete","average" (= UPGMA), "mcquitty" (= WPGMA), "median" (= WPGMC) or "centroid" (= UPGMC).</p></dd>


<dt>k_num</dt>
<dd><p>Numeric: the number of groups for cutting the tree. Default: 5.</p></dd>


<dt>palette</dt>
<dd><p>Character: color palette used in heatmap. Default: "Spectral", options: 'Spectral', 'BrBG', 'PiYG', 'PRGn', 'PuOr', 'RdBu', 'RdGy', 'RdYlBu', 'RdYlGn'.</p></dd>


<dt>cluster_pal</dt>
<dd><p>Character: color palette used for the cluster. Default: "Set1", options: 'Set1', 'Set2', 'Set3', 'Accent', 'Dark2', 'Paired', 'Pastel1', 'Pastel2'.</p></dd>


<dt>gaps_col</dt>
<dd><p>Character: vector of columns indices that show where to put gaps into heatmap. Default: "NULL".</p></dd>


<dt>angle_col</dt>
<dd><p>Numeric: angle of the column labels. Default: 45.</p></dd>


<dt>label_size</dt>
<dd><p>Numeric: fontsize for the plot. Default: 10, min: 0.</p></dd>


<dt>base_size</dt>
<dd><p>Numeric: base font size. Default: 12, min: 0.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Plot: Heatmap cluster plot for visualizing clustered gene expression data.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>wei dong</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># 1. Library TOmicsVis package</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://benben-miao.github.io/TOmicsVis/" class="external-link">TOmicsVis</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># 2. Use example dataset gene_exp</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_exp</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_exp</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              M1       M2       M3       M4       M5       M6        M7       M8</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RGL4   8.454808 8.019389 8.990836 9.718631 7.908075 4.147051  4.985084 4.576711</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MPP7   8.690520 8.630346 7.080873 9.838476 8.271824 5.179200  5.200868 3.266993</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> UGCG   8.648366 8.600555 9.431046 7.923021 8.309214 4.902510  5.750804 4.492856</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CYSTM1 8.628884 9.238677 8.487243 8.958537 7.357109 4.541605  6.370533 4.246651</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANXA2  4.983769 6.748022 6.220791 4.719403 3.284346 8.089850 10.637472 7.214912</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENDOD1 5.551640 5.406465 4.663785 3.550765 4.103507 8.393991  9.538503 9.069923</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              M9      M10</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> RGL4   4.930349 4.293700</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MPP7   5.565226 4.300309</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> UGCG   4.659987 3.306275</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CYSTM1 4.745769 3.449627</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ANXA2  9.002710 5.123359</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ENDOD1 8.639664 7.106392</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># 3. Default parameters</span></span></span>
<span class="r-in"><span><span class="fu">heatmap_cluster</span><span class="op">(</span><span class="va">gene_exp</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="heatmap_cluster-1.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using Cluster, gene as id variables</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The `fun.y` argument of `stat_summary()` is deprecated as of ggplot2 3.3.0.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Please use the `fun` argument instead.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">TOmicsVis</span> package.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>   Please report the issue at</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>   <span style="color: #0000BB; font-style: italic;">&lt;</span><a href="https://github.com/benben-miao/TOmicsVis/issues/"><span style="color: #0000BB; font-style: italic;">https://github.com/benben-miao/TOmicsVis/issues/</span></a><span style="color: #0000BB; font-style: italic;">&gt;</span>.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Graphs cannot be horizontally aligned unless the axis parameter is set. Placing graphs unaligned.</span>
<span class="r-plt img"><img src="heatmap_cluster-2.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="heatmap_cluster-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># 4. Set palette = "RdBu"</span></span></span>
<span class="r-in"><span><span class="fu">heatmap_cluster</span><span class="op">(</span><span class="va">gene_exp</span>, palette <span class="op">=</span> <span class="st">"RdBu"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="heatmap_cluster-4.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using Cluster, gene as id variables</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Graphs cannot be horizontally aligned unless the axis parameter is set. Placing graphs unaligned.</span>
<span class="r-plt img"><img src="heatmap_cluster-5.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="heatmap_cluster-6.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># 5. Set cluster_pal = "Accent"</span></span></span>
<span class="r-in"><span><span class="fu">heatmap_cluster</span><span class="op">(</span><span class="va">gene_exp</span>, cluster_pal <span class="op">=</span> <span class="st">"Accent"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="heatmap_cluster-7.png" alt="" width="700" height="433"></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using Cluster, gene as id variables</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Graphs cannot be horizontally aligned unless the axis parameter is set. Placing graphs unaligned.</span>
<span class="r-plt img"><img src="heatmap_cluster-8.png" alt="" width="700" height="433"></span>
<span class="r-plt img"><img src="heatmap_cluster-9.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Benben Miao, Wei Dong.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

