<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference • TOmicsVis</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/JetBrains_Mono-0.4.7/font.css" rel="stylesheet"><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="https://github.com/benben-miao/TOmicsVis/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">TOmicsVis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.2.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">
    <span class="fa fa-passport"></span>
     
    Reference
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--articles">
    <span class="fa fa-file-code"></span>
     
    Articles
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--articles">
    <h6 class="dropdown-header" data-toc-skip>Get Start</h6>
    <a class="external-link dropdown-item" href="https://mp.weixin.qq.com/s/g8sRcK_ExlsOFniMWEJnVQ">
      <span class="fa fa-weixin"></span>
       
      LongArticle
    </a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Tutorials</h6>
    <a class="dropdown-item" href="../articles/Tutorials.html">
      <span class="fa fa-file-code"></span>
       
      Tutorials
    </a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Blog</h6>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">
    <span class="fa fa-code-branch"></span>
     
    Versions
  </a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown--courses">
    <span class="fa fa-film"></span>
     
    Courses
  </a>
  <div class="dropdown-menu" aria-labelledby="dropdown--courses">
    <h6 class="dropdown-header" data-toc-skip>Platforms</h6>
    <a class="external-link dropdown-item" href="https://space.bilibili.com/34105515/channel/series">
      <span class="fa fa-youtube"></span>
       
      BiliBili
    </a>
    <a class="external-link dropdown-item" href="https://mp.weixin.qq.com/s/mVXJmHPAnC9J1-zMj7eG7g">
      <span class="fa fa-weixin"></span>
       
      WeChat
    </a>
  </div>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/OmicsSuite/">
    <span class="fa fa-dna"></span>
     
    OmicsSuite
  </a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="nav-link" href="https://github.com/benben-miao/TOmicsVis/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="-samples-statistics">①Samples Statistics<a class="anchor" aria-label="anchor" href="#-samples-statistics"></a></h2>
      
      <p class="section-desc"></p><p>Statistical sample outliers, correlation among samples, distance among samples, etc.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="quantile_plot.html">quantile_plot()</a></code> 
        </dt>
        <dd>Quantile plot for visualizing data distribution.</dd>
      </dl><dl><dt>
          
          <code><a href="box_plot.html">box_plot()</a></code> 
        </dt>
        <dd>Box plot support two levels and multiple groups with P value.</dd>
      </dl><dl><dt>
          
          <code><a href="violin_plot.html">violin_plot()</a></code> 
        </dt>
        <dd>Violin plot support two levels and multiple groups with P value.</dd>
      </dl><dl><dt>
          
          <code><a href="survival_plot.html">survival_plot()</a></code> 
        </dt>
        <dd>Survival plot for analyzing and visualizing survival data.</dd>
      </dl></div><div class="section level2">
      <h2 id="-traits-analysis">②Traits Analysis<a class="anchor" aria-label="anchor" href="#-traits-analysis"></a></h2>
      
      <p class="section-desc"></p><p>Statistical analysis of traits, such as adding R and P values to box plots and violin plots, and more dimensionality reduction analysis.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="corr_heatmap.html">corr_heatmap()</a></code> 
        </dt>
        <dd>Correlation Heatmap for samples/groups based on Pearson algorithm.</dd>
      </dl><dl><dt>
          
          <code><a href="pca_analysis.html">pca_analysis()</a></code> 
        </dt>
        <dd>PCA dimensional reduction analysis for RNA-Seq.</dd>
      </dl><dl><dt>
          
          <code><a href="pca_plot.html">pca_plot()</a></code> 
        </dt>
        <dd>PCA dimensional reduction visualization for RNA-Seq.</dd>
      </dl><dl><dt>
          
          <code><a href="tsne_analysis.html">tsne_analysis()</a></code> 
        </dt>
        <dd>TSNE analysis for analyzing and visualizing TSNE algorithm.</dd>
      </dl><dl><dt>
          
          <code><a href="tsne_plot.html">tsne_plot()</a></code> 
        </dt>
        <dd>TSNE plot for analyzing and visualizing TSNE algorithm.</dd>
      </dl><dl><dt>
          
          <code><a href="umap_analysis.html">umap_analysis()</a></code> 
        </dt>
        <dd>UMAP analysis for analyzing RNA-Seq data.</dd>
      </dl><dl><dt>
          
          <code><a href="umap_plot.html">umap_plot()</a></code> 
        </dt>
        <dd>UMAP plot for analyzing and visualizing UMAP algorithm.</dd>
      </dl><dl><dt>
          
          <code><a href="dendro_plot.html">dendro_plot()</a></code> 
        </dt>
        <dd>Dendrograms for multiple samples/groups clustering.</dd>
      </dl></div><div class="section level2">
      <h2 id="-differential-expression-analyais">③Differential Expression Analyais<a class="anchor" aria-label="anchor" href="#-differential-expression-analyais"></a></h2>
      
      <p class="section-desc"></p><p>Analysis and visualization based on gene expression data and differentially expressed genes.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="venn_plot.html">venn_plot()</a></code> 
        </dt>
        <dd>Venn plot for stat common and unique gene among multiple sets.</dd>
      </dl><dl><dt>
          
          <code><a href="upsetr_plot.html">upsetr_plot()</a></code> 
        </dt>
        <dd>UpSet plot for stat common and unique gene among multiple sets.</dd>
      </dl><dl><dt>
          
          <code><a href="flower_plot.html">flower_plot()</a></code> 
        </dt>
        <dd>Flower plot for stat common and unique gene among multiple sets.</dd>
      </dl><dl><dt>
          
          <code><a href="volcano_plot.html">volcano_plot()</a></code> 
        </dt>
        <dd>Volcano plot for visualizing differentailly expressed genes.</dd>
      </dl><dl><dt>
          
          <code><a href="ma_plot.html">ma_plot()</a></code> 
        </dt>
        <dd>MversusA plot for visualizing differentially expressed genes.</dd>
      </dl><dl><dt>
          
          <code><a href="heatmap_group.html">heatmap_group()</a></code> 
        </dt>
        <dd>Heatmap group for visualizing grouped gene expression data.</dd>
      </dl><dl><dt>
          
          <code><a href="circos_heatmap.html">circos_heatmap()</a></code> 
        </dt>
        <dd>Circos heatmap plot for visualizing gene expressing in multiple samples.</dd>
      </dl><dl><dt>
          
          <code><a href="chord_plot.html">chord_plot()</a></code> 
        </dt>
        <dd>Chord plot for visualizing the relationships of pathways and genes.</dd>
      </dl></div><div class="section level2">
      <h2 id="-advanced-analysis">④Advanced Analysis<a class="anchor" aria-label="anchor" href="#-advanced-analysis"></a></h2>
      
      <p class="section-desc"></p><p>Advanced analysis of Transcriptome includes exploring gene expression trends of multiple groups of samples, constructing co expression modules for genes, and exploring genes related to traits in expression modules.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="gene_rank_plot.html">gene_rank_plot()</a></code> 
        </dt>
        <dd>Gene ranking dotplot for visualizing differentailly expressed genes.</dd>
      </dl><dl><dt>
          
          <code><a href="gene_cluster_trend.html">gene_cluster_trend()</a></code> 
        </dt>
        <dd>Gene cluster trend plot for visualizing gene expression trend profile in multiple samples.</dd>
      </dl><dl><dt>
          
          <code><a href="trend_plot.html">trend_plot()</a></code> 
        </dt>
        <dd>Trend plot for visualizing gene expression trend profile in multiple traits.</dd>
      </dl><dl><dt>
          
          <code><a href="wgcna_pipeline.html">wgcna_pipeline()</a></code> 
        </dt>
        <dd>WGCNA analysis pipeline for RNA-Seq.</dd>
      </dl><dl><dt>
          
          <code><a href="network_plot.html">network_plot()</a></code> 
        </dt>
        <dd>Network plot for analyzing and visualizing relationship of genes.</dd>
      </dl><dl><dt>
          
          <code><a href="heatmap_cluster.html">heatmap_cluster()</a></code> 
        </dt>
        <dd>Heatmap cluster for visualizing clustered gene expression data.</dd>
      </dl></div><div class="section level2">
      <h2 id="-go-and-kegg-enrichment">⑤GO and KEGG Enrichment<a class="anchor" aria-label="anchor" href="#-go-and-kegg-enrichment"></a></h2>
      
      <p class="section-desc"></p><p>GO and KEGG enrichment analysis and visualization.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="go_enrich.html">go_enrich()</a></code> 
        </dt>
        <dd>GO enrichment analysis based on GO annotation results (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="go_enrich_stat.html">go_enrich_stat()</a></code> 
        </dt>
        <dd>GO enrichment analysis and stat plot (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="go_enrich_bar.html">go_enrich_bar()</a></code> 
        </dt>
        <dd>GO enrichment analysis and bar plot (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="go_enrich_dot.html">go_enrich_dot()</a></code> 
        </dt>
        <dd>GO enrichment analysis and dot plot (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="go_enrich_net.html">go_enrich_net()</a></code> 
        </dt>
        <dd>GO enrichment analysis and net plot (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="kegg_enrich.html">kegg_enrich()</a></code> 
        </dt>
        <dd>KEGG enrichment analysis based on KEGG annotation results (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="kegg_enrich_bar.html">kegg_enrich_bar()</a></code> 
        </dt>
        <dd>KEGG enrichment analysis and bar plot (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="kegg_enrich_dot.html">kegg_enrich_dot()</a></code> 
        </dt>
        <dd>KEGG enrichment analysis and dot plot (None/Exist Reference Genome).</dd>
      </dl><dl><dt>
          
          <code><a href="kegg_enrich_net.html">kegg_enrich_net()</a></code> 
        </dt>
        <dd>KEGG enrichment analysis and net plot (None/Exist Reference Genome).</dd>
      </dl></div><div class="section level2">
      <h2 id="-tables-operations">⑥Tables Operations<a class="anchor" aria-label="anchor" href="#-tables-operations"></a></h2>
      
      <p class="section-desc"></p><p>Quickly manipulate specific Transcriptome data structures.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="table_split.html">table_split()</a></code> 
        </dt>
        <dd>Table split used for splitting a grouped column to multiple columns.</dd>
      </dl><dl><dt>
          
          <code><a href="table_merge.html">table_merge()</a></code> 
        </dt>
        <dd>Table merge used to merge multiple variables to on variable.</dd>
      </dl><dl><dt>
          
          <code><a href="table_filter.html">table_filter()</a></code> 
        </dt>
        <dd>Table filter used to filter row by column condition.</dd>
      </dl><dl><dt>
          
          <code><a href="table_cross.html">table_cross()</a></code> 
        </dt>
        <dd>Table cross used to cross search and merge results in two tables.</dd>
      </dl></div><div class="section level2">
      <h2 id="example-datasets-in-tomicsvis">Example datasets in TOmicsVis<a class="anchor" aria-label="anchor" href="#example-datasets-in-tomicsvis"></a></h2>
      
      <p class="section-desc"></p><p>Example datasets in TOmicsVis.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="weight_sex.html">weight_sex</a></code> 
        </dt>
        <dd>Weight and Sex traits dataframe.</dd>
      </dl><dl><dt>
          
          <code><a href="traits_sex.html">traits_sex</a></code> 
        </dt>
        <dd>Length, Width, Weight, and Sex traits dataframe.</dd>
      </dl><dl><dt>
          
          <code><a href="survival_data.html">survival_data</a></code> 
        </dt>
        <dd>Survival data as example data for survival_plot function.</dd>
      </dl><dl><dt>
          
          <code><a href="gene_expression.html">gene_expression</a></code> 
        </dt>
        <dd>All genes in all samples expression dataframe of RNA-Seq.</dd>
      </dl><dl><dt>
          
          <code><a href="gene_expression2.html">gene_expression2</a></code> 
        </dt>
        <dd>Shared DEGs of all paired comparisons in all samples expression dataframe of RNA-Seq.</dd>
      </dl><dl><dt>
          
          <code><a href="gene_expression3.html">gene_expression3</a></code> 
        </dt>
        <dd>Shared DEGs of all paired comparisons in all groups expression dataframe of RNA-Seq.</dd>
      </dl><dl><dt>
          
          <code><a href="samples_groups.html">samples_groups</a></code> 
        </dt>
        <dd>Samples and groups for gene expression.</dd>
      </dl><dl><dt>
          
          <code><a href="degs_lists.html">degs_lists</a></code> 
        </dt>
        <dd>Paired comparisons differentially expressed genes (degs) among groups.</dd>
      </dl><dl><dt>
          
          <code><a href="degs_stats.html">degs_stats</a></code> 
        </dt>
        <dd>All DEGs of paired comparison CT-vs-LT12 stats dataframe.</dd>
      </dl><dl><dt>
          
          <code><a href="degs_stats2.html">degs_stats2</a></code> 
        </dt>
        <dd>All DEGs of paired comparison CT-vs-LT12 stats2 dataframe.</dd>
      </dl><dl><dt>
          
          <code><a href="network_data.html">network_data</a></code> 
        </dt>
        <dd>Network data from WGCNA tan module top-200 dataframe.</dd>
      </dl><dl><dt>
          
          <code><a href="gene_go_kegg.html">gene_go_kegg</a></code> 
        </dt>
        <dd>GO and KEGG annotation of background genes.</dd>
      </dl><dl><dt>
          
          <code><a href="gene_go_kegg2.html">gene_go_kegg2</a></code> 
        </dt>
        <dd>GO and KEGG annotation of background genes.</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Benben Miao, Wei Dong.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

